<!--
--- UCSF Chimera Copyright ---
Copyright (c) 2010-2015 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
--- UCSF Chimera Copyright ---
-->

<html><head>
<title>Chimera Interface to Modeller</title>
<style type="text/css">
A.nounder:link {text-decoration: none}
A.nounder:visited {text-decoration: none}
A.nounder:hover {text-decoration: underline; color: red;}
</style>
</head><body>

<a name="top"></a>
<h2>Chimera Interface to Modeller</h2>
<p>
Chimera provides a graphical interface to running the program
<a href="https://www.salilab.org/modeller/" target="_blank">Modeller</a>,
either locally or via a web service hosted by the
<a href="https://www.cgl.ucsf.edu/Overview/index.html" target="_blank">UCSF
RBVI</a>. Two types of calculations are available:
<ul>
<li><a href="#comparative"><b>Comparative (homology) modeling</b></a>.
Theoretical models of a protein are generated using at least one known related 
structure and a sequence alignment of the known and unknown structures.
The protein to be modeled is the <b><i>target</i></b>, and a related 
known structure used for modeling is a <b><i>template</i></b>.
The inputs for comparative modeling can be generated in
<a href="multalignviewer.html#approaches">several ways</a>.
<br><br>
<li><a href="#building"><b>Building</b></a> 
parts of a protein without using a template.
Missing segments can be built <i>de novo</i>, or existing segments
<b><i>refined</i></b> by generating additional possible conformations.
Parts that need building or refinement are often loop regions.
</ul>
Modeller is developed by the 
<a href="https://www.salilab.org" target="_blank">Sali Lab</a>.
Use of Modeller, whether a previously
<a href="https://www.salilab.org/modeller/download_installation.html"
target="_blank">downloaded</a> copy or via web service, 
requires a <a name="modeller-license"><b><i>license key</i></b></a>.
Academic users can <a href="https://www.salilab.org/modeller/registration.html"
target="_blank">register</a> free of charge to receive a license key. 
(Commercial entities and government research labs, please see
<a href="https://www.salilab.org/modeller/accelrys.html" 
target="_blank">Modeller licensing</a>.) Modeller users should cite:
<blockquote>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/8254673" target="_blank">
Comparative protein modelling by satisfaction of spatial restraints.</a>
&Scaron;ali A, Blundell TL.
<i>J Mol Biol.</i> 1993 Dec 5;234(3):779-815.
</blockquote>
See also: 
<a href="../editing/editing.html"><b>Build Structure</b></a>,
<a href="../rotamers/rotamers.html"><b>Rotamers</b></a>,
<a href="framemav.html" target="_top"><b>Multalign Viewer</b></a>,
<a href="../../UsersGuide/midas/mda.html"><b>mda</b></a>,
<a href="../../UsersGuide/modbase.html">ModBase fetch</a>,
<a href="../../UsersGuide/savemodel.html">modifying and saving data</a>, the
<a href="../../UsersGuide/tutorials/dor.html">Comparative Modeling tutorial</a>
</p>

<h3><a href="#top" class="nounder">&larr;</a>
<a name="comparative">Comparative Modeling with Modeller</a></h3>
<p>
Comparative modeling requires a structure to serve as a template,
and a target-template sequence alignment.
There are <a href="multalignviewer.html#approaches">several ways</a>
to obtain these inputs using Chimera.  Sequence alignments are shown in 
<a href="framemav.html" target="_top"><b>Multalign Viewer</b></a>.
The Chimera interface to comparative modeling with Modeller
can be started by choosing
<a href="multalignviewer.html#mavmenu-structure"><b>Structure...
Modeller (homology)</b></a> from the <b>Multalign Viewer</b> menu,
or called with <a href="../../UsersGuide/midas/mda.html"><b>mda</b></a>.
With this interface,
*<b>only a single chain or subunit can be modeled at a time</b>.*
Modeling a multimer or complex requires running Modeller outside of Chimera.
<ul>
<li style="list-style-type:none"><b>Choose the target (sequence to be modeled)</b>
- the name of the target sequence should be chosen from the pulldown menu
of all sequences in the current alignment
<br><br>
<li style="list-style-type:none"><b>Choose at least one template</b>
- at least one structure to use as a template should be chosen from the table,
by clicking or dragging with the left mouse button to highlight the
corresponding rows.  <b>Ctrl</b>-click toggles the status of a single row. 
The choices are actually the sequences in the alignment, and 
choosing a sequence indicates using its associated structure(s).
Some of the columns are blank initially, but clicking 
<a href="#fetch"><b>Fetch Structures/Annotations</b></a> 
loads the structures into Chimera and fills in the table, where possible.
The table can be sorted by the contents of any column by clicking the
column header:
  <ul>
  <li><b>Sequence</b> - sequence name; 
  choosing a sequence indicates using its associated structure as a template
  (the table lists all of the sequences in the alignment except the target)
  <li><b>Structure ID</b> - identifier for the structure associated 
  with the sequence, if any; usually a 4-character PDB ID, but could be a 
  <a href="../../UsersGuide/scop.html">SCOP domain ID</a>
  (which includes a PDB ID)
  <li><b>%ID</b> - percent sequence identity of the sequence as compared to
  the target, computed from the alignment
  <li><b>Title</b> - title of structure (from PDB entry)
  <li><b>Organism</b> - source organism of structure (from PDB entry)
  </ul>
Clicking <a name="fetch"><b>Fetch Structures/Annotations</b></a>:
  <ol>
  <li>fetches the structure for any sequence not already associated with a
  structure, if that structure can be deduced from the sequence name (as in the 
  <a href="multalignviewer.html#mavpref-structure"><b>Structure</b>
  preferences</a> in
  <a href="framemav.html" target="_top"><b>Multalign Viewer</b></a>)
  <li>uses the structure IDs to look up additional information
  </ol> 
Before comparative modeling with multiple templates fetched by
<a href="../../UsersGuide/midas/mda.html"><b>mda</b></a>, use the command
<b>reset overlay</b> to position the template structures for best results.
<br><br>
<li style="list-style-type:none">Choose where to run Modeller:
<br><br>
  <ul>
  <li><a name="runweb"><b>Run Modeller via web service</b></a>
    <ul>
    <li><b>Modeller license key</b> 
    - a <a href="#modeller-license">license key</a> is required 
    to run the program; the <a href="https://www.salilab.org/modeller/" 
    target="_blank">Modeller home page</a> includes links
    to register for a key and to download the program
    </ul>
  <li><a name="runlocal"><b>Run Modeller locally</b></a>
    <ul>
    <li><b>Location of Modeller executable</b>
    - the location of the Modeller executable file; 
    the <a href="#modeller-license">license key</a> will
    have been entered somewhere already during local installation
    <li><b>Modeller script file (optional, overrides dialog)</b>
    - use the specified Modeller script to control the calculation; this
    will override the settings in the dialog.  The script corresponding
    to the current dialog settings can be viewed in
    <a href="../idle/idle.html"><b>IDLE</b></a> by clicking 
    <b>Get Current Modeller Script</b>, saved to a file using the IDLE menu,
    and edited by hand as desired. For more details on scripting Modeller, 
    consult the <a href="https://www.salilab.org/modeller/manual/manual.html"
    target="_blank">Modeller manual</a>.
    </ul>
  </ul>
  <br>
<li style="list-style-type:none"><b>Advanced Options:</b>
<br><br>
  <ul>
  <li><b>Number of output models [<i>N</i>] (max 1000)</b>
  (default <b>5</b>)
  <li><b>Include non-water HETATM residues from template</b>
  (off by default)
  - whether to include HETATM residues other than water (ligands, ions, 
  detergent molecules, <i>etc.</i>) from the template in the output models.
  This option will propagate all qualifying residues, even from multiple
  templates; those not desired in the output should be
  <a href="../editing/editing.html#deletion">deleted</a> 
  from the template(s) beforehand.
  <li><b>Include water molecules from template</b>
  (off by default)
  - whether to include water residues from the template in the output models.
  This option will propagate all qualifying residues, even from multiple
  templates; those not desired in the output should be
  <a href="../editing/editing.html#deletion">deleted</a> 
  from the template(s) beforehand.
  <li><b>Build models with hydrogens (warning: slow)</b>
  (off by default)
  <li><b>Use fast/approximate mode (produces only one model)</b>
  (off by default)
  - use fast/approximate mode (~3 times faster) to get a rough idea of 
  model appearance or to confirm that the alignment is reasonable.  
  This mode does not randomize the starting structure (generates only a
  single model) and performs very little optimization of the target function.
  <li><b>Use thorough optimization (recommended with MDA)</b>
  - optimize more thoroughly than the default, as recommended for modeling
  large targets with <b>M</b>ulti<b>D</b>omain <b>A</b>ssembler
  (the <a href="../../UsersGuide/midas/mda.html"><b>mda</b></a> command)
  <li><b>Temporary folder location (optional)</b>
  - use the specified location for temporary files; otherwise, a
  location will be generated automatically
  <li><b>Distance restraints file (optional)</b>
  - Specify an input file containing distance restraints 
  (see example file <a href="distres.txt">distres.txt</a>). 
  Each line in the file should be of the format:
  <blockquote>
  <i>res1 &nbsp;res2 &nbsp;dist &nbsp;stdev</i>
  </blockquote>
  where <i>res1</i> and <i>res2</i> are residue numbers or 
  ranges of residue numbers in the target sequence, <i>dist</i> is the 
  distance in &Aring;, and <i>stdev</i> is the standard deviation. 
  If a single residue is specified, its C&alpha; will be used to 
  anchor the restraint. If a residue range (<i>e.g.</i> 233-275) is
  specified, the range's center of mass will be used to anchor the restraint.
  </ul>
</ul>
<b>OK</b> starts the calculation and dismisses the panel,
while <b>Apply</b> starts the calculation without dismissing the panel.
<b>Close</b> dismisses the panel without performing any calculation.
<b>Help</b> brings up this manual page in a browser window.
</p><p>
Running Modeller is a background task.  
Clicking the information icon <img src="../../UsersGuide/taskicon.png"> 
in the Chimera <a href="../../UsersGuide/chimerawindow.html#statusline">status 
line</a> will bring up the <a href="../idle/task.html"><b>Task Panel</b></a>,
in which the job can be canceled if desired.
</p><p>
After the calculation has finished, the comparative models are opened 
in Chimera and can be <a href="../../UsersGuide/savemodel.html">saved</a>
in the usual ways.  The models are automatically superimposed onto the template 
(or the lowest-numbered of multiple templates) using 
<a href="../../UsersGuide/midas/mmaker.html"><b>matchmaker</b></a>
defaults, and the view is focused on that template. 
The models are <a href="multalignviewer.html#association">associated</a> 
with the target sequence, and the <b>RMSD</b>
<a href="multalignviewer.html#headers">header</a> displayed in the
<a href="multalignviewer.html#seqwindow">sequence window</a>.
Model scores are shown in a
<a href="../../UsersGuide/modbase.html#modellist"><b>Model List</b></a>,
the same dialog used
for comparative models <a href="../../UsersGuide/fetch.html">fetched</a>
from <a href="https://modbase.compbio.ucsf.edu/modbase-cgi/index.cgi"
target="_blank">ModBase</a>.  Loops or other parts of a model
can be subjected to further <a href="#building">refinement</a>.
</p><p>
Running Modeller with identical inputs on different machines may give
different (but equally valid) results, due to small numerical differences
that can lead to finding different local optima of the 
modeling objective function.
</p>

<h3><a href="#top" class="nounder">&larr;</a>
<a name="building">Building/Refinement with Modeller</a> (Model/Refine Loops 
<img src="modloops.png" alt="Model/Refine Loops icon">&nbsp;)</h3>
<p>
The only required input for Modeller building or refinement 
is a protein structure.  
Missing segments can be built <i>de novo</i>, or existing segments
<b><i>refined</i></b> by generating additional possible conformations.
Building and refinement can be applied to
protein structures regardless of whether they were
<a href="#comparative">modeled</a> or determined experimentally.
</p><p>
The Chimera interface to Modeller for building or refinement can be accessed
by <a href="../../UsersGuide/extension.html">starting</a>
<b>Model/Refine Loops</b>, a tool in the <b>Structure Editing</b> category, or
by choosing <a href="multalignviewer.html#mavmenu-structure"><b>Structure...
Modeller (loops/refinement)</b></a> from the
<b>Multalign Viewer</b> (<b>Sequence</b> tool) menu.
Using <b>Model/Refine Loops</b> is equivalent to using the
<a href="framemav.html" target="_top"><b>Sequence</b></a> tool
to show the sequence of a chain, then using its menu to show the interface.
If the structure chain is
<a href="multalignviewer.html#association">associated</a> with a sequence in an
alignment in <a href="framemav.html" target="_top"><b>Multalign Viewer</b></a>,
however, the alignment sequence can be used instead of the individual sequence.
<ul>
<li style="list-style-type:none">
The output models will include any water molecules and other 
HETATM residues from the input model; those not desired in the output 
should be <a href="../editing/editing.html#deletion">deleted</a> 
from the input beforehand.
<br><br>
<li><b>Model/remodel</b> - which parts of the input model to build or refine; 
they will be considered collectively even if composed of multiple
noncontiguous sequence segments
<br><br>
  <ul>
  <li><b>active region</b> - residues in the
  <a href="multalignviewer.html#activeregion">active region</a></b> in
  <a href="framemav.html" target="_top"><b>Multalign Viewer</b></a>
  (the region most recently drawn or clicked); this
  <a href="multalignviewer.html#regions">region</a></b> 
  must contain only residues from one sequence, and the model 
  <a href="multalignviewer.html#association">associated</a> with that sequence 
  will be used as the input 
  <li><b>Chimera selection region</b> - the currently
  <a href="../../UsersGuide/selection.html">selected</a> amino acid residues,
  as represented by the <b>Chimera selection</b> region in
  <a href="framemav.html" target="_top"><b>Multalign Viewer</b></a>; this
  <a href="multalignviewer.html#regions">region</a></b>
  must contain only residues from one sequence, and the model 
  <a href="multalignviewer.html#association">associated</a> with that sequence 
  will be used as the input
  <li><b>non-terminal missing structure</b>
   - <b><i>missing structure</i></b> refers to residues that appear in PDB 
  SEQRES records but are missing from the coordinates section of a PDB file.
  The model of interest should be chosen from the pulldown menu on the right.
  By default, missing structure is indicated in the sequence
  with red outline boxes, corresponding to the <b>missing structure...</b> 
  <a href="multalignviewer.html#regions">region</a></b> for the model.
  The <b><i>non-terminal</i></b> missing structure consists of only the 
  missing segments that are constrained at both ends by existing structure.
  <p>
  ** Note: if the input model contains a non-terminal missing segment that
  the user opts not to build, inevitably Modeller will bring the
  existing structure on either side together to &ldquo;close the gap&rdquo;
  in the output models. **
  </p>
  <li><b>all missing structure</b>
  - all missing segments, including N-terminal and C-terminal parts
  only constrained at one end by existing structure; 
  all residues in the <b>missing structure...</b> 
  <a href="multalignviewer.html#regions">region</a></b> for the
  model chosen from the pulldown menu on the right
  </ul>
<br>
<li><b>Allow this many residues adjacent to missing regions to move</b> 
(default <b>1</b>) - if one of the <b>missing structure</b> options is used,
how many existing residues at each missing/existing junction 
to allow to move relative to the input structure
<li><b>Number of models to generate</b> (default
<b>5</b>)
<li><b>Loop modeling protocol</b> 
  <ul>
  <li><b>standard</b> (default)
  <li><b>DOPE</b> - Discrete Optimized Protein Energy score (see
<a href="https://www.ncbi.nlm.nih.gov/pubmed/17075131" target="_blank">Shen
and Sali</a>, <i>Protein Sci</i> <b>15</b>:2507 (2006)) with Lennard-Jones
  potential and GB/SA implicit solvent interaction; generally higher quality
  but more computationally expensive than the standard method, and more
  prone to calculation failure (resulting in fewer models than requested)
  <li><b>DOPE-HR</b> - same as DOPE, except higher precision
  </ul>
<li><b>Run modeller using</b>
  <ul>
  <li><b>web service</b> (default)
  <li><b>local installation</b>
  </ul>
If web service:
   <ul>
   <li><b>Modeller license key</b> 
    - a <a href="#modeller-license">license key</a> is required 
    to run the program; the <a href="https://www.salilab.org/modeller/" 
    target="_blank">Modeller home page</a> includes links
    to register for a key and to download the program
    </ul>
If local installation:
    <ul>
    <li><b>Location of Modeller executable</b>
    - the location of the Modeller executable file; 
    the <a href="#modeller-license">license key</a> will
    have been entered somewhere already during local installation
    <li><b>Custom Modeller script file (optional, overrides dialog)</b>
    - use the specified Modeller script to control the calculation; this
    will override the settings in the dialog.  The script corresponding
    to the current dialog settings can be viewed in
    <a href="../idle/idle.html"><b>IDLE</b></a> by clicking 
    <b>Get Current Modeller Script</b>, saved to a file using the IDLE menu,
    and edited by hand as desired. For more details on scripting Modeller, 
    consult the <a href="https://www.salilab.org/modeller/manual/manual.html"
    target="_blank">Modeller manual</a>.
    </ul>
  <li><b>Temporary folder location (optional)</b>
  - use the specified location for temporary files; otherwise, a
  location will be generated automatically
  </ul>
</ul>
<b>OK</b> starts the calculation and dismisses the panel,
while <b>Apply</b> starts the calculation without dismissing the panel.
<b>Close</b> dismisses the panel without performing any calculation.
<b>Help</b> brings up this manual page in a browser window.
</p><p>
Running Modeller is a background task.  
Clicking the information icon <img src="../../UsersGuide/taskicon.png"> 
in the Chimera <a href="../../UsersGuide/chimerawindow.html#statusline">status 
line</a> will bring up the <a href="../idle/task.html"><b>Task Panel</b></a>,
in which the job can be canceled if desired.
</p><p>
After the calculation has finished, the models (each including the
unchanged parts of the protein in addition to what was built or refined) 
are opened in Chimera and can be 
<a href="../../UsersGuide/savemodel.html">saved</a> in the usual ways.  
The models are <a href="multalignviewer.html#association">associated</a> 
with the sequence, and the <b>RMSD</b>
<a href="multalignviewer.html#headers">header</a> displayed in the
<a href="multalignviewer.html#seqwindow">sequence window</a>.
Model scores are shown in a
<a href="../../UsersGuide/modbase.html#modellist"><b>Model List</b></a>,
the same dialog used
for comparative models <a href="../../UsersGuide/fetch.html">fetched</a>
from <a href="https://modbase.compbio.ucsf.edu/modbase-cgi/index.cgi"
target="_blank">ModBase</a>.
</p>
<hr>
<address>UCSF Computer Graphics Laboratory / July 2015</address>
</body></html>
