<!--
--- UCSF Chimera Copyright ---
Copyright (c) 2002-2015 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
--- UCSF Chimera Copyright ---
-->

<html>
<head>
<TITLE>Multalign Viewer (Sequence)</TITLE>
</head>
<body>
<a name="top">
<h3>Multalign Viewer <img src="mulign.png" alt="Multalign Viewer icon">
&nbsp;&nbsp;(Sequence <img src="seqicon.png" alt="Sequence icon">&nbsp;)</h3>
</a>

<table border="1px" align="right" style="margin:4px 8px 8px 4px"
cellpadding="0" cellspacing="0">
<tr><td><a href="../matchalign/match3seq.png">
<img src="../matchalign/match3seq.png"
width="340px" title="click for larger image..."></a></td></tr>
</table>

<p>
<b>Multalign Viewer</b> shows amino acid and nucleotide sequences:
<ul>
<li>multiple sequence alignments
  <ul>
  <li>read from files
(<a href="../../UsersGuide/filetypes.html#alignment">several formats</a>)
  <li>created by other tools in Chimera
(<a href="../matchmaker/matchmaker.html"><b>MatchMaker</b></a>,
<a href="../matchalign/matchalign.html"><b>Match&nbsp;->&nbsp;Align</b></a>,
<a href="../align/align.html"><b>Align&nbsp;Chain&nbsp;Sequences</b></a>)
  <li>pseudo-multiple alignments from
<a href="../../UsersGuide/blast.html"><b>Blast Protein</b></a>
  </ul>
<br>
<li>individual sequences
  <ul>
  <li>of structures in Chimera (when started as <b>Sequence</b>)
  <li>read from files 
   (<a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp#search"
  target="_blank">FASTA</a>,
  <a href="https://www.bioinformatics.nl/tools/crab_pir.html" 
  target="_blank">NBRF/PIR</a> formats)
  <li><a href="../../UsersGuide/fetch.html">fetched</a> from 
  <a href="https://www.uniprot.org/" target="_blank">UniProt</a>
  </ul>
</ul>
The sequences are automatically <a href="#association">associated</a>
with structures in Chimera.  Associated structures are not required,
but association enables useful sequence-structure 
<a href="#crosstalk">crosstalk</a> such as bidirectional selection.
Measures of sequence conservation and spatial variability can be shown
as <a href="#headers">headers</a> above the sequences, and 
sequence feature <a href="#annotations">annotations</a> from
<a href="https://www.uniprot.org/" target="_blank">UniProt</a> 
can be shown as colored boxes.
An <a href="#modeller">interface to Modeller</a> allows 
comparative (homology) modeling using a template structure,
as well as building missing segments without a template.
<b>Multalign Viewer</b> status and alignment data,
but not <a href="#mavprefs">preference settings</a>,
are included in saved
<a href="../../UsersGuide/sessions.html#sesdef">sessions</a>.
</p><p>
See also:
<a href="../../UsersGuide/frametut.html" target="_top">Chimera tutorials</a>
(in particular,
<a href="../../UsersGuide/tutorials/super.html">Sequences and Structures</a>,
<a href="../../UsersGuide/tutorials/alignments.html">Superpositions and
Alignments</a>,
<a href="../../UsersGuide/tutorials/dor.html">Comparative Modeling</a>,
and on the Chimera website,
<a href="https://www.rbvi.ucsf.edu/chimera/data/tutorials/systems/outline.html"
target="_blank">Mapping Sequence Conservation</a>)
and the following reference:
<blockquote>
<a href="https://www.biomedcentral.com/1471-2105/7/339" target="_blank">
Tools for integrated sequence-structure analysis with UCSF Chimera.</a>
Meng EC, Pettersen EF, Couch GS, Huang CC, Ferrin TE.
<i>BMC Bioinformatics.</i> 2006 Jul 12;7:339.
</blockquote>
</p>

<a name="startup">
<h4>STARTUP AND INPUT</h4></a>
<p>
There are <a href="../../UsersGuide/extension.html">several ways to start</a>
<b>Multalign Viewer</b>, a tool in the <b>Sequence</b> category.
Explicitly starting <b>Multalign Viewer</b> brings up a
<a href="../../UsersGuide/opensave.html">dialog</a> for opening a
<a href="../../UsersGuide/filetypes.html#alignment">sequence alignment</a>.
In fact, simply <a href="../../UsersGuide/filetypes.html">opening</a>
a file in any of the registered 
<a href="../../UsersGuide/filetypes.html#alignment">sequence alignment 
formats</a> (without starting <b>Multalign Viewer</b> first) 
will automatically use it to show the alignment.
Several alignments can be open at the same time, each in a
separate <a href="#seqwindow">sequence window</a>.
</p><p>
<a name="individual">
Individual sequences of structures in Chimera</a> can be shown:
<ul>
<li>by <a href="../../UsersGuide/extension.html">starting</a> 
the <b>Sequence</b> tool (<b>Sequence</b> category)
<li>by using the command
<a href="../../UsersGuide/midas/listen.html#sequence"><b>sequence</b></a>
<li>by choosing one or more molecule models in the left
side of the <a href="../../UsersGuide/modelpanel.html"><b>Model Panel</b></a> 
and clicking the <b>sequence...</b> button in the right side of the panel
</ul>
If there is more than one bonded multiresidue chain in the model(s), the chains 
will be listed in a dialog for designating which sequences to <b>Show</b>.  
The sequence of each chosen chain will be shown in a separate
<a href="#seqwindow">window</a>, including any residues not in
the structure itself but in SEQRES records in the input structure file.
Residues with standard names are assigned the corresponding standard
one-letter codes, while certain nonstandard residues are assigned the
same one-letter codes as closely related standard residues:
<ul>
<li> ASH (protonated ASP) = D
<li> CYX (disulfide-bonded CYS) = C
<li> GLH (protonated GLU) = E
<li> HID/HIE/HIP (different protonation states of HIS) = H
<li> HYP (hydroxyproline) = P
<li> MSE (selenomethionine) = M
<li> +N (methylated nucleic acid) = N, where N = A/C/G/T/U
<li> additional relationships described in MODRES records in PDB input files
</ul>
In addition, the two unusual genetically encoded amino acids 
are equivalenced as follows for alignment scoring with
amino acid similarity matrices:
<ul>
<li>U (selenocysteine) is scored as C
<li>O (pyrrolysine) is scored as X
</ul>
</p>

<a name="seqwindow">
<h4>THE SEQUENCE WINDOW</h4></a>

<table border="1" align="right" style="margin:8px 8px 8px 8px"
cellpadding="0" cellspacing="0">
<tr><td>
<img src="multal.png" width="390px" 
title="sequence window showing two regions">
</td></tr>
</table>
<p>
The figure shows part of the sequence window contents for the input
file <a href="apoex.fa">apoex.fa</a>.
In addition to the sequences and their <a href="#seqnames">names</a>, 
<a href="#numbering">numbering</a> and <a href="#headers">header lines</a>
may be displayed.  Font size, sequence wrapping behavior, and
the coloring of residue one-letter codes can be
controlled in the <a href="#mavpref-layout"><b>Appearance</b> preferences</a>.
Colored boxes enclosing one or more residues are called
<a href="#regions">alignment regions</a>; double-clicking a region 
opens the <a href="#regionbrowser"><b>Region Browser</b></a>.
The <a href="#mavmenu-edit"><b>Edit</b> menu</a> allows
<a href="#seqsearch">searching</a> for a string or pattern of residues 
in the sequences, or by full or partial sequence name.
</p><p>
When the mouse focus is in the sequence window,
the <b>Page Down</b> key (or <b>space</b>) moves the view down
to start with the block below the topmost block whose beginning
is currently visible;
<b>Page Up</b> (or <b>Shift</b>-<b>space</b>) moves the view up
to start with the block above the topmost block whose beginning is
currently visible.
Vertical alignment scrolling can also be done with the mouse wheel.
</p><p>
Mousing over sequence names and residues shows information near the bottom 
of the window. Mousing over a sequence name reports any
<a href="#association">associated structure(s)</a>, the vertical position 
in the alignment (sequence is Nth of M sequences), and sequence length.
Mousing over a residue symbol reports the corresponding residue in
associated structure(s), if any, otherwise the sequence position.  
Reporting the position in non-structure-associated sequences is turned off 
for larger alignments (&gt;250,000 characters).
The system &ldquo;copy&rdquo; shortcut (for example, command-C on Mac) 
will copy the contents of the <a href="#activeregion">active region</a></b>, 
if any, otherwise the entire sequence(s), into the text buffer.
</p><p>
The <b>Hide</b> button closes the sequence window without changing
the state of <b>Multalign Viewer</b>.
The sequence window can be reinvoked using the <b>Raise</b> option
for the <b>Multalign Viewer</b>
<a href="../../UsersGuide/instances.html">instance in the
<b>Tools</b> menu</a> (abbreviated <b>MAV</b>).
This is also useful when the window has become obscured by other windows.
<b>Help</b> opens this manual page in a browser window,
and <b>Quit</b> exits from <b>Multalign Viewer</b>.
</p>

<a name="seqnames">
<h4>SEQUENCE NAMES</h4></a>
<p>
Initial sequence names are taken from the input file or based on the
corresponding structure chain. A sequence name can be changed by choosing
<a href="#editname"><b>Edit... Edit Sequence Name</b></a> from the
<b>Multalign Viewer</b> menu, and in the resulting dialog,
indicating the desired sequence and entering a new name.
Double-clicking a name in the <a href="#seqwindow">sequence window</a> 
brings up the same dialog, already set to that sequence. 
<a href="#namesearch"><b>Edit... Find Sequence Name</b></a> 
allows searching by full or partial sequence name.
In the <a href="#seqwindow">sequence window</a>, ellipses (...)
are used to shorten names longer than the maximum
(default 30 characters) specified in the
<a href="#mavpref-layout"><b>Appearance</b></a> section of the 
<a href="#mavprefs"><b>Multalign Viewer</b> preferences</a>.
</p>

<a name="numbering">
<h4>NUMBERING</h4></a>
<p>
Numbering can be displayed over the alignment and to the left and/or right
of the sequences.  Numbering displays can be controlled with the
<a href="#mavmenu-numberings"><b>Numberings</b> menu</a>.
</p><p>
<b><i>Alignment numbering</i></b> (for the alignment as a whole)
includes gap positions and always starts at 1.  
Whether it should be shown initially can be set in the
the <a href="#mavpref-analysis"><b>Headers</b> preferences</a>.
When present, alignment numbering is shown above any
<a href="#headers">headers</a>. 
</p><p>
<b><i>Sequence numbering</i></b> (for individual sequences)
includes only non-gap positions and can start with numbers other than 1.  
Sequence start numbers are set to 1 when an alignment is opened, except:
<ul>
<li>when a sequence is generated from a structure chain in Chimera
(for example, with 
<a href="../matchalign/matchalign.html"><b>Match -> Align</b></a>),
its start number is set to the number of the first residue in the 
structure chain
<li>when all sequence names in the input file
are of the form <b><i>string/N&ndash;M</i></b>,
the trailing <b><i>/N&ndash;M</i></b> is stripped to generate
the <a href="#seqnames">names</a> shown in the 
<a href="#seqwindow">sequence window</a>, and the
sequence start numbers are set to the respective values of <b><i>N</i></b>
</ul>
Sequence start numbers can be changed arbitrarily with
<a href="#seqnum"><b>Numberings... Adjust Sequence Numberings</b></a>.
Even with a correct start number, however, a sequence's numbering 
will be incorrect following any missing internal residues (for example, 
in a sequence derived from a structure missing a loop).
When an alignment is <a href="#saving">saved</a>, 
<a name="append-num">
<b><i>/N&ndash;M</i></b> can be appended to each sequence name,</a>
where <b><i>N</i></b> is the sequence's start number and 
<b><i>M</i></b> is calculated from <b><i>N</i></b> and the total number
of residues in the sequence.  
</p>
<!--
Note added Oct 2012:
As structures associate/disassociate from sequences, if the numbering of 
the remaining associated structures agrees (i.e. starts with the same number) 
then the sequence numbering start is changed to correspond. If they disagree, 
the sequence numbering starts at 1. If no sequences remain associated, 
the number is unchanged from what it was before any disassociation.
An exception is the Sequence tool. The numbering will not be changed even if 
other structures are associated. However, if additional sequences are added,
the Sequence tool loses it "magic" status and the numbering rules above apply.
-->

<table border="1" align="right" style="margin:8px 8px 8px 8px"
cellpadding="0" cellspacing="0">
<tr><td><a href="customheaders.png"><img src="customheaders.png" 
width="400px" title="click for larger image..."></a></td></tr>
</table>

<a name="headers">
<h4>ALIGNMENT HEADERS</h4></a>
<p>
Headers are lines of information above the sequences in the
<a href="#seqwindow">sequence window</a>.
They can be hidden/shown with the checkboxes in the
<b>Multalign Viewer</b> <a href="#mavmenu-headers"><b>Headers</b> menu</a>.
Some headers are available automatically, and
custom headers can be <a href="defineheader.html">defined</a>
using a <a href="defineheader.html#headerformat">simple format</a>.
The <a href="#mavpref-analysis"><b>Headers</b> preferences</a> control 
how <b>Consensus</b> and <b>Conservation</b> header values are calculated.
</p><p>
A header may contain numerical quantities (shown as a histogram),
characters, or symbols.
If the values in a numeric header fall within the range of 0 to 1, 
they will be used directly as histogram bar heights.  
If the range extends below 0 or above 1, the values will be
converted into histogram bar heights ranging from 0 to 1 according to:
    <blockquote>
    for values &ge; 0,&nbsp;&nbsp;&nbsp; 1 &ndash; &frac12;(<i>e</i><sup>&ndash;
value</sup>)
    </blockquote><blockquote>
    for values &lt; 0,&nbsp;&nbsp;&nbsp; &frac12;(<i>e</i><sup>value</sup>)
    </blockquote>
However, the original values (not those used for histogram display) are
assigned as <a href="#mavAttributes">residue attributes</a>
and written out when header contents are <a href="#saveheader">saved</a>.
</p><p>
Headers may be <b><i>dynamic</i></b>, with values that update automatically 
when independent variables such as the sequence alignment or structure 
<a href="#association">associations</a> are changed, 
or <b><i>static</i></b>, with values that stay the same.
</p><p>
The following dynamic headers are available by default:
<ul>
<li><b>Consensus</b> - <a href="#consensus">consensus sequence</a>
as defined in the <a href="#mavpref-analysis"><b>Headers</b> preferences</a>
<li><b>Conservation</b> - <a href="#conservation">conservation</a>
as specified in the <a href="#mavpref-analysis"><b>Headers</b> preferences</a>
<li><b>Charge variation</b> 
- range of residue formal charges, assuming &ndash;1 for
D/E, +1 for H/K/R, and 0 for all other types
<li><a name="rmsd-headers">
RMSD (root-mean-square distance) per-column spatial variation among the 
<a href="#association">associated</a> structures, 
</a>
only calculated for columns with at least two associated structure residues:
  <ul>
  <li><b>RMSD: ca</b> - using <a href="#onepoint">one point per residue</a>,
  CA in amino acids, usually C4' in nucleic acids
  <li><b>RMSD: backbone</b> - based on peptide backbone atoms (N, CA, C, O) 
  in amino acids, minimal backbone (O5', C5', C4', C3', O3', P) in nucleic acids
  <li><b>RMSD: full</b> - for residues of the same name, based on all
  distances between atoms of the same name (without correcting for symmetries 
  between equivalent atoms with different names, <i>e.g.</i>,
  OD1 and OD2 in aspartic acid); if not all structure residues associated
  with the column have the same name, falls back to the <b>RMSD: backbone</b>
  calculation
  </ul>
See also: <a href="#assessment">superposition assessment</a>,
<a href="#mavAttributes">residue attributes</a>
</ul>
Arbitrary, user-defined headers can also be shown:
<ul>
<li> Static headers can be added by reading in a
<a href="defineheader.html#headerformat">header definition file</a>
with <a href="#mavmenu-headers"><b>Headers... Load</b></a>
in the <b>Multalign Viewer</b> menu.
Loading values for a header with the same name as an
existing header will create an additional header line,
not replace the values in the existing header line.
GC (Generic per-Column) <a href="#annotations">annotations</a>
in alignments in
<a href="https://en.wikipedia.org/wiki/Stockholm_format"
target="_blank">Stockholm format</a> are also treated as static headers.
<br><br>
<li> Dynamic headers can be defined with Python code.  
For examples, see the file <b>share/MAVHeader/ChimeraExtension.py</b> 
within the Chimera installation 
(it defines the <b>Charge variation</b> and <b>RMSD</b> headers 
described <a href="#headers">above</a>).
Rather than changing this file in the <b>MAVHeader</b> module, which might
be overwritten whenever a new version of Chimera is installed, users should
create an analogous file named <b>ChimeraExtension.py</b> 
in a different directory.
The <a href="../../UsersGuide/preferences.html#Tools"><b>Tools</b> 
preferences</a> can be used to add that directory's location 
(the directory above it) to the places Chimera should look for extensions.
</ul>
The following apply to headers other than 
<a href="#numbering">alignment numbering</a>:
<ul>
<li> Headers can be reordered (a header line is placed below the others
when turned on with the <a href="#mavmenu-headers"><b>Headers</b> menu</a>).
<li> Header line contents can be saved to a file
(<a href="#mavmenu-headers"><b>Headers... Save</b></a>
in the <b>Multalign Viewer</b> menu).
<li> Clicking within a header 
<a href="../../UsersGuide/selection.html">selects</a> any
structure residues <a href="#association">associated</a> 
with that column of the alignment.
<li> Values of headers are available as
<a href="#mavAttributes">residue attributes</a>
named <b>mav<i>HeaderName</i></b>
in any <a href="#association">associated</a> structures.
Numerical attributes will appear in the attribute lists of
<a href="../render/render.html"><b>Render/Select by Attribute</b></a>;
character attributes will be listed only in the
<a href="../render/render.html#select"><b>Select by Attribute</b></a> portion.
It may be necessary to 
<a href="../render/render.html#refreshing"><b>Refresh</b></a> 
the attribute menus or values in these tools to update them with any changes
in <b>Multalign Viewer</b> header information.
</ul>
</p>

<a name="regions">
<h4>ALIGNMENT REGIONS</h4></a>
<p>
Alignment <b><i>regions</i></b> are colored boxes or outlines
that enclose one or more residue symbols in the 
<a href="#seqwindow">sequence window</a>. 
A single region can contain
any number of disjoint and/or abutting rectangular blocks.
Pausing the cursor over a region (but not directly over a residue symbol)
shows the region name in a pop-up balloon
(see the <a href="#mavpref-regions"><b>Regions</b> preferences</a>),
and double-clicking a region opens the 
<a href="#regionbrowser"><b>Region Browser</b></a>.
</p><p>
Several actions in <b>Multalign Viewer</b> and other tools
generate regions, and they can also be created manually.
<a href="https://www.uniprot.org/" target="_blank">UniProt</a> sequence
feature annotations are loaded automatically as regions for sequences
<a href="../../UsersGuide/fetch.html">fetched</a> from UniProt,
and annotations from UniProt 
can be added to other sequences (those already open in Chimera) using
<a href="#mavmenu-tools"><b>Info... UniProt Annotations</b></a>.
</p><p>
The initial display of manually created regions and those created by
<a href="../../UsersGuide/selection.html">selection</a>
or <a href="#association">structure association</a> 
can be set in the <a href="#mavpref-regions"><b>Regions</b> preferences</a>.
The display and colors of existing regions can be controlled with the
<a href="#regionbrowser"><b>Region Browser</b></a>.
Be aware that <b><a href="#reorder">reordering the sequences</a>
will delete existing multisequence regions</b>.
</p><p>
<a name="manual">
A region can be created manually</a> by dragging with
the left mouse button within the <a href="#seqwindow">sequence window</a>.
Dragging downward into the following block highlights to the end of
the preceding block.
<b>Shift</b>-dragging with the left mouse button <b><i>adds to</i></b> the
<a href="#activeregion">active region</a>.
<b>Ctrl</b>-dragging <b><i>creates</i></b> a new region and makes it
the <a href="#activeregion">active region</a>.
</p><p>
<a name="activeregion">
The <b><i>active region</i></b> is the region most recently</a>
created manually, clicked on in the 
<a href="#seqwindow">sequence window</a>, or designated as 
<b>A</b>ctive in the <a href="#regionbrowser"><b>Region Browser</b></a>.
The corresponding parts of any <a href="#association">
associated</a> structures are highlighted in the main Chimera window by
becoming <a href="../../UsersGuide/selection.html">selected</a>.
Only one region can be active at a time.
In the <a href="#seqwindow">sequence window</a>, the active region is
indicated with a dashed outline; clicking the active region
deactivates it, and clicking a different region deactivates the
former active region and makes the new region active.
A region with no interior color
is only responsive to clicks on its borders.  
Where regions <a href="#overlap">overlap</a>,
only the highest is responsive to clicks.
</p><p>
<a name="regionbrowser">
The <b>Region Browser</b></a> can be opened by double-clicking a region in the 
<a href="#seqwindow">sequence window</a> or by choosing
<a href="#mavmenu-tools"><b>Info... Region Browser</b></a> 
from the <b>Multalign Viewer</b> menu.
</p>
<table border="1" style="margin:8px 8px 8px 8px"
cellpadding="0" cellspacing="0">
<tr><td><img src="multreg.png" width="580px"></td></tr>
</table>
<p>
Regions applicable to the <b>entire alignment</b> or to any of the
individual sequences in the alignment can be listed using the pulldown menu 
near the top of the dialog.  Regions applicable to the entire alignment 
include the <b>Chimera selection</b> and any hand-drawn regions, 
even if they only enclose parts of a single sequence. 
Regions applicable to an individual sequence include 
<a href="#annotations">UniProt annotations</a> 
and regions created by <a href="#association">structure association</a>. 
If <a href="#annotations">UniProt annotations</a> have been added,
checkboxes under <b>Data source</b> allow including/excluding specific sets
of regions from the listing.
</p><p>
<b>Region Browser</b> columns:
<ul>
<li><b>A</b>ctive - whether the region is the 
<a href="#activeregion">active region</a>
<li><b>S</b>hown - whether the region is shown
in the <a href="#seqwindow">sequence window</a> (however, it could
still be obscured by <a href="#overlap">overlapping</a> regions)
<li>region interior color 
(a <a href="../../UsersGuide/coloring.html#colorwell">color well</a>)
<li>region border color 
(a <a href="../../UsersGuide/coloring.html#colorwell">color well</a>)
<li><b>Name</b> - region name
<li><b>RMSD</b> - region <a href="#overall-rmsd">overall RMSD</a>,
if applicable (entire-alignment regions only) calculated using
<a href="#onepoint">one point per residue</a> (typically CA atoms)
<li><b>Start</b> - position in <a href="#numbering">alignment numbering</a>
of the first residue in the region (individual-sequence regions only)
<li><b>End</b> - position in <a href="#numbering">alignment numbering</a>
of the last residue in the region (individual-sequence regions only)
</ul>
<a name="chosenregions">
Clicking a row in the <b>Region Browser</b> highlights the line
and <b><i>chooses</i></b> the region</a>.
More than one region can be chosen at a time, and checkbox settings
specify whether <b>Choosing in table</b>
shows all chosen regions, hides other regions, and/or makes the last chosen
region the <a href="#activeregion">active region</a>.
</p><p>
If <b>Include gaps</b> is false, 
the <a href="#chosenregions">chosen region(s)</a> will be drawn 
to enclose only residues, not gap positions.
Although a given region may then appear as disjoint blocks,
it will still be a single region.
</p><p>
<a name="overlap">
Since regions may overlap, a region can be considered higher or
lower than another.</a> When the mouse focus is in the 
<a href="#seqwindow">sequence window</a>, the <b>up arrow</b> and
<b>down arrow</b> keys can be used to raise and lower the
<a href="#activeregion">active region</a>, respectively,
and pressing the <b>Delete</b> key will delete the 
<a href="#activeregion">active region</a>. Alternatively, 
<b>Region Browser</b> buttons can be used to act on the
<a href="#chosenregions">chosen region(s)</a>:
<ul>
<li><b>Raise</b> - place region on top (in front) of any overlapping 
regions in the <a href="#seqwindow">sequence window</a> and move it
to the top of the list in the <b>Region Browser</b> 
<li><b>Lower</b> - place region below (behind) any overlapping regions 
regions in the <a href="#seqwindow">sequence window</a> and move it 
to the bottom of the list in the <b>Region Browser</b> 
<li><b>Copy</b> - copy region to user-specified name
<li><b>Rename</b> - change region name
<li><b>Delete</b> - remove region; the <b>Chimera selection</b> region
cannot be deleted, but can contain zero residues
<li><b>Info</b> - report region contents in the 
<a href="../../UsersGuide/reply.html"><b>Reply Log</b></a>, 
in terms of <a href="#numbering">alignment numbering</a>, and positions
within each sequence (<a href="#numbering">sequence numbering</a>),
and residue numbers within any associated structures
</ul>
Many other operations create or modify regions, including:
<ul>
<li>sequence-structure <a href="#association">association</a>
<li><a href="#selregion">selection</a>
within sequence-associated structures
<li><a href="#seqsearch">searching</a>
for occurrences of a particular string or pattern of residues
<li>showing <a href="#ssregion">secondary structure</a>,
<i>actual</i> in structure-associated sequences,
<i>predicted</i> in sequences not associated with structures
<li>reading a 
<a name="seqsel">
sequence coloring format (SCF) file or seqsel file</a>
(<a href="#mavmenu-file"><b>File... Load SCF/Seqsel File</b></a>) 
generated by the program <a href="jevtrace.html">JEvTrace</a>.
The two <a href="jevtrace.html">formats</a>
are slightly different, but both contain information
for coloring regions in the sequence and in any <a href="#association">
associated</a> structures.
The file reader automatically determines which format is being used.
The formats are simple enough that users wishing to color residues alike
in sequence and structure (for any reason) may wish to
create a file manually and read it in by this route.
</ul>
</p>

<a name="trees">
<h4>TREES</h4></a>
<p>
A <a href="https://evolution.genetics.washington.edu/phylip/newicktree.html"
target="_blank">Newick-format</a>
tree corresponding to the alignment can be read and displayed to
the left of the sequences with 
<a href="#mavmenu-tree"><b>Tree... Load</b></a>
in the <b>Multalign Viewer</b> menu.
A newly loaded tree will replace any previously shown tree.
The tree can be hidden/shown with
<a href="#mavmenu-tree"><b>Tree... Show Tree</b></a>.
</p><p>
The sequence names in the tree must be the same as 
the <a href="#seqnames">sequence names</a> in the alignment, 
and neither set of names can be a subset of the other. 
If the order of sequences in the tree differs from that in the alignment, 
a dialog will appear, requesting permission to reorder the alignment 
sequences for clearer display (to avoid tree branch crossings).
<b>Reordering will delete multisequence <a href="#regions">regions</a></b>,
although the <b>Chimera selection</b> region
(if any) will subsequently reappear.
</p><p>
When the tree includes branch lengths, both solid and dotted lines 
are used for display, with horizontal solid line lengths proportional 
to branch lengths.  When the tree does not include branch lengths, 
only solid lines are used.
</p><p>
<a name="chosen-node">
Clicking a node in the tree makes it red and
chooses it for subsequent operations.</a>
<a href="#mavmenu-tree"><b>Tree... Extract Subalignment</b></a>
copies the sequences defined by the chosen node
into a separate <a href="#seqwindow">sequence window</a>.
The alignment of the sequences is kept exactly the same as in the
full alignment, including any all-gap columns, so that the subalignment 
is initially the same length as the full alignment.  If desired,
all-gap columns can be deleted with the 
<a href="#mavmenu-edit"><b>Edit</b> menu</a>.
</p>

<a name="editing">
<h4>EDITING AND SAVING THE ALIGNMENT</h4></a>
<p>
Sequences can be <a href="#add">added</a>,
<a href="#delete">deleted</a>, <a href="#reorder">reordered</a>,
<a href="#realign">realigned</a>, and <a href="#editname">renamed</a>
using the <b>Multalign Viewer</b> <a href="#mavmenu-edit"><b>Edit</b> menu</a>.
A sequence to <a href="#add">add</a> can be pasted as plain text,
read from a file, inferred from a structure, or retrieved from 
<a href="https://www.uniprot.org/" target="_blank">UniProt</a>
along with its feature annotations. Several parameters can be adjusted 
to control how the new sequence is aligned with the others.
Sequence <a href="#reorder">reordering</a> <b>will delete multisequence
<a href="#regions">regions</a></b>,
although the <b>Chimera selection</b> region 
(if any) will subsequently reappear.  
Gap columns can be 
<a href="#insert">inserted</a> and <a href="#delete">removed</a>
using the <a href="#mavmenu-edit"><b>Edit</b> menu</a>,
and a sequence can be <a href="#copy">copied as plain text</a> 
for pasting into another application window.
The <a href="#activeregion">active region</a> contents (if single-block)
can be opened in a new <a href="#seqwindow">sequence window</a>
with <a href="#new-window"><b>Edit... Region &rarr; New Window</b></a>.
</p><p>
<a href="#realign"><b>Edit... Realign Sequences</b></a>
uses a Clustal Omega or MUSCLE web service to generate a new alignment
from all of the sequences in the current alignment.
</p><p>
Manual editing allows rearranging residues and gaps without 
changing the sequences.  In other words, residues <i>within</i>
a sequence cannot be created, deleted, mutated, or reordered,
but gap positions can be created and deleted.
</p>
<a name="editkeys">
<table border="1" align="right" style="margin:8px 8px 8px 8px"
cellpadding="0" cellspacing="0">
<tr><td> <img src="editkeys.png"> </td></tr>
</table>
</a>
<p>
<a href="#showkeys"><b>Edit... Show Editing Keys</b></a>
in the <b>Multalign Viewer</b> menu brings up a dialog summarizing
<b>Ctrl</b>-key editing functions.
The <a href="#activeregion">active region</a> can be moved one
column at a time with <b>Ctrl</b>-left arrow and <b>Ctrl</b>-right arrow,
or as far as possible in one step with <b>Shift-Ctrl</b>-arrow (until the 
active region abuts either some other residues or the end of the alignment).
If the active region already abuts the alignment start or end,
new columns will be created as needed to accommodate the movement.  
Such movements can be undone step-by-step with <b>Ctrl</b>-down arrow
or collectively with <b>Ctrl-Esc</b>.  Only <b>Ctrl</b>-key editing 
can be undone, not the larger-scale operations described below
(sequence addition, deletion, and reordering).
Such larger-scale operations clear the editing history, but otherwise
all <b>Ctrl</b>-key editing operations are retained.
A previously undone movement can be redone with <b>Ctrl</b>-up arrow. 
Undo/redo repositions residues but not the
<a href="#activeregion">active region</a> box, since the
active region could have been changed during editing.
</p><p>
<a name="saving">
Choosing <a href="#mavmenu-file"><b>File... Save As</b></a>
from the <b>Multalign Viewer</b> menu</a>
brings up a <a href="../../UsersGuide/opensave.html">dialog</a> for
saving the sequence alignment to a file.  It is possible to save just the
<a href="#activeregion">active region</a> to a file; 
the region may consist of disjoint sections,
but each sequence (row) included must contain the same set of columns.
All-gap columns (for example, arising when a region to be saved includes
only a subset of the sequences) can be omitted from the output,
and <a href="#append-num">numbering can be appended</a> to the 
<a href="#seqnames">sequence names</a>.
The <a href="../../UsersGuide/filetypes.html#alignment">sequence alignment
formats</a> available for saving are the same as those that can be read.
An alignment saved in
<a href="https://en.wikipedia.org/wiki/Stockholm_format"
target="_blank">Stockholm format</a>
will automatically include
<a href="#stocksave">annotations describing the secondary structure</a>
of any <a href="#association">associated</a> structures.
</p><p>
The <a href="#seqwindow">sequence window</a> contents can also be
saved as an EPS file (<a href="#mavmenu-file"><b>File... Save EPS</b></a>
in the <b>Multalign Viewer</b> menu).
</p>

<a name="pid">
<h4>PERCENT IDENTITY</h4></a>
<p>
Pairwise percent identity can be calculated all-by-all or for specified
pairs of sequences in the alignment.
<a href="#mavmenu-tools"><b>Info... Percent Identity</b></a>
in the <b>Multalign Viewer</b> menu opens a dialog for
choosing which sequences to compare (from functionally interchangeable
pulldown menus labeled <b>Compare</b> and <b>with</b>):
<ul>
<li><b>all sequences</b> (default) 
<li>individual sequences listed by name
</ul>
There is also a choice of denominator, what to <b>divide by</b>:
<ul>
<li><b>shorter sequence length</b> (default)
<li><b>longer sequence length</b>
<li><b>non-gap columns in common</b>
</ul>
<b>Apply</b> calculates the percent identity without dismissing the dialog,
while <b>OK</b> calculates the percent identity and dismisses the dialog.
The percent identity is reported in the
status area near the bottom of the <a href="#seqwindow">sequence window</a>
and the Chimera <a href="../../UsersGuide/reply.html"><b>Reply Log</b></a>.
<b>Close</b> simply dismisses the dialog,
and <b>Help</b> opens this manual page in a browser window.
</p>

<a name="annotations">
<h4>ANNOTATIONS</h4></a>
<p>
Sequence feature annotations from 
<a href="https://www.uniprot.org/" target="_blank">UniProt</a>
are treated as <a href="#regions">regions</a> that can be displayed as
colored boxes in the <a href="#seqwindow">sequence window</a>.  
Sequence annotations from UniProt are present automatically for sequences
<a href="../../UsersGuide/fetch.html">fetched</a> from UniProt,
and UniProt annnotations can be added to other sequences 
(those already open in Chimera) using 
<a href="#mavmenu-tools"><b>Info... UniProt Annotations</b></a>.
</p><p>
The remainder of this section refers to annotations not handled as 
<a href="#regions">regions</a>.
</p><p>
<a name="anno-ali">
Annotations and comments associated with an alignment as a whole</a>
(as opposed to individual columns or sequences)
can be viewed and edited using
<a href="#editanno"><b>Edit... Alignment Annotations</b></a>
in the <b>Multalign Viewer</b> menu.
Changes are not assessed for correctness.
</p><p>
Alignment <a name="annodef"><b><i>annotations</i></b></a> are
shown in the upper part of the dialog.  Each has two parts,
a name and a value (generally text).  For example, <b>MSF length</b>
is a possible name and <b>366</b> is a possible associated value.
A new annotation can be added by clicking <b>New</b>, specifying a name, and
then entering a value in the adjacent field.  
Values can have multiple lines even though only one is shown;
pressing return in a value field starts a new line,
but the previous information is retained.
Clicking <b>Delete</b> and then choosing an annotation
name removes the corresponding annotation.
</p><p>
Alignment <a name="comdef"> <b><i>comments</i></b></a>, shown in the
lower part of the dialog, may consist of multiple lines of free-form text.
</p><p>
Not all
<a href="../../UsersGuide/filetypes.html#alignment">sequence alignment 
formats</a>
can accommodate annotations.
Thus, <a href="#saving">saved</a>
files may not include this information or reflect any changes
that have been made.  Only the
<a href="https://en.wikipedia.org/wiki/Stockholm_format"
target="_blank">Stockholm format</a>
accommodates arbitrary <a href="#annodef">annotations</a>, 
whereas both Stockholm and RSF formats
allow for <a href="#comdef">comments</a>.  
</p><p>
Markups in 
<a href="https://en.wikipedia.org/wiki/Stockholm_format"
target="_blank">Stockholm format</a> are handled as follows:
<ul>
<li>GF (Generic per-File) markups are 
interpreted as alignment <a href="#annodef">annotations</a>
or <a href="#comdef">comments</a> (GF CC indicates the latter).
<li>GC (Generic per-Column) markups are treated as 
<a href="#headers">headers</a>; they can be shown above the alignment
and are available as <a href="#mavAttributes">residue attributes</a>
in any <a href="#association">associated</a> structures.
<li>GS (Generic per-Sequence) markups are not displayed by 
<b>Multalign Viewer</b>.
<li>GR (Generic per-Sequence and per-Column) markups are not displayed 
by <b>Multalign Viewer</b>, but 
<a name="stocksave">
when an alignment is <a href="#saving">saved</a> in Stockholm format,
</a>
the file will automatically include GR
markup lines describing the secondary structure of any 
<a href="#association">associated</a> structures.
<p>
The GR markup line for a sequence and its
<a href="#association">associated</a> structure starts with
<pre>
#=GR <i>seq-name</i> Chimera_actual_SS_<i>struct-name</i>
</pre>
The structure name is included because there can be more than 
one structure associated with a given sequence.  Chimera-generated
GR secondary structure markups may include the symbols:
</p>
<table border align="center" cellpadding="3" cellspacing="0">
<tr><th align="center">symbol</th> <th align="center">meaning</th></tr>
<tr><td align="center">.</td> <td align="center">gap in sequence</td></tr>
<tr><td align="center">H</td> <td align="center">helix</td></tr>
<tr><td align="center">E</td> <td align="center">strand</td></tr>
<tr><td align="center">C</td> 
<td align="center">other structure</td></tr>
<tr><td align="center">X</td>
<td align="center">sequence residue not associated 
with structure residue</td></tr>
</table>
<p>
If a sequence did not have a GR SS markup in the
original input file and is associated with only one structure, another markup
line with the same contents but named <b>SS</b> instead of 
<b>Chimera_actual_SS_<i>struct-name</i></b> will also be included.
</ul>
</p>

<a name="association">
<h4>SEQUENCE-STRUCTURE ASSOCIATION</h4></a>
<p>
When a sequence alignment and one or more structures are opened in Chimera
(in any order), a structure chain will be associated automatically
with a sequence in the alignment if their sequences can be aligned without 
too many mismatches (<a href="#assoc-details">details...</a>).
Associations can be <a href="#assocpanel">changed or added</a>
if the automated procedure does not give the desired result.
However, when the sequence of a structure in Chimera is shown
<a href="#individual">individually</a> via the <b>Sequence</b> tool,
the sequence and the corresponding structure chain are associated
and cannot be dissociated.
</p><p>
The <a href="#seqnames">names</a> of structure-associated sequences 
are shown in bold over a box indicating the 
<a href="../../UsersGuide/hierarchy.html">model-level color</a>
of the structure.
For example, <b>APE_HUMAN</b> in the <a href="#seqwindow">sequence window
figure</a> is associated with a structure whose model-level color is tan.
If multiple models are associated with the same sequence, 
the box is shown as a dark green dashed outline.
</p><p>
<a name="crosstalk">
Association enables several types of sequence-structure crosstalk:
</a>
<ul>
<li>
<b>Associated structure reported on mouseover</b>.
When the cursor is placed over the <i>name</i> of a sequence,
information on the sequence and any associated structure(s) is shown in the
status area near the bottom of the <a href="#seqwindow">sequence window</a>.
When the cursor is placed over a sequence <i>residue</i> that is associated
with a residue in one or more structures, information on the structure
residue(s) is given.
Association information can be <a href="#saveassoc">saved to a file</a>.
<li>
<b>Clicking/dragging on sequence &rarr; structure selection</b>.
 <ul>
 <li>When a <a href="#regions">region</a> is made the
 <a href="#activeregion">active region</a> 
 (for example, when newly created by dragging),
 any associated structure residues
 will be <a href="../../UsersGuide/selection.html">selected</a>.
 <li>Clicking within a <a href="#headers">header</a> line
 will <a href="../../UsersGuide/selection.html">select</a> any 
 structure residues associated with that column of the alignment.
 </ul>
<li>
<a name="selregion"><b>Structure selection &rarr; sequence region</b>.</a>
By default, making a <a href="../../UsersGuide/selection.html">selection</a>
in one or more sequence-associated structures will create a 
<a href="#regions">region</a> named <b>Chimera selection</b> in the
associated sequences.  Every residue with at least
one atom or bond selected will be included in the region.
The color of the region and whether it should be created
can be controlled in the
<a href="#mavpref-regions"><b>Regions</b> preferences</a>.
The region will not be created when the selection
merely reflects the current <a href="#activeregion">active region</a>.
<li>
<b>Focusing the view</b>.
 <ul>
 <li>Clicking with the right mouse button on a structure-associated
 residue in the <a href="#seqwindow">sequence window</a> will focus
 the graphics window view on the corresponding structure residue(s).
 <li><b>Shift</b>-right-clicking
 within a <a href="#regions">region</a> (or right-clicking within
 the region but not on a residue) will focus the view on all of the 
 structure residues associated with the region.
 </ul>
</ul>
When the current <a href="../../UsersGuide/selection.html">selection</a>
includes one or more sequence-associated residues,
<a href="#mavmenu-structure"><b>Structure... 
Expand Selection to Columns</b></a> will expand the selection to include 
all residues associated with the same column(s) of the sequence alignment.
This can be reversed with <b>Select... Undo</b> in the main Chimera menu.
</p><p>
<a name="assoc-details">
The allowable number of mismatches for automatic association is user-specified 
as a proportion of the total number of residues in the structure chain
</a>
(1/10 by default; see 
the <a href="#mavpref-structure"><b>Structure</b> preferences</a>).
For automatic association, gaps in the structure sequence
relative to the sequence in the alignment file can only occur where
residues are missing from the structure (for example, a flexible
loop with insufficient density for coordinates to be determined).
The order in which the sequence and structure files are opened
does not matter.  Associations are reported in the
status area near the bottom of the <a href="#seqwindow">sequence window</a>
and the Chimera <a href="../../UsersGuide/reply.html"><b>Reply Log</b></a>.
A structure (even if it has multiple chains) cannot be associated with more
than one sequence, but a single sequence can be associated with 
more than one structure.
If more than one sequence matches a given structure chain,
the single best-matching sequence is associated.
</p><p>
Sequence-structure association may create new <a href="#regions">regions</a>.
If all residues of a sequence are matched, no region is created.
Otherwise, regions are created containing the matched segment(s),
any mismatches, and any gaps in the structure relative to the sequence.
The region names report the associated structure chain 
and the number of gaps or mismatches.
The initial colors and display status (shown or hidden) of these regions can be
controlled in the <a href="#mavpref-regions"><b>Regions</b> preferences</a>.
</p>

<a name="assocpanel">
<p><b>Changing Associations</b></p>
</a>
Associations can be changed or added using 
<a href="#mavmenu-structure"><b>Structure... Associations</b></a>
in the <b>Multalign Viewer</b> menu. The resulting dialog
lists the molecule models open in Chimera.  An association
is specified by choosing a chain (if the model has more than one chain) 
and the name of the sequence to associate with that chain.
The choices for association include the sequences in the alignment
displayed by <b>Multalign Viewer</b> and <b>none</b>.
When the setting is <b>none</b>, there is an option to 
<b>associate with best match</b>, <i>i.e.</i> to compare the structure
chain with all of the sequences in the alignment and associate it with the 
one that yields the fewest mismatches.
</p><p>
For the first submodel in an ensemble of structures
(multiple submodels with the same primary model number, such as from NMR),
there will be an additional button to <b>Propagate</b> its 
association setting to all of the other submodels in the ensemble.
This facilitates associating the whole ensemble with the same sequence
for purposes of <a href="#superposition">superposition</a> and/or calculating 
<a href="#rmsd-headers">RMSD headers</a>.
</p><p>
Associations will be made as specified, no matter how inappropriate.
<b>Apply</b> performs the associations without dismissing the panel, while
<b>OK</b> performs the associations and dismisses the panel.
<b>Close</b> dismisses the panel without changing the associations.
<b>Help</b> opens this manual page in a browser window.
</p>

<p><b>Automatic Structure Loading</b></p>
<p>
<a href="#mavmenu-structure"><b>Structure... Load Structures</b></a> 
in the <b>Multalign Viewer</b> menu
can be used to open structure files corresponding to sequences
that are not already structure-associated.  The corresponding structure
files are determined from the <a href="#seqnames">sequence names</a>
using the rules
given in the <a href="#mavpref-structure"><b>Structure</b> preferences</a>,
then retrieved and opened (as described for 
<a href="../../UsersGuide/fetch.html"><b>Fetch by ID</b></a>).
If <b>Automatically load Structures</b> is turned on
(also in the <a href="#mavpref-structure"><b>Structure</b> preferences</a>),
this will occur as soon as an alignment is read.
</p>

<p><b><a name="ssregion">SECONDARY STRUCTURE</a></b></p>
<p>
In sequences associated with structures,
<a href="#regions">regions</a> named <b>structure helices</b> 
(pale yellow with gold outline)
and <b>structure strands</b> (pale green with darker outline)
can be shown/hidden with the <b>Multalign Viewer</b> menu setting
<a href="#mavmenu-structure"><b>Structure... Secondary Structure... 
show actual</b></a>.
This setting is initially turned on  for individual sequences 
displayed with the <b>Sequence</b> tool.
</p><p>
Protein helix and strand assignments are taken from the input structure
file or generated with 
<a href="../../UsersGuide/midas/ksdssp.html"><b>ksdssp</b></a>.
Helix and strand regions can overlap when more than one structure 
is associated with a sequence.
(Regardless of whether the regions exist, however,
an alignment saved in
<a href="https://en.wikipedia.org/wiki/Stockholm_format"
target="_blank">Stockholm format</a>
will automatically include
<a href="#stocksave">annotations describing the secondary structure</a>
of any associated structures.)  
</p><p>
In sequences not associated with structures,
<a href="#regions">regions</a> named <b>predicted helices</b> (gold outline) 
and <b>predicted strands</b> (green outline)
can be shown/hidden with the <b>Multalign Viewer</b> menu setting
<a href="#mavmenu-structure"><b>Structure... Secondary Structure... 
show predicted</b></a>.
The prediction is done with <a href="gor.html"><b>GOR</b></a>.
</p>

<a name="mavAttributes">
<h4>RESIDUE ATTRIBUTES</h4></a>
<p>
Structure residues <a href="#association">associated</a> with
residues in a sequence alignment shown by <b>Multalign Viewer</b>
are assigned
<a href="../defineattrib/defineattrib.html#attribdef">attributes</a>:
<ul>
<li><a name="mavPercentConserved"><b>mavPercentConserved</b></a>
- the value for a residue is
the percent conservation of the most prevalent residue type 
at the corresponding position in the alignment.
<li><a name="mavHeader"><b>mav<i>Header</i></b>, where <i>Header</i>
is the name of a <a href="#headers">header</a> line
- the value for a residue is the character or number 
(shown with bar height in a histogram) in the header line 
at the corresponding position of the alignment. For example:
  <ul>
  <li>if <a href="#conservation"><b>Conservation</b></a> is shown 
  as a histogram, there will be a <b>mavConservation</b> attribute with 
  numerical values; if it is shown with characters,
  the attribute values will be characters  
  <li>if the <a href="#rmsd-headers"><b>RMSD: ca</b></a> header is shown,
  there will be a corresponding residue attribute named <b>mavRMSDca</b>
  </ul>
</ul>
Structure residues not associated with residues in the sequence
alignment are not assigned attribute values.
</p><p>
<a href="#mavmenu-structure"><b>Structure... Select</b> (or <b>Render</b>)
<b>by Conservation</b></a> in the <b>Multalign Viewer</b> menu opens
the corresponding portion of
<a href="../render/render.html"><b>Render/Select by Attribute</b></a>,
set to the residue attribute <b>mavConservation</b> if numerical,
otherwise <b>mavPercentConserved</b>. 
</p><p>
Numerical attributes will appear in the attribute lists of
<a href="../render/render.html"><b>Render/Select by Attribute</b></a>;
character attributes will be listed only 
<a href="../render/render.html#select"><b>Select by Attribute</b></a> portion.
It may be necessary to 
<a href="../render/render.html#refreshing"><b>Refresh</b></a> 
the attribute menus or values in these tools to update them with any changes
in <b>Multalign Viewer</b> header information.
</p><p>
<b>Note</b>: if a structure is associated with more than one alignment,
the attribute values are based on the alignment for which association
occurred or changed most recently.  
Generally this is the alignment most recently created or opened, but 
a different alignment can be forced as the &ldquo;winner&rdquo; by choosing 
<a href="#mavmenu-structure"><b>Structure... Associations</b></a>
from its menu and toggling an existing association to <b>none</b> and
then back (each time clicking <b>Apply</b>).
</p>

<a name="superposition">
<h4>STRUCTURAL SUPERPOSITION</h4></a>
<p>
<a href="#mavmenu-structure"><b>Structure... Match</b></a> in the
<b>Multalign Viewer</b> menu
allows structures to be superimposed based on
the alignment of their <a href="#association">associated</a> sequences.
The dialog contains two subpanels, each listing the structures 
<a href="#association">associated</a> with sequences in the alignment.
One reference structure should be chosen in the left side, but any number
of structures to be matched (superimposed)
with it can be chosen in the right side.
Residues in each <b>Structure to match</b> are paired with
the aligned (in the sequence alignment) residues of the <b>Reference
structure</b>.  If both structures are <a href="#association">associated</a>
with the same sequence, the correspondence is even more obvious.
<a name="onepoint">
Fitting uses one point per residue</a>:  CA in
amino acid residues and C4' in nucleic acid residues.
If a nucleic acid residue lacks a C4' atom 
(some lower-resolution structures are P traces), 
its P atom will be paired with the P atom of the aligned residue. 
The number of atom pairs fitted and the resulting RMSD are reported in the
Chimera <a href="../../UsersGuide/reply.html"><b>Reply Log</b></a> and the
status area near the bottom of the <a href="#seqwindow">sequence window</a>.
</p><p>
<b>Match highly conserved residues only</b> causes only the well-conserved
(at least 80%) positions in the alignment to be used for the least-squares fit.
These are the positions shown as capital letters in the <a href="#consensus">
consensus sequence</a>.
</p><p>
<b>Match active region only</b> causes only the positions in the current
<a href="#activeregion">active region</a> of the alignment to be used
for matching.
</p><p>
<b>Use pseudobonds to show matched atoms</b> indicates that lines
(<a href="../../UsersGuide/pbpanel.html#pbdef">pseudobonds</a>) 
should be drawn between the matched atoms.
For each matched pair of structures, a
<a href="../../UsersGuide/pbpanel.html#pbdef">pseudobond</a> group
is created and colored uniquely (in order, the
<a href="../../UsersGuide/colortables.html#allcolors">named colors</a>
dark green, dodger blue, sienna, yellow, spring green, purple, gray,
and coral are used).  Each group
is named <b>matches of...</b>, where the rest
of the name indicates the structures and chains that were matched.
The <a href="../../UsersGuide/pbpanel.html"><b>PseudoBond Panel</b></a>
can be used to change the appearance of or delete the
<a href="../../UsersGuide/pbpanel.html#pbdef">pseudobonds</a>.
</p><p>
<b>Iterate by pruning long atom pairs until no pair exceeds [<i>x</i>]
angstroms</b>
refers to an iterative fitting procedure: in each cycle,
atom pairs are removed from the match list and the remaining
pairs are fitted, until no matched pair is
more than <i>x</i> apart (default <b>2.0</b> &Aring;).
The atom pairs removed are either the 10% farthest apart of all pairs
or the 50% farthest apart of all pairs exceeding the cutoff, whichever
is the lesser number of pairs.
The result is that the best-matching &ldquo;core&rdquo; regions are maximally
superimposed; conformationally dissimilar regions such as flexible
loops are not included in the final fit, even though they may be aligned
in the sequence alignment.
</p><p>
<b>Create region showing matched residues</b>
indicates that the residues paired to generate
the final fit should be shown as a <a href="#regions">region</a>
named <b>matched residues</b>.
</p><p>
<b>Apply</b> performs the matching (superposition) without dismissing the
dialog.  <b>OK</b> performs the matching and dismisses the dialog,
<b>Cancel</b> dismisses the dialog without performing a match,
and <b>Help</b> opens this manual page in a browser window.
See also: <a href="../../UsersGuide/superposition.html">Chimera superposition
methods</a> 
</p>

<a name="assessment">
<h4>SUPERPOSITION ASSESSMENT</h4></a>
<p>
The spatial variation among multiple superimposed structures
can be shown and analyzed using one or more of the
<a href="#rmsd-headers">RMSD headers</a> and corresponding
<a href="#mavAttributes">residue attributes</a>
(<b>mavRMSDca</b>, <b>mavRMSDbackbone</b>, and <b>mavRMSDfull</b>).
All of the structure residues associated with a column
in the alignment are assigned the same value of a given attribute.
</p><p>
All other RMSD calculations in <b>Multalign Viewer</b> (besides headers)
use <a href="#onepoint">one point per residue</a>.
</p><p>
<a name="overall-rmsd">
An <b><i>overall</i> RMSD</b> value</a> is calculated for each
<a href="#regions">region</a> in which a block encloses at least one column 
<a href="#association">associated</a> with at least two structure residues.  
All pairwise within-column distances are included in the calculation of the
overall RMSD, and all distances are weighted equally (columns may not be
weighted equally, as they may be associated with differing numbers of 
structure residues).
The current overall RMSD for a region is reported at the bottom of the 
<a href="#seqwindow">sequence window</a> and in the
<a href="../../UsersGuide/reply.html"><b>Reply Log</b></a>
as the region is created, and values that are updated as needed
are shown in the <b>RMSD</b> column of the
<a href="#regionbrowser"><b>Region Browser</b></a>.
</p><p>
Pairwise assessments of spatial variation can be performed by choosing
<a href="#mavmenu-structure"><b>Structure... Assess Match</b></a> from the
<b>Multalign Viewer</b> menu.  Each pairwise comparison creates an
attribute of the residues of one structure containing the distances
from the sequence-aligned residues of a reference structure.
</p><p>
A <b>Reference structure</b> should be chosen from the pulldown
menu of structures
<a href="#association">associated</a> with sequences in the alignment.
One or more <b>Structures to evaluate</b> should be chosen
from the list of remaining structures.  
The structures should already be superimposed, but the superposition can
have been generated in any way, including manually (not necessarily using
<a href="#superposition"><b>Structure... Match</b></a>).
</p><p>
The distance between each reference-evaluation pair of residues
aligned in the sequence alignment (or corresponding to the same residue, if the
reference and evaluation structures are <a href="#association">associated</a>
with the same sequence) is measured.  The distances are assigned as
a <a href="#mavAttributes">residue attribute</a> of the evaluation structure(s),
named <b>matchDist</b> by default.  After attribute values have been
assigned, <a href="../render/render.html#select"><b>Select by Attribute</b></a>
will appear, set to the new residue attribute.
</p><p>
<b>OK</b> performs the comparison and dismisses the dialog,
while <b>Close</b> just dismisses the dialog.
<b>Help</b> opens this manual page in a browser window.
</p>

<a name="modeller">
<h4>INTERFACE TO MODELLER</h4></a>
<p>
The program
<a href="https://www.salilab.org/modeller/" target="_blank">Modeller</a>
performs protein comparative modeling and related calculations.
Chimera provides a graphical interface to running Modeller,
either locally or via a web service hosted by the
<a href="https://www.cgl.ucsf.edu/Overview/index.html" target="_blank">UCSF
RBVI</a>. Modeller use requires a license key, available free of charge to
academics upon <a href="https://www.salilab.org/modeller/registration.html"
target="_blank">registration</a>.
Two types of calculations are available:
<ol>
<li><a href="modeller.html#comparative"><b>Comparative 
(homology) modeling</b></a>, interface shown by choosing
<a href="#mavmenu-structure"><b>Structure... Modeller (homology)</b></a> 
from the <b>Multalign Viewer</b> menu.  In comparative modeling,
theoretical models of a protein are generated using at least one known related
structure and a sequence alignment of the known and unknown structures.
<a name="approaches">
The protein to be modeled is the <b><i>target</i></b>, and a related
known structure used for modeling is a <b><i>template</i></b>.
The inputs for comparative modeling can be generated in several ways</a>:
</p>
<ul>
<li>Using a pre-existing target-template sequence alignment:
  <ul>
  <li><a href="../../UsersGuide/filetypes.html">open</a> the
<a href="../../UsersGuide/filetypes.html#alignment">sequence alignment file</a>
  and template structure(s)
  <li>make sure the template structure(s) are
  <a href="#association">associated</a> with the alignment
  </ul>
<br>
<a name="known-template"></a>
<li>Starting with a structure already known to be a suitable template:
    <ul>
    <li><a href="../../UsersGuide/filetypes.html">open</a> the structure(s)
    <li>if a single template structure,
    start <a href="#top"><b>Sequence</b></a> to show its sequence
    <li>if multiple template structures, use the
    <a href="../align/align.html"><b>Align Chain Sequences</b></a> tool
    to generate a multiple alignment from their sequences
    <li><a href="#editing">add</a> the target sequence 
    to align it to the template sequence(s)
    (<a href="#mavmenu-edit"><b>Edit... Add Sequence</b></a> in the
    <b>Multalign Viewer</b> menu)
    <li>if needed, <a href="realign.html">realign</a> the sequences
    to further improve the multiple alignment
    (<a href="#realign"><b>Edit... Realign Sequences</b></a> in the
    <b>Multalign Viewer</b> menu)
    </ul>
<br>
<li>Blast-searching<a href="#blast-caveats"><sup>1</sup></a>
the PDB for structures related to the target sequence 
(via individual steps as described below, or in a combined fashion using
<a href="../../UsersGuide/midas/mda.html"><b>mda</b></a><a href="#mda-overlay"><sup>2</sup></a>):
  <ul>
  <li>start <a href="../../UsersGuide/blast.html"><b>Blast Protein</b></a>
  (under <b>Tools... Sequence</b> in the main Chimera menu),
  paste in the target sequence as plain text,
  search the <b>pdb</b> database for similar sequences, <i>i.e.</i>,
  structures that could be used as templates; from the list of results,
  choose one or more hits, show them in MAV (<b>Multalign Viewer</b>) as a 
  sequence alignment with the query, and load the corresponding structure(s)
  <br> &ndash; or &ndash;
  <li><a href="../../UsersGuide/fetch.html">fetch</a> the target sequence
  from <a href="https://www.uniprot.org" target="_blank">UniProt</a>,
  use the <b>Multalign Viewer</b>
  <a href="#mavmenu-tools"><b>Info</b> menu</a> to start
  <a href="../../UsersGuide/blast.html"><b>Blast Protein</b></a>,
  proceed as above
  <br> &ndash; or &ndash;
  <li><a href="../../UsersGuide/filetypes.html">open</a>
  a file containing just the target sequence
   (<a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp#search"
  target="_blank">FASTA</a> or
  <a href="https://www.bioinformatics.nl/tools/crab_pir.html" 
  target="_blank">NBRF/PIR</a> format), use the <b>Multalign Viewer</b>
  <a href="#mavmenu-tools"><b>Info</b> menu</a> to start
  <a href="../../UsersGuide/blast.html"><b>Blast Protein</b></a>,
  proceed as above
  </ul>
<p>
<a name="blast-caveats"><sup>1</sup></a>
An alignment from Blast is only a pseudo-multiple alignment,
a consolidation of the pairwise alignments of individual hits to the query.
Since alignment quality is important for comparative modeling, it may be
helpful to <a href="realign.html">realign</a> the sequences with true 
multiple-alignment software 
(<a href="#realign"><b>Edit...  Realign Sequences</b></a> in the
<b>Multalign Viewer</b> menu) before modeling.
Further, Blast may omit parts of the sequences of hit structures.
To generate a multiple alignment containing the entire sequences,
load the structures of the Blast hits of interest, 
then proceed as described <a href="#known-template">above</a>
for starting with a structure.
</p><p>
<a name="mda-overlay"><sup>2</sup></a>
Before comparative modeling with multiple templates fetched by
<a href="../../UsersGuide/midas/mda.html"><b>mda</b></a>, use the command 
<b>reset overlay</b> to position the template structures for best results.
</p>
</ul>
<br>
<li><a href="modeller.html#building"><b>Building</b></a>
parts of a protein without using a template.  The interface is shown
by choosing <a href="#mavmenu-structure"><b>Structure... Modeller 
(loops/refinement)</b></a> from the <b>Multalign Viewer</b> menu.
Missing segments can be built <i>de novo</i>, or existing segments
<b><i>refined</i></b> by generating additional possible conformations.
Parts that need building or refinement are often (but not always)
loops between secondary structure elements.
</ol>

<a name="mavprefs">
<h4>PREFERENCES</h4></a>
<p>
The <b>Preferences</b></a> section of the
<a href="#mavmenu"><b>Multalign Viewer</b> menu</a>
controls preferences specific to <b>Multalign Viewer</b>.
Settings are saved to the Chimera
<a href="../../UsersGuide/preferences.html#preffile">
<b>preferences</b> file</a> as soon as they are changed.
</p><p>
<b>Multalign Viewer</b> preferences are grouped together
in sections shown as index cards:
<ul>
<li><a href="#mavpref-layout"><b>Appearance</b></a>
- sequence display parameters
<li><a href="#mavpref-structure"><b>Structure</b></a>
- structure-loading rules
<li><a href="#mavpref-analysis"><b>Headers</b></a>
- calculation of sequence alignment <a href="#headers">headers</a>
<li><a href="#mavpref-regions"><b>Regions</b></a>
- creation and display of <a href="#regions">regions</a> (colored boxes)
</ul>
<b>Close</b> dismisses the <b>Multalign Viewer</b> preferences tool,
and <b>Help</b> opens this manual page in a browser window.
</p><p>
Note that a typed-in value will not be applied until <b>Apply</b>, 
<b>OK</b> (which also dismisses the preferences tool),
or the Enter (return) key has been pressed.
Changes in other types of settings take effect and are saved immediately.
</p><p>
The <a name="mavpref-layout"><b>Appearance</b></a> section of
the <a href="#mavprefs"><b>Multalign Viewer</b> preferences</a>
controls sequence text arrangement and coloring. 
Most of the preferences in this section can be set separately for
<b>Multiple alignments</b> and <b>Single sequences</b>.
<ul>
<li><a name="wrapping"><b>Line wrapping</b></a>
- whether sequences should be shown as a single horizontal bar
or wrapped to form successive blocks
  <ul>
  <li><b>Wrap to new line every [<i>n</i>]0 residues</b> 
  (default for single sequences)
  - wrap at the specified interval (default <i>n</i>=<b>5</b>, or 50 positions)
  regardless of the number of sequences
  <li><b>Wrap if [<i>k</i>] or fewer sequences</b> 
  (default for multiple alignments, not relevant to single sequences)
   - wrap at the interval specified above if the alignment
   contains no more than <i>k</i> sequences (default <b>8</b>)
  <li><b>Never wrap</b>
  <li><b>Space between wrapped blocks</b> 
  - whether to pad blocks vertically
  (default true for alignments, false for single sequences)
  </ul>
<li><b>Font</b>
  <ul>
  <li><b>Use [<i>m</i>] point</b> (Courier/<b>Helvetica</b>/Times),
  optionally <b>Bold</b> (<i>m</i>=<b>12</b> by default)
  </ul>
<li><b>Spacing</b>
  <ul>
  <li><b>Column separation (pixels)</b> 
  (default <b>0</b> for alignments, <b>&ndash;2</b> for single sequences)
  <li><b>Line separation (pixels)</b>
  (default <b>1</b> for alignments, <b>8</b> for single sequences)
  <li><b>Space after every 10 residues</b> 
  (default true for alignments, false for single sequences)
  </ul>
<li><a name="coloring"><b>Residue letter coloring</b></a>
  <ul>
  <li><b>Color scheme</b>
   <ul>
   <li><b>black</b> (default for single sequences)
   - color all letters black
   <li><b>Clustal X</b> (default for alignments)
    - use <a href="cxcolor.html">Clustal X coloring</a>, which depends on 
    both residue type and the pattern of conservation within a column
   <li><b>ribbon</b> - color one-letter codes to match the
    <a href="#association">associated</a> structure residues,
    the average color where multiple structures are associated, and black
    where no structures are associated;
    <a href="../../UsersGuide/representation.html#ribbon">ribbons</a> will
    match the lettering, but need not be displayed for this option to work.
    Possible uses include making the letters match a structure that has been
    colored <a href="../render/render.html#render">by attribute</a>, by domain,
    or over a <a href="../rainbow/rainbow.html">rainbow</a> range of hues.
   <li><b>Kyte-Doolittle hydrophobicity</b> - color one-letter codes by
   <a href="../../UsersGuide/midas/hydrophob.html">Kyte-Doolittle 
   hydrophobicity</a>, ranging from red for the most hydrophobic 
   amino acids to blue for the most hydrophilic, with black in the middle
   <li>custom coloring schemes can be listed here by describing them 
   in the Clustal X <a href="cxcolor.html#format">parameter file format</a> 
   and reading them in with 
   <a href="#mavmenu-file"><b>File... Load Color Scheme</b></a>
   </ul>
  </ul>
<li><b>Sequence names</b> 
  <ul>
  <li><b>Use ellipsis for names longer than [<i>j</i>]</b> 
  (<i>j</i>=<b>30</b> characters by default)
  </ul>
</ul>
The <a name="mavpref-structure"><b>Structure</b></a> section of
the <a href="#mavprefs"><b>Multalign Viewer</b> preferences</a>
sets rules for automatic
<a href="#association">sequence-structure association</a>
and structure loading.
<ul>
<li><b>Auto-association</b> - criteria for automatic 
<a href="#association">sequence-structure association</a>
  <ul>
  <li><b>Allow one mismatch per [<i>n</i>] structure residues</b>
  (<i>n</i>=<b>10</b> by default)
  </ul>
<li><b>Structure loading</b> 
- whether/how <a href="#seqnames">sequence names</a> relate to 
the names of the corresponding structure files; whether to retrieve
and open such files 
(as described for <a href="../../UsersGuide/fetch.html"><b>Fetch by ID</b></a>)
as soon as a sequence alignment is read.
No more than one structure per sequence will be loaded.
If a sequence name corresponds to both a SCOP file and a PDB file,
the SCOP file will be loaded
(unless the <b>Load SCOP...</b> option is turned off).
  <ul>
  <li><b>Load SCOP file for each unassociated sequence whose name appears
      to be a <a href="../../UsersGuide/scop.html">SCOP ID (sid)</a></b>
      (on by default)
  <li><b>Load PDB file for each unassociated sequence whose name...</b>
    <ul>
    <li><b>is &lt; pdb code &gt;<br>
         (e.g."1gcn")</b>
    <li><b>starts with &lt; pdb code &gt;<br>
         (e.g."1k6w_A")</b>
    <li><b>contains "pdb|" + &lt; pdb code &gt;<br>
         (NCBI-style; e.g. "gi|1421399|pdb|1FTZ")</b>
    <li><b>starts with [<i>string</i>] + &lt; pdb code &gt;</b><br>
    where <b><i>string</i></b> is user-specified
    </ul>
    These correspondence rules are not mutually exclusive; all but
    the last are on by default.
    <li><b>Automatically load structures</b> (off by default) - whether to
    retrieve and open structures as soon as a sequence alignment is read;
    if not, the procedure can still be invoked with
    <a href="#mavmenu-structure"><b>Structure... Load Structures</b></a> in the
    <b>Multalign Viewer</b> menu
    <li><b>Do not autoload if more than [<i>N</i>] structures would load</b>
     (on and <i>N</i>=<b>10</b> by default)
  </ul>
</ul>
</p><p>
The <a name="mavpref-analysis"><b>Headers</b></a> section of
the <a href="#mavprefs"><b>Multalign Viewer</b> preferences</a>
controls how alignment <a href="#headers">headers</a> are calculated 
and whether alignment <a href="#numbering">numbering</a> is shown initially.
<ul>
<li><a name="consensus"><b>Consensus style</b></a>
 (<b>majority in column including gaps</b>/majority in column ignoring gaps) 
- what to show in the <a href="#headers">header</a> line named 
<b>Consensus</b>.  
The consensus sequence shows the residue type most prevalent
(not necessarily >50%) at each position in the alignment;
highly conserved residues (80% or greater) are capitalized
and shown in purple, except that the completely conserved residues
are shown in red.
If several residue types are equally the most prevalent,
one is chosen at random to appear in the consensus sequence.
When gaps are included, however,
the consensus will show a gap at positions where a gap
is the most prevalent or equally the most prevalent 
as one or more residue types.
<br><br>
<li><a name="conservation"><b>Conservation style</b></a>
(AL2CO/Clustal histogram/Clustal characters/<b>identity histogram</b>) 
- what to show in the <a href="#headers">header</a> line named 
<b>Conservation</b> and assign as the
<a href="#mavAttributes"><b>mavConservation</b> attribute</a> of residues in
<a href="#association">associated</a> structures.
The values can be <a href="#saveheader">saved to a file</a>.
  <blockquote>
  In a <b>Clustal histogram</b>, full bar height indicates complete
  identity, 2/3 bar height indicates Clustal
  <a href="clustgroups.html">strong group</a> conservation,
  and 1/3 bar height indicates Clustal
  <a href="clustgroups.html">weak group</a> conservation.
  </blockquote><blockquote>
  <b>Clustal characters</b> are the characters used in Clustal format
  to indicate conservation: "*" for complete identity, ":" for
  <a href="clustgroups.html">strong group</a> conservation, and "." for
  <a href="clustgroups.html">weak group</a> conservation.
  </blockquote><blockquote>
  An <b>identity histogram</b> shows what proportion of the sequences
  have the most prevalent nongap character per alignment column.
  Histogram bar heights are proportional to (M-1)/(N-1), where
  M is the number of occurrences of the most prevalent residue type at 
  a position in the alignment and N is the number of sequences in the 
  alignment. Thus, if every sequence has a different residue at a 
  given position, the bar height is zero, not 1/N.
  However, M/N values (not the values used for display purposes) are
  used for the <a href="#mavAttributes"><b>mavConservation</b> attribute</a>
  and written out when conservation values are <a href="#saveheader">saved</a>.
  </blockquote><blockquote>
  <b>AL2CO</b> (<b>AL</b>ignment <b>2</b> <b>CO</b>nservation)
  provides more sophisticated options for calculating conservation, including
  choice of equation (entropy, <i>etc.</i>), sequence weighting, 
  and smoothing over windows. <a name="al2coref">Users should cite</a>:
    <blockquote>
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/11524371" target="_blank">
    AL2CO: calculation of positional conservation in a 
    protein sequence alignment.</a>
    Pei J, Grishin NV.
    <i>Bioinformatics.</i> 2001 Aug;17(8):700-12.
   </blockquote>
  <b>AL2CO parameters</b>:
  <ul>
  <li><b>Frequency estimation method</b> (unweighted/modified Henikoff &
  Henikoff/<b>independent counts</b>) - whether/how to weight the sequences
  <li><a name="al2comeas"><b>Conservation measure</b></a> 
  (<b>entropy-based</b>/variance-based/sum of pairs) 
  - which of the AL2CO equations to use for conservation 
  <li><b>Averaging window</b> (default <b>1</b>) - window width, or
  how many alignment positions to average in a sliding window.
  The result is assigned to the central column of the window
  when the width is odd, the column left of center when the width is even 
  (<i>e.g.</i>, the fourth position in a window of 8), 
  with various adjustments at the termini.
  The authors of AL2CO recommend a width of 3 for motif analyses.
  <li><a name="gapfract"><b>Gap fraction</b></a> 
  (default <b>0.5</b>) - maximum fraction of sequences with gaps in a column 
  for conservation to still be calculated; for example, if the 
  <b>gap fraction</b> is 0.25, conservation values will only be calculated 
  for columns in which at least three-fourths of the sequences have residues
  <li><b>Sum-of-pairs matrix</b> [choices:
  BLOSUM-30, BLOSUM-35, BLOSUM-40, BLOSUM-45, BLOSUM-50, BLOSUM-55,
  BLOSUM-60, <b>BLOSUM-62</b> (default), BLOSUM-65, BLOSUM-70, BLOSUM-75,
  BLOSUM-80, BLOSUM-85, BLOSUM-90, BLOSUM-100, BLOSUM-N, HSDM, identity,
  PAM-40, PAM-120, PAM-150, PAM-250, SDM] - which amino acid similarity matrix 
  to use with the AL2CO sum of pairs 
  <a href="#al2comeas"><b>Conservation measure</b></a>
  <li><b>Matrix transformation</b> (<b>none</b>/normalization/adjustment)
  - whether/how to modify the matrix used with the AL2CO sum of pairs 
  <a href="#al2comeas"><b>Conservation measure</b></a>
  <blockquote>
  normalization: &nbsp;&nbsp;&nbsp; <i>S'<sub>ab</sub></i> = <i>S<sub>ab</sub></i> / (<i>S<sub>aa</sub>S<sub>bb</sub></i>)<sup>&frac12;</sup>
  </blockquote><blockquote>
  adjustment: &nbsp;&nbsp;&nbsp; <i>S''<sub>ab</sub></i> = 2<i>S<sub>ab</sub></i> &ndash; &frac12;(<i>S<sub>aa</sub></i> + <i>S<sub>bb</sub></i>)
  <blockquote>
  </ul>
  Conservation values
  from AL2CO are in standard deviations from the mean (sometimes called 
  Z-scores) and can range from &ndash;&infin; (least conserved) to 
  +&infin; (most conserved).  
  </blockquote>
<li><b>Show numbering initially</b> (<b>true</b>/false) 
- whether to show <a href="#numbering">alignment numbering</a>
when an alignment is opened (does not apply to single sequences)
</ul>
The <a name="mavpref-regions"><b>Regions</b></a> section of
the <a href="#mavprefs"><b>Multalign Viewer</b> preferences</a>
controls whether a <a href="../../UsersGuide/selection.html">selection</a>
will be shown as a <a href="#regions">region</a> and sets initial region colors.
  <ul>
  <li><b>Show Chimera selection as region</b> (<b>true</b>/false) -
    whether to show a <a href="../../UsersGuide/selection.html">selection</a>
    within structures as a <a href="#regions">region</a>
    named <b>Chimera selection</b>
    within <a href="#association">associated</a> sequences (if any);
    a residue will be included in the region if any of its atoms are selected
  <li><b>Chimera selection region border color</b>
    (a <a href="../../UsersGuide/coloring.html#colorwell">color well</a>,
    <b>No color</b> by default) 
    - initial border color for the region showing the Chimera
    <a href="../../UsersGuide/selection.html">selection</a>
  <li><b>Chimera selection region interior color</b>
    (a <a href="../../UsersGuide/coloring.html#colorwell">color well</a>,
    <a href="../../UsersGuide/colortables.html#topcolors"><b>green</b></a> 
    by default)
    - initial interior color for the region showing the Chimera
    <a href="../../UsersGuide/selection.html">selection</a>
  <li><b>New region border color</b>
    (a <a href="../../UsersGuide/coloring.html#colorwell">color well</a>,
     <b>No color</b> by default)
    - initial border color for <a href="#manual">manually created regions</a>
  <li><b>New region interior color</b>
    (a <a href="../../UsersGuide/coloring.html#colorwell">color well</a>,
    <a href="../../UsersGuide/colortables.html#topcolors"><b>white</b></a>
    by default)
    - initial interior color for <a href="#manual">manually created regions</a>
  <li><b>Show balloon help for regions</b> (<b>true</b>/false) -
   whether to show the region name in a balloon when the cursor is paused
   over the region (but not over a residue symbol, as that would instead
   report the <a href="#association">associated</a> structure residue)
  </ul><ul>
  <li><b>Structure association</b> - whether and how to show regions created
during <a href="#association">association</a>
    <ul>
    <li><b>Draw region depicting successful matches</b> (on/<b>off</b>;
    <a href="../../UsersGuide/coloring.html#colorwell">color wells</a>
    for the border and interior, default
    <b>black</b> and <b>No color</b>, respectively)
    <li><b>Draw region depicting mismatches</b> (<b>on</b>/off;
    <a href="../../UsersGuide/coloring.html#colorwell">color wells</a>
    for the border and interior, default
    <b>No color</b> and <b>red</b>, respectively)
    <li><b>Draw region depicting missing structure</b> (<b>on</b>/off;
    <a href="../../UsersGuide/coloring.html#colorwell">color wells</a>
    for the border and interior, default
    <b>red</b> and <b>No color</b>, respectively)
    </ul>
  </ul>
</ul>
</p>

<a name="mavmenu">
<h4>MENU LISTING</h4>
</a>
The <b>Multalign Viewer</b> menu includes the following sections:
<ul>
<li><a href="#mavmenu-file"><b>File</b></a>
<li><a href="#mavmenu-edit"><b>Edit</b></a>
<li><a href="#mavmenu-structure"><b>Structure</b></a>
<li><a href="#mavmenu-headers"><b>Headers</b></a>
<li><a href="#mavmenu-numberings"><b>Numberings</b></a>
<li><a href="#mavmenu-tree"><b>Tree</b></a>
<li><a href="#mavmenu-tools"><b>Info</b></a>
<li><a href="#mavprefs"><b>Preferences</b></a>
</ul>

<a name="mavmenu-file">
<h4>File</h4></a>
<ul>
<li><b>Save As...</b> bring up a
<a href="../../UsersGuide/opensave.html">dialog</a> for 
<a href="#saving">saving the alignment</a> to a file.
The <a href="../../UsersGuide/filetypes.html#alignment">sequence alignment
formats</a> available for saving are the same as those that can be read.
Options:
  <ul>
  <li><b>Restrict save to <a href="#activeregion">active region</a></b>
  (<b>off</b> by default)
  <li><b>Omit all-gap columns</b> (default <b>on</b>)
  <li><b><a href="#append-num">Append sequence numberings</a>
  to sequence names</b> (default <b>on</b> if <a href="#numbering">sequence
  numbering</a> is displayed, otherwise <b>off</b>)
  </ul>
<li><b>Save EPS...</b> bring up a
<a href="../../UsersGuide/opensave.html">dialog</a> for
saving an Encapsulated PostScript file of the
<a href="#seqwindow">sequence window</a> contents.  Options:
  <ul>
  <li><b>color mode</b> (<b>color</b>/grayscale/black & white)
  <li><b>rotate 90</b> (true/<b>false</b>)
  <li><b>extent</b> (<b>visible region</b>/entire alignment)
  <li><b>hide tree control nodes</b> (<b>true</b>/false) - whether to omit
the small squares from a <a href="#trees">tree</a> display
  </ul>
If the alignment is <a href="#wrapping">wrapped</a>
(or would wrap if it were longer), one
EPS file will be saved with the specified name.  
Otherwise, the sequence <a href="#seqnames">names</a>
and alignment will be saved in two separate EPS files with
"-names" and "-alignment" appended to the specified name
(before the ".eps" suffix, if present).
<li><a name="saveassoc"><b>Save Association Info...</b></a>
save a text file that states which structure is associated with
which sequence and lists the correspondences between alignment positions
and structure residues.  The <b>Residue naming style</b>
indicates how residues will be specified in the file:
  <ul>
  <li><b>simple</b> (for example, <b>PHE 16</b>)
  <li><a href="../../UsersGuide/midas/atom_spec.html#basic"><b>command-line 
  specifier</b></a> (for example, <b>:16</b>)
  </ul>
</ul><ul>
<li><b>Load SCF/Seqsel File...</b>
read a <a href="jevtrace.html">JEvTrace</a> file; color
sequence <a href="#regions">regions</a> and the corresponding
parts of any associated structures accordingly
<li><b>Load Color Scheme...</b>
read a <a href="cxcolor.html#format">parameter file</a> to set the
coloring of residue codes in the <a href="#seqwindow">sequence window</a>.
When a scheme is loaded, the residue codes are colored accordingly
and the name of the custom coloring file is added to the list of 
choices for <a href="#coloring"><b>Residue letter coloring</b></a>
in the <a href="#mavpref-layout"><b>Appearance</b> preferences</a>.
The <a href="../../UsersGuide/opensave.html">dialog</a> for loading the
file includes an option to <b>Make this scheme the default</b>.
The location of each custom coloring file is saved in the
<a href="../../UsersGuide/preferences.html#preffile">
<b>preferences</b> file</a>.  If a custom coloring file is later
renamed, moved, or deleted, trying to load the corresponding scheme 
will generate an error.
</ul><ul>
<li><b>Hide</b>
- close the sequence window without changing
the state of <b>Multalign Viewer</b>
(the window can be reopened using the <b>Raise</b> option
for the <b>Multalign Viewer</b> <a href="../../UsersGuide/instances.html">
instance in the Chimera <b>Tools</b> menu</a>, abbreviated MAV)
<li><b>Quit</b>
- exit from <b>Multalign Viewer</b>
</ul>

<a name="mavmenu-edit">
<h4>Edit</h4></a>
<ul>
<li><a name="copy"><b>Copy Sequence...</b></a> 
copy a sequence as text that can be pasted into another application window.
Options:
  <ul>
  <li><b>Remove gaps from copy</b> (default <b>on</b>) - exclude gap characters
  (typically present when the sequence is part of a multiple alignment)
  <li><b>Restrict copy to <a href="#activeregion">active region</a></b>
  (default <b>off</b>) - only include residues that are within the
  <a href="#activeregion">active region</a></b>, 
  even if they form disjoint segments
  </ul>
Note: when the mouse focus is in the <a href="#seqwindow">sequence window</a>,
the system &ldquo;copy&rdquo; shortcut (for example, command-C on Mac) 
will copy the contents of the <a href="#activeregion">active region</a></b>, 
if any, otherwise the entire sequence(s), into the text buffer.
Multiple sequences or their parts will be separated by newlines.
<li><a name="reorder"><b>Reorder Sequences</b></a> 
- open a dialog for reordering sequences.
The dialog lists the names of the sequences in their current order. The
order can be changed by clicking on a name and moving it to the <b>top</b>,
<b>bottom</b>, <b>one higher</b>, or <b>one lower</b>.  Any order can be
achieved through combinations of these operations.
<b>OK</b> and <b>Apply</b> perform the reordering
in the <a href="#seqwindow">sequence window</a> with and without
dismissing the dialog, respectively.
Sequence reordering <b>will delete multisequence
<a href="#regions">regions</a></b>,
although the <b>Chimera selection</b> region 
(if any) will subsequently reappear.
<li><a name="insert"><b>Insert All-Gap Columns</b></a> - open a dialog
for inserting all-gap columns into the alignment (presumably to facilitate 
subsequent <a href="#editing">editing</a> 
that moves residues into the new space). 
The dialog allows specification of how many gap columns to add, 
where to insert them, and the character to use for gap positions.
<b>OK</b> and <b>Apply</b> insert the gap columns with and without
dismissing the dialog, respectively.
<li><a name="delete"><b>Delete Sequences/Gaps...</b></a>
open a dialog for deleting sequences and/or all-gap columns.
One or more sequences can be slated for deletion by choosing their names
from the list.  <b>Ctrl</b>-click toggles the status of a sequence.
To choose a block of sequences without dragging,
click on the first (or last) and then <b>Shift</b>-click 
on the last (or first) in the desired block.
<b>Delete all-gap columns</b> indicates that columns containing only
gaps (evaluated after sequence deletion, if any) should be removed.
Clicking <b>Apply</b> (or <b>OK</b>, which also closes the dialog) 
performs the indicated deletions.
<br><br>
<li><a name="add"><b>Add Sequence</b></a> -
open a dialog for adding a new sequence to the alignment. 
<b>The existing alignment will not be changed, except that gaps
may be inserted <i>between</i> its columns.</b>
Unless simply appended as described below,
a new sequence will be added to the alignment using the Needleman-Wunsch
algorithm and the indicated <a href="#parameters">scoring parameters</a> as
applied to <b>All sequences</b> or only the <b>Specified sequences</b>.
Either way, the result will include all sequences, but sometimes a better
alignment can be obtained by omitting hard-to-align sequences 
(those not similar to the new sequence) from the scoring calculations.
Clicking <b>Apply</b> (or <b>OK</b>, which also closes the dialog) 
adds the sequence to the alignment.
<br><br>
Sequence input options:
  <ul>
  <li> <a name="addplain"><b>Plain Text</b></a> - add a sequence
  that has been entered or pasted in the <b>Sequence</b> field as plain text.
  It will be converted to upper case and any spaces removed automatically.
  The sequence should not contain gap characters, unless (with gaps) it is
  exactly the same length as the existing alignment and the option to
  <b>Simply append...</b> is used.
  <li> <a name="addstruct"><b>From Structure</b></a>
  - add the sequence of a structure chain open in Chimera 
  <li> <a name="addfile"><b>From File</b></a>
  - add one or more sequences from a file:
    <ul>
    <li>A single sequence or multiple unaligned sequences 
    (no gap characters) can be read from a file in 
   <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp#search"
    target="_blank">FASTA</a> or 
    <a href="https://www.bioinformatics.nl/tools/crab_pir.html" 
    target="_blank">NBRF/PIR</a> format.
    The sequences do not need to be of equal length;
    they will be added one by one in their order of occurrence 
    in the file, each addition leaving the preceding alignment 
    unchanged except by the possible insertion of gap columns.
    <li>An alignment (possibly containing gap characters) in any of the
    <a href="../../UsersGuide/filetypes.html#alignment">known formats</a>
    can be appended if it is exactly the same length as the existing alignment
    and the option to <b>Simply append...</b> is used.
    </ul>
  <li> <a name="adduniprot"><b>From UniProt</b></a>
  - add a sequence from 
  <a href="https://www.uniprot.org/" target="_blank">UniProt</a>
  and assign its feature annotations as <a href="#regions">regions</a> 
  (much as described for
  <a href="../uniprot/uniprot.html#seqanno"><b>PDB/UniProt Info</b></a>,
  except independent of structure). UniProt's
  <a href="https://www.uniprot.org/uploadlists/" target="_blank">ID 
  mapping service</a>
  can be used to obtain UniProt identifiers from other sequence
  database identifiers.
  </ul><br>
<a name="parameters">Alignment parameters</a>:
  <ul>
  <li> <a name="AAscore"><b>Matrix</b> (default <b>BLOSUM-62</b>)</a> - 
  <a href="../../UsersGuide/midas/mmaker.html#matrix">substitution matrix</a>
  for scoring the alignment.  The score for aligning a residue in the new 
  sequence with a column in the existing alignment is the sum of the 
  matrix values for the residue versus each of the residues in the column 
  (and zero versus a gap position)
  divided by the number of sequences (rows) in the existing alignment. 
  This contribution over the entire alignment is the <b><i>residue
  similarity score</i></b>.  In some cases there will also be a
  <a href="#SSscore"><b><i>secondary structure score</i></b></a>.
  <li><a name="penalty"><b>Opening penalty</b> (default <b>12</b>)</a>
   - penalty for opening
  a gap in the existing alignment, regardless of whether it is already
  flanked by gaps in any of the sequences. 
  For opening a gap in the new sequence, 
  this penalty is scaled by the fraction of sequences
  in the existing alignment not gapped at that position.
  When <a href="#SSscore">secondary structure scoring</a> is used, 
  secondary-structure-specific gap opening penalties apply instead.
  <li> <b>Extension penalty</b> (default <b>1</b>) - penalty for extending 
  a gap between columns of the existing alignment.  
  For extending a gap in the new sequence, 
  this penalty is scaled by the fraction of sequences
  in the existing alignment not gapped at that position.
  <li> <b>Character</b> (default is the most prevalent gap character in the 
  existing alignment, or the character <b>.</b> if the alignment lacks gaps)
  - character for displaying 
  gap positions created during alignment of the new sequence
  <li><a name="SSscore"><b>Include secondary structure score (<i>N</i>%)</b></a>
  - include a secondary structure term in the alignment score
  and use secondary-structure-specific gap opening penalties.
  This option only applies when one is adding a sequence 
  <a href="#addstruct">from a structure</a> and
  at least one other sequence in the alignment is already
  <a href="#association">associated</a> with a structure.
  Detailed settings can be shown by clicking <b>Show parameters</b>, 
  hidden again by clicking <b>Hide parameters</b>.  
  <b><i>N</i></b> reflects the relative weights of the terms,
  which can be adjusted by moving the slider.  If N is 30, for example,
  <blockquote>
total score = 0.70(residue similarity score) + 0.30(secondary structure score)
&ndash; gap penalties
</blockquote>
The values in the secondary structure <b>Scoring matrix</b>
(for all pairwise combinations of <b>H</b> helix, <b>S</b> strand, and
<b>O</b> other) and the secondary-structure-specific
<b>Gap opening penalties</b> can be adjusted.
The secondary structure score is computed in the same way as described for the
<a href="#AAscore">residue similarity score</a>, except that sequences
without associated structures contribute neither to the score nor to the
sequence count used for normalization.  
When secondary structure scoring is used (even if the slider is
set to zero), 
<ul>
<li>the secondary-structure-specific <b>Gap opening penalties</b>
are used instead of the single <a href="#penalty"><b>Opening penalty</b></a>
<li>the <b>Extension penalty</b> remains the same
<li>the total penalty is computed as described <a href="#penalty">above</a>,
except that sequences without associated structures contribute neither to
the penalty nor to the sequence count used for normalization
</ul>
Contributions from multiple structures associated with the same sequence in 
the alignment are averaged so that the sequence only counts &ldquo;once.&rdquo;
<p>
The alignment parameter values last used for adding a sequence are retained. 
However, <b>Reset...</b> buttons are provided for restoring the general
alignment parameters (those other than secondary structure scoring) and
secondary structure scoring parameters to their factory defaults.
</p>
</blockquote>
  </ul>
<li><a name="realign"><b>Realign Sequences...</b></a>
use a Clustal Omega or MUSCLE web service to <a href="realign.html">realign</a>
all of the sequences in the current alignment, either replacing the
current alignment or opening the result in a new
<a href="#seqwindow">sequence window</a>
<li><a name="editanno"><b>Alignment Annotations...</b></a>
open a dialog for viewing and editing
<a href="#anno-ali">alignment annotations</a>
<li><a name="editname"><b>Edit Sequence Name...</b></a>
allow entering a new <a href="#seqnames">name</a> 
for the sequence chosen from the pulldown menu
<li><a name="showkeys"><b>Show Editing Keys...</b></a>
bring up a dialog summarizing <a href="#editkeys"><b>Ctrl</b>-key 
editing functions</a>
<li><a name="new-window"><b>Region &rarr; New Window</b></a>
- open <a href="#activeregion">active region</a> contents in a new
<a href="#seqwindow">sequence window</a> (single-block regions only)
</ul>
<a name="seqsearch">
The next three options allow searching for occurrences of a specified 
string or pattern of residues in one or all sequences</a>; matches
are made into a <a href="#regions">region</a> named <b>search result</b>:
<ul>
<li><b>Find Subsequence...</b> find matches to a string,
potentially with <b>Ambiguity codes</b>:
<ul>
<li><b>none</b> - only allow exact matches of one-letter codes
<li><b>protein</b> - in addition to exact one-letter code matching,
allow B to match aspartic acid (D) and asparagine (N)
and Z to match glutamic acid (E) and glutamine (Q)
<li><b>nucleic acid</b> - in addition to exact one-letter code matching,
allow R to match the standard purine nucleotides (A, G), Y to match the
standard pyrimidine nucleotides (C, T, U), and N to match any of the five
(A, C, G, T, U)
</ul>
<li><b>Find Regular Expression...</b>
find matches to a pattern specified as a 
<a href="https://www.regular-expressions.info/quickstart.html"
target="_blank">regular expression</a>
<li><b>Find PROSITE Pattern...</b> find matches to a pattern specified with 
"https://prosite.expasy.org/scanprosite/scanprosite_doc.html"
target="_blank">PROSITE pattern syntax</a>
(residue codes must be capitalized; extended syntax involving an asterisk, 
<i>e.g.</i> <b><{C}*></b>, is not supported)
<br><br>
<a name="namesearch">
<li><b>Find Sequence Name...</b> find sequences whose names contain or 
exactly match an entered string, where capitalization is important.</a>
Clicking <b>Find</b> shows a matching sequence name in red and
scrolls the alignment vertically to center on it. The sequence's
position in the alignment (#N of M) and number of residues is reported
near the bottom of the sequence window. If there are multiple matches,
clicking <b>Find</b> again locates the next one.
</ul>

<a name="mavmenu-structure">
<h4>Structure</h4></a>
<ul>
<li><b>Load Structures</b> - for sequences not already
<a href="#association">associated</a> with a structure, retrieve
and open the corresponding structure files (as described
for <a href="../../UsersGuide/fetch.html"><b>Fetch by ID</b></a>);
correspondence is determined from <a href="#seqnames">sequence names</a>
according to the rules
given in the <a href="#mavpref-structure"><b>Structure</b> preferences</a>
<li><b>Match...</b> open a dialog for <a href="#superposition">
superimposing structures</a> based on the sequence alignment
<li><b>Assess Match...</b>
open a dialog for <a href="#assessment">evaluating</a>
how well the sequence-aligned residues of structures are superimposed in space
<li><b>Modeller (homology)...</b> show interface to
<a href="modeller.html#comparative">comparative modeling with Modeller</a>
<li><b>Modeller (loops/refinement)...</b> show interface to
<a href="modeller.html#building">building or refinement with Modeller</a>
<li><b>Associations...</b>
open a dialog for <a href="#assocpanel">adding or changing associations</a>
between structures and sequences
<li><b>Secondary Structure</b>
 <ul>
 <li><b>show actual</b> - whether to show <a href="#regions">regions</a> 
 named <b>structure helices</b> (pale yellow with gold outline)
 and <b>structure strands</b> (pale green with darker outline)
 in sequences with <a href="#association">associated</a> structures.
 Protein helix and strand assignments are taken from the input structure
 file or generated with
 <a href="../../UsersGuide/midas/ksdssp.html"><b>ksdssp</b></a>.
 <li><b>show predicted</b> - whether to show <a href="#regions">regions</a> 
 named <b>predicted helices</b> (gold outline) 
 and <b>predicted strands</b> (green outline) 
 in sequences without <a href="#association">associated</a> structures.
 Helices and strands in these sequences are
 <a href="gor.html">predicted using <b>GOR</b></a>.
 </ul>
 <li><b>Select by Conservation...</b> open the 
 <a href="../render/render.html#select"><b>Select by Attribute</b></a> tool
 to <a href="../../UsersGuide/selection.html">select</a> 
 residues in sequence-<a href="#association">associated</a>
 structures by <a href="#mavPercentConserved">conservation</a>
 <li><b>Render by Conservation...</b> open the 
 <a href="../render/render.html#render"><b>Render by Attribute</b></a> tool
 to color or otherwise change the appearance of 
 residues in sequence-<a href="#association">associated</a>
 structures by <a href="#mavPercentConserved">conservation</a>
 <li><b>Expand Selection to Columns</b>
 - if any sequence-<a href="#association">associated</a> residues are
 <a href="../../UsersGuide/selection.html">selected</a>, 
 expand the selection to include all residues associated
 with the same column(s) of the sequence alignment
 </ul>
</ul>

<a name="mavmenu-headers">
<h4>Headers</h4></a>
<ul>
<li><a name="saveheader"><b>Save...</b></a>
save a <a href="defineheader.html">header definition file</a>
containing alignment positions and header values.
Header values are the characters or numbers 
(shown with bar heights in a histogram) 
in a <a href="#headers">header</a> line such as
<a href="#conservation"><b>Conservation</b></a>
in the <a href="#seqwindow">sequence window</a>.
No-value positions (for example, 
when the AL2CO <a href="#conservation"><b>Conservation style</b></a>
is used, positions with a gap in more than 
<a href="#gapfract"><b>gap fraction</b></a> of the sequences)
can be omitted from the file.
<li><a name="loadheader"><b>Load...</b></a>
read a <a href="defineheader.html">header definition file</a> to
show a new header line and have its values available as
<a href="#mavAttributes">residue attributes</a> in any 
<a href="#association">associated</a> structures
</ul>
The available <a href="#headers">headers</a> 
are listed in the bottom part of the menu;
checkboxes control which are shown in the
<a href="#seqwindow">sequence window</a>:
<ul>
<li> <a href="#consensus"><b>Consensus</b></a>
<li> <a href="#conservation"><b>Conservation</b></a>
<li> <b><i>another_header_name</i></b>
<li> (<i>etc.</i>)
</ul>

<a name="mavmenu-numberings">
<h4>Numberings</h4></a>
<ul>
<li><b>Overall Alignment</b> - whether to show 
<a href="#numbering">alignment numbering</a> 
</ul>
<ul>
<li><b>Left Sequence</b> - whether to show 
<a href="#numbering">sequence numbering</a>
to the left of each row of sequences
<li><b>Right Sequence</b> - whether to show
<a href="#numbering">sequence numbering</a>
to the right of each row of sequences
<li><a name="seqnum"><b>Adjust Sequence Numberings...</b></a>
open a dialog for specifying sequence <a href="#numbering">start numbers</a>.
One or more sequences can be chosen with the mouse.
<b>Ctrl</b>-click toggles the status of a sequence.
To choose a block of sequences without dragging,
click on the first (or last) and then <b>Shift</b>-click 
on the last (or first) in the desired block.
Clicking <b>Apply</b> (or <b>OK</b>, which also closes the dialog)
applies the new <b>Start</b> number to the chosen sequence(s).
</ul>

<a name="mavmenu-tree">
<h4>Tree</h4></a>
<ul>
<li><b>Load...</b>
read a <a href="#trees">tree</a> in
<a href="https://evolution.genetics.washington.edu/phylip/newicktree.html"
target="_blank">Newick format</a> and display it to the left of the sequences 
in the <a href="#seqwindow">sequence window</a>
<li><b>Show Tree</b>
- hide/show the previously loaded tree
</ul>
<ul>
<li><b>Extract Subalignment</b> - copy the subalignment defined by the
<a href="#chosen-node">chosen node</a> of the <a href="#trees">tree</a> 
into a separate <a href="#seqwindow">sequence window</a>
</ul>

<a name="mavmenu-tools">
<h4>Info</h4></a>
<ul>
<li><b>Percent Identity...</b> 
calculate the <a href="#pid">percent identity</a>
between pairs of sequences in the alignment
<li><a href="#regionbrowser"><b>Region Browser</b></a>
- manage the existing <a href="#regions">regions</a>
<li><b>Blast Protein...</b> perform a 
<a href="../../UsersGuide/blast.html">protein BLAST</a> search
with a specified sequence
<li><a name="unifetch"><b>UniProt Annotations...</b></a>
map feature annotations from 
<a href="https://www.uniprot.org/" target="_blank">UniProt</a>
onto a sequence as <a href="#regions">regions</a>; 
if the matching database entry lacks feature annotations, or no match is found, 
no regions will be generated.
<p>
<a href="https://www.uniprot.org/" target="_blank"><b>UniProt</b></a>
annotations are fetched as an XML file that can be cached and reused 
as needed depending on the 
<a href="../../UsersGuide/preferences.html#Fetch"><b>Fetch</b> preferences</a>
and whether the option to <b>Ignore any cached data</b> is checked.
The relevant entry can be identified:
</p>
  <ul>
  <li><b>from UniProt ID</b>
- directly by the specified ID, for example, <b>Q92731</b> or <b>esr2_human</b>
  <li><b>from PDB code and chain ID</b>
- using the same lookup mechanism as the
<a href="../uniprot/uniprot.html"><b>PDB/UniProt Info</b></a> tool.
Namely, structure annotations and any PDB-UniProt mapping information 
(UniProt ID and residue number correspondences) are fetched from the
<a href="https://www.rcsb.org" target="_blank">RCSB PDB</a>.
The UniProt ID is reported in the
<a href="../../UsersGuide/reply.html"><b>Reply Log</b></a>.
  <li><b>by Blast search of UniProt (may take minutes)</b>
- <a href="https://www.uniprot.org/help/uniref" target="_blank">UniRef100</a> 
is BLAST-searched for sequences similar to the query; this may take several 
minutes and is handled as a <a href="../idle/task.html">background task</b></a>.
Only the top hit is used, regardless of the quality or presence of feature 
annotations. The UniProt ID and E-value of the top hit are given in the 
<a href="../../UsersGuide/reply.html"><b>Reply Log</b></a>.
<b><i>
The top hit may not be identical to the query sequence, and scientific
judgement must be exercised as to the transferability of any annotations.
</i></b>
  </ul>
<p>
Possible actions are to annotate the sequence with the UniProt 
sequence features and/or show the corresponding UniProt web page(s).
In the <a href="#regionbrowser"><b>Region Browser</b></a>, 
the initial ordering of annotations is alphabetical,
right after any missing structure region.
</p>
</ul>

<a href="#mavprefs"><h4>Preferences</h4></a>

<hr>
<address>UCSF Computer Graphics Laboratory / April 2019</address>
</body>
</html>
