<!doctype html public "-//W3C//DTD HTML 4.0//EN">
<!-- vim: sw=1:
>>> UCSF Chimera Copyright
Copyright (c) 2010 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
<<< UCSF Chimera Copyright
-->
<html>
 <head>
  <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
  <link rel="icon" href="../chimera16.png" type="image/png">
  <title>UCSF Chimera Version 1.5 Release Notes</title>
 </head>

 <body>
  <p>
  <img src="../titleChimera.png" width="370" height="55" alt="logo">
  </p>

  <font size="-1">
   <a href="http://www.cgl.ucsf.edu/chimera/docs/relnotes.html">Release History</a>
  </font>

  <h2>UCSF Chimera Version 1.5 Release Notes</h2>
  <ul>
   <li><a href="#1.5.0">Version 1.5 (build 32133, 11 Dec 2010)</a>
   <li><a href="#1.5.1">Version 1.5.1 (build 32314, 12 Jan 2011)</a>
   <li><a href="#1.5.2">Version 1.5.2 (build 32411, 25 Jan 2011)</a>
   <li><a href="#1.5.3">Version 1.5.3 (build 33475, 24 May 2011)</a>
 </ul>

  <h3>
   <a name="1.5.0">Changes in release 1.5</a>
   since the <a href="1.4.html">previous production release (1.4)</a>:
  </h3>


  <blockquote>

   <h4>Contents</h4>
   <ul>
    <li><a href="#general">General Changes</a>
    <li><a href="#new_tools">New Tools</a>
    <li><a href="#tool_changes">Tool Changes</a>
    <li><a href="#new_commands">New Commands</a>
    <li><a href="#command_changes">Command Changes</a>
    <li><a href="#bugfixes">Notable Bug Fixes</a>
    <li><a href="#bugs">Known Bugs</a>
   </ul>

   <h4><a name="general">General Changes:</a></h4>
   <ul>
    <li>Updated many of the bundled software packages, most notably:
    Python 2.7,
    ScientificPython 2.9.0,
    AmberTools 1.4
    <li><b>Actions... Color</b> menu simplified, <b>all options...</b>
    opens separate dialog with target settings and full set of colors
    <li><b>Actions... Atoms/Bonds</b> menu includes controls for ring fill,
    aromaticity display, nucleotide display
    <li>start-of-line aliases (defined with <b>alias ^</b><i>name</i>) now
    available from top-level <b>Aliases</b> menu
    <li>selection broadening/narrowing cascades now include "nothing,"
    one level narrower than the original selection
    <li>new <b>coil</b> selector specifies all non-helix, non-strand
    amino acid residues
    <li>default label offsets increased for better readability
    <li>appearance of small sticks improved
    <li><b>subdivision quality</b> default 
    (<b>Effects</b>, interactive presets) raised to 1.5
    <li><b>Focus</b> (menu and command) and <b>window</b> command with
    specified targets now consider window height, not just width
    <li>for purposes of <b>Focus</b> (menu and command), 
    ribbon display implies display of backbone atoms rather than all 
    atoms of a residue 
    <li>residue label position can be based on primary atom or backbone only
    rather than all displayed atoms
    <li>ModBase fetch shows additional model scores, where available
    <li>PDB fetching improvements &mdash; personal PDB directories 
    can be flat or hierarchical, files compressed or not
    <li>saving to multiple PDB or Mol2 files allows file naming by model name 
    <li>Mol2 files can be written with Amber/GAFF atom types
    (from <b>Add Charge</b>) instead of Sybyl atom types
    <li>can read PQR format as produced by PDB2PQR v1.6, assign atomic
    charge and radius attributes accordingly
    <li>can read Gromos87 format (single structure, not trajectory)
    <li>added VRML level-of-detail (LOD) support
    <li>added <b>General</b> preference, whether to try to place dialogs 
    beside the main window instead of on top (on by default)
    <li>commands can be copied from the <b>Command History</b> 
    for pasting into another application window
    <li>most histograms (<i>e.g.,</i> <b>Render by Attribute</b>) now automatically
    adjust their displayed range when a value outside the range is entered
    <li><b>Selector Construction Panel</b> removed
    <li>the --script option (used at startup from the system command-line)
    now uses quotes to delimit which arguments go to the script, behaves
    like the <b>runscript</b> command
    <li>the startup option --script and command <b>runscript</b> run scripts in
    Chimera's current directory instead of the directory containing the script
   </ul>

   <h4><a name="new_tools">New Tools:</a></h4>
   <ul>

    <li><b>Metal Geometry</b> (Structure Analysis) &mdash;
    analyze metal coordination geometry

    <li>Modeller interface - graphical interface
    to the <b>Modeller</b> homology modeling program (Sali Lab, UCSF)
    <ul>
     <li>interface currently accessible from the <b>Structure</b> menu
     in <b>MultAlign Viewer</b>
     <li>the <b>SaliLab Model Evaluation Server</b> can be called to compute
     additional model scores
    </ul>

    <li><b>Ramachandran Plot</b> 
    (only accessible from <b>Model Panel</b> currently)

    <li><b>PDB/UniProt Info</b> (Sequence) &mdash; 
    retrieve annotations for PDB entries using a web service from the RCSB 
    (tool is in early stage, change significantly)

    <li><b>Segment Map</b>, <b>Fit to Segments</b> (Volume Data) &mdash; 
    these comprise the <b>Segger</b> package (thanks to Greg Pintilie,
    National Center for Macromolecular Imaging) 

    <li><b>Small-Angle X-Ray Profile</b> (Higher-Order Structure) &mdash; 
    calculate theoretical SAXS profile, compare with experimental profile

   </ul>

   <h4><a name="tool_changes">Tool Changes:</a></h4>
   <ul>

    <li><b>2D Labels</b> (Utilities) &mdash; 
    <ul>
     <li>includes arrows that can be drawn in any orientation
     <li>2D labels including text, symbols, and arrows
     can be written to and read from files (allows transferring
     labels between sessions, hand-editing, saving labels without saving 
     the entire session, <i>etc.</i>)
    </ul>

    <li><b>Add Charge</b> (Structure Analysis, Structure Editing) &mdash; 
    added option to show partial charges as labels

    <li><b>Axes/Planes</b> (Structure Analysis) &mdash; 
    expanded to <b>Axes/Planes/Centroids</b>

    <li><b>Blast Protein</b> (Sequence) &mdash; 
    query from structure chain includes residues from SEQRES (if present)
    even if missing from the coordinates

    <li><b>Build Structure</b> (Structure Editing) &mdash; 
    <ul>
     <li>new <b>Join Models</b> section bonds and merges two models, 
     moving one of them to form the appropriate bond 
     (previously required laborious hand-positioning) 
     <li>other sections renamed to better reflect their uses: 
     <b>Start Structure</b>, <b>Modify Structure</b>, <b>Adjust Bonds</b>
     <li>each section now uses an <b>Apply</b> button
    </ul>

    <li><b>Color Editor</b> (Utilities) &mdash; 
    if a color component value &gt; 1 and &le; 255 is entered, 
    it is automatically divided by 255 to convert it to the 0&ndash;1 range

    <li><b>Coulombic Surface Coloring</b> (Surface/Binding Analysis) &mdash;
    can generate a grid of the values (a volume dataset), thus allowing
    coloring surface caps, showing isosurfaces, saving to a file, <i>etc.</i>

    <li><b>Distances</b>, <b>Angles/Torsions</b> (Structure Analysis) &mdash; 
    <ul>
     <li>added option to select the corresponding atoms/bonds 
     when a measurement is chosen in the table
     <li>angle monitors saved in sessions
    </ul>

    <li><b>Fit in Map</b> (Volume Data) &mdash; 
    correlation can be calculated about zero (now default) or mean values

    <li><b>Find Clashes/Contacts</b> 
    (Structure Analysis, Surface/Binding Analysis) &mdash; 
    allows checking for interactions between two independently designated 
    sets of atoms (not just a designated set against itself or all other atoms)

    <li><b>FindHBond</b> 
    (Structure Analysis, Surface/Binding Analysis) &mdash; 
    <ul>
     <li>allows finding only those H-bonds between the selection and 
     an independently designated set of atoms
     <li>pseudobonds drawn to hydrogens if present instead of donor
     heavy atoms
    </ul>

    <li><b>Lighting</b> (Viewing Controls) &mdash; 
    <ul>
     <li>new modes with different numbers/types of lights,
     including ambient-only
     <li>interface reorganized and simplified, advanced section removed
     (new <b>lighting</b> command provides access to those settings)
    </ul>

    <li><b>Match &rarr; Align</b> (Structure Comparison, Sequence) &mdash; 
    in addition to the RMSD and number of equivalenced positions (alignment
    length), reports Structural Distance Measure (SDM) and Q-score values

    <li><b>MD Movie</b> (MD/Ensemble Analysis) &mdash; 
    <ul>
     <li>support added for "particle" trajectory format, 
     designed for coarse-grained systems
     <li>can read trajectory from multiple XYZ files, one per frame
     <li>can read/concatenate multiple DCD trajectory files
     <li>reads atomic charges from some topology formats (PSF, Amber prmtop)
     <li>can perform clustering (same method as <b>Ensemble Cluster</b> tool)
     <li>better integrated with <b>coordset</b> command
     <li>controller dialog will restore from a saved session if the
     entire trajectory was read in before saving
    </ul>

    <li><b>Measure and Color Blobs</b> (Surface/Binding Analysis, Volume Data)
    &mdash; 
    <ul>
     <li>can generate a new color for each blob
     <li>can draw bounding box aligned with principal axes of inertia
    </ul>

    <li><b>Minimize Structure</b> (Structure Editing) &mdash; 
    can remember settings of secondary dialogs (<b>Dock Prep</b> 
    <i>etc.</i>) for optional use in subsequent minimizations

    <li><b>Model Panel</b> (General Controls) &mdash; 
    added <b>biological unit</b> and <b>Ramachandran plot</b> functions

    <li><b>Movie Recorder</b> (Utilities) &mdash; 
    variable bit rate encoding (Q=8) now default

    <li><b>Multalign Viewer</b> (Sequence) &mdash; 
    <ul>
     <li>Blast protein search available from Tools menu
     <li>trees saved in sessions
     <li>single-sequence files can be read/written
     <li>can read alignments (only) from HSSP files
     <li>vertical scrolling with mouse scroll wheel
     <li>can add sequence(s) from a file 
     (previously just from structure or pasted as plain test)
     <li>alignment headers contain symbols (circle, square, diamond, star)
    </ul>

    <li><b>Phantom Force Feedback</b> (Volume Data) &mdash; removed

    <li><b>Render by Attribute</b> (Depiction, Structure Analysis) &mdash; 
    for convenience in setting colors, added built-in palettes and the
    ability to reverse color order with a single click

    <li><b>ResProp</b> (Structure Analysis) &mdash; 
    coloring section removed

    <li><b>Rotamers</b> (Structure Editing) &mdash; 
    the existing sidechains can be retained

    <li><b>Surfnet</b> (Surface/Binding Analysis) &mdash; 
    simplified into a single dialog without loss of functionality

    <li><b>Thermal Ellipsoids</b> (Structure Analysis) &mdash; 
    <ul>
     <li>anisotropic B-factors read from CIF/mmCIF (previously only PDB)
     <li>depictions saved in sessions
     <li>possible depictions include ellipsoid principal axes 
     and/or principal ellipses in addition to the ellipsoids themselves
     <li>includes <b>Presets</b> menu for rapid access to 
     predefined combinations of options
    </ul>

    <li><b>Transform Coordinates</b> (Movement) &mdash; 
    renamed from <b>Transform Molecule Coordinates</b>,
    now works on all model types

    <li><b>Volume Filter</b> (Volume Data) &mdash; 
    <ul>
     <li>Gaussian filtering more memory-efficient, 
     allows specifying output value type
     <li>added Flatten filter
    </ul>
   </ul>

   <h4><a name="new_commands">New Commands:</a></h4>
   <ul>
    <li><b>coulombic</b> &mdash; command-line implementation of 
    <b>Coulombic Surface Coloring</b>
    <li><b>lighting</b> &mdash; adjust lighting and shininess
   </ul>

   <h4><a name="command_changes">Command Changes:</a></h4>
   <ul>
    <li><b>2dlabels</b> &mdash; 
    <ul>
     <li>includes 2D arrows in any orientation
     <li>2D labels including text, symbols, and arrows
     can be written to and read from files (allows transferring
     labels between sessions, hand-editing, saving labels without saving 
     the entire session, <i>etc.</i>)
    </ul>
    <li><b>align</b> &mdash; can align an axis or plane object 
    (previously worked only on atoms)
    <li><b>angle</b> &mdash; can measure angles between axes and/or planes
    <li><b>aniso</b> &mdash; added options for display of thermal ellipsoid 
    principal axes and/or principal ellipses, other changes as listed above 
    for <b>Thermal Ellipsoids</b>
    <li><b>close</b> &mdash; more than one model number can be given
    <li><b>define</b> &mdash; can calculate centroids 
    (previously just axes and planes)
    <li><b>distance</b> &mdash; can measure distances involving axes, planes,
    and/or centroids
    <li><b>findclash</b> &mdash; 
    allows checking for interactions between two independently specified 
    sets of atoms (not just a specified set against itself or all other atoms)
    <li><b>findhbond</b> (<b>hbonds</b>) &mdash; 
    allows finding only those H-bonds between the selection and 
    an independently designated set of atoms; pseudobonds drawn to hydrogens 
    if present instead of donor heavy atoms
    <li><b>help</b> &mdash; use without arguments opens the search dialog
    <li><b>measure</b> &mdash; added <b>spine</b> calculation of a path along
    the center line of a segmentation region from <b>Segment Map</b>
    <li><b>morph</b> &mdash; added <b>nogui true</b> option to suppress
    <b>MD Movie</b> dialog
    <li><b>movie</b> &mdash; 
    <ul>
     <li>variable bit rate encoding (Q=8) now default
     <li>new <b>crossfade</b> option for transition by linear interpolation
    </ul>
    <li><b>nucleotides</b> &mdash; added <b>~nucleotides</b> for convenience
    (equivalent to <b>nuc side atoms</b>)
    <li><b>rock, roll, turn</b> &mdash; a center of rotation method can
    be specified just for executing the command 
    (for example, allows rotating around the center of view with the command
    even if a different center is used for rotation with the mouse)
    <li><b>setattr</b> &mdash; added M flag, all applicable models
    <li><b>shape</b> &mdash; can generate ellipsoids, rectangles, flat ribbons,
    apply specified rotations to some shapes
    <li><b>swapaa</b> &mdash; the existing sidechains can be retained
    <li><b>sym</b> &mdash; option to generate low-res surfaces instead of 
    atomic copies for efficiency on large systems
    <li><b>vop</b> &mdash; added <b>flatten</b> and <b>unbend</b> operations,
    <b>gaussian</b> allows specifying output value type
    <li>redundant commands removed: <b>load</b>, <b>source</b>, <b>rescolor</b>,
    <b>resrepr</b>, <b>x3dsave</b>, <b>brotation</b>, <b>push</b>, <b>pop</b>  
   </ul>

   <h4><a name="bugfixes">Notable Bug Fixes:</a></h4>
   <ul>
    <li>transparent ribbons no longer contain extraneous disks
    <li>eliminated blocky artifacts that could occur around labels
    in a scene with transparency
    <li>when possible, shadows are smoother
    <li>shadows and labels can be used at the same time
    <li>saving images now uses OpenGL framebuffer objects
    for off-screen rendering to work around problems with some platforms
    (mostly older Macs and some Linux systems)
    <li>avoid crashes and reduce memory use when saving ribbon images
    by not subdividing as much &mdash; if you notice a difference,
    use the <b>Effects</b> tool to increase the subdivision quality as needed
   </ul>

   <h4><a name="bugs">Known Bugs:</a></h4>
   <ul>
    <li>surfaces fail for some structures
    <li>POV-Ray incorrectly renders some of the characters in the fonts
    Chimera uses, especially the zero
    <li>the inside ribbon color is not exported
    (<i>e.g.,</i> POV-ray does not show it)
    <li>Mac OS X problems:
    <ul>
     <li> in Mac OS X 10.6 Chimera sometimes crashes in the Mac OpenGL
     driver when saving images or zooming in on molecule ribbons or when
     displaying large surfaces.
     <li> in Mac OS X 10.6.4,
     the display of large surfaces (&gt; 15 million triangles)
     can cause the machine to freeze requiring a reboot.  This is a
     graphics driver bug effecting both Nvidia and ATI graphics cards.
     This appears to be fixed in Mac OS 10.6.5.
     <li> in Mac OS X 10.5, opening and then closing IDLE adds 
     the IDLE menus to the Chimera menus, 
     and trying to use the extra menus generates errors
     <li>in sequential stereo, the drag selection outline is only visible
     for a fraction of a second when the mouse button is released
    </ul>
    <li>Mac OS X with X Windows problems:
    <ul>
     <li>Observed with 10.5 (Leopard):
     <ul>
      <li> sometimes display does not update
      (can be forced by slightly moving the view contents with the mouse)
      <li> sometimes saved images have a corrupt or "mis-tiled" appearance.
      Workarounds to try include:
      <ul>
       <li>showing the <b>Command Line</b> if it is not shown 
       (or if it is shown, hiding it) and then re-saving the image
       <li>saving the position with command <b>savepos</b>, jiggling
       the view with the mouse, restoring the position with <b>reset</b>,
       and then re-saving the image
       <li>saving a session, then using the native Mac version of Chimera
       (non-X11) to restore the session and re-save the image
      </ul>
     </ul>
     <li>Observed with 10.6 (Snow Leopard):
     <ul>
      <li> <b>Side View</b> come up blank, 
      but will be filled in when structures are moved
      <li>closing the Viewing dialog (<b>Side View</b>, <b>Lighting</b>, etc.)
      causes chimera to crash on next graphics redraw
     </ul>
    </ul>
   </ul>

   <hr>
   <h3><a name="1.5.1">Changes in release 1.5.1</a> relative to 
    <a href="#1.5.0">release (1.5)</a>:</h3>

   <h4>Major Bugs Fixed:</h4>
   <ul>
    <li>On Linux, installing as root
    would create unreadable .desktop files on the user's desktop.
    <li>For <i>swapna</i> command, better geometry for swapped-in base rings.
    <li>On Mac Aqua, bug where dialogs would be placed almost completely offscreen
    if they couldn't be placed without overlapping graphics window fixed.
    <li>Fixed problem where Write Prmtop/Add Ions would complain about "missing GAFF types" for NMR structures with missing residues.
    <li>Fixed bug in hbond-guided hydrogen addition where metal-coordinating atoms were also involved in possible h-bonds.
    <li>Fixed problem with too many dashes in POV-Ray output.
    <li>Fixed problem restoring session with a uncolored molecular surface.
    <li>Various Unicode fixes.
   </ul>

   <hr>
   <h3><a name="1.5.2">Changes in release 1.5.2</a> relative to 
    <a href="#1.5.1">release 1.5.1</a>:</h3>

   <h4>Major Bugs Fixed:</h4>
   <ul>
    <li>On all platforms, registration no longer uses expired certificate.
    <li>Prevent invalid PubChem IDs from producing a traceback.
    <li>More Unicode fixes.
   </ul>

   <hr>
   <h3><a name="1.5.3">Changes in release 1.5.3</a> relative to 
    <a href="#1.5.2">release 1.5.2</a>:</h3>

   <h4>Major Bugs Fixed:</h4>
   <ul>
    <li>Regression: adjust line width when saving images so they are the same relative size
    <li>Regression: fix width and height arguments to copy command to work as documented (if only one is given, use aspect ratio to compute the other)
    <li>On Windows, user names with Unicode characters work.
    <li>In exported data, normalize the rotation axis (<i>e.g.,</i> for cylinders).
   </ul>

   <h4>Additional Known Bugs:</h4>
   <ul>
    <li>On Linux, pressing control-MouseWheel causes chimera to crash.
    <li>On Macs with ATI graphics, row-interleaved stereo doesn't work.
    <li>On Windows, chimera will not work if installed in a directory with non-ASCII characters in the path.
    <li>On Windows XP, tear-off menus don't work.
    <li>On Mac OS X 10.6, the <b>Write Prmtop</b> tool can fail for systems with non-standard residues.
   </ul>

  </blockquote>
 </body>
</html>
