<!doctype html public "-//W3C//DTD HTML 4.0//EN">
<!-- vim: sw=1:
>>> UCSF Chimera Copyright
Copyright (c) 2012 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
<<< UCSF Chimera Copyright
-->
<html>
 <head>
  <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
  <link rel="icon" href="../chimera16.png" type="image/png">
  <title>UCSF Chimera Version 1.6 Build 35700 Release Notes</title>
 </head>

 <body>
  <p>
  <img src="../titleChimera.png" width="370" height="55" alt="logo">
  </p>

<font size="-1">
<a href="http://www.cgl.ucsf.edu/chimera/docs/relnotes.html">Release History</a>
</font>

  <h2>UCSF Chimera Version 1.6 Release Notes</h2>
  <ul>
   <li><a href="#1.6.0">Version 1.6 (build 35700, 29 Feb 2012)</a>
   <li><a href="#1.6.1">Version 1.6.1 (build 35849, 13 March 2012)</a>
   <li><a href="#1.6.2">Version 1.6.2 (build 36855, 6 July 2012)</a>
 </ul>

  <h3>
   <a name="1.6.0">Changes in release 1.6</a>
   since the <a href="1.5.html">previous production release (1.5)</a>:
  </h3>

  <ul>
   <li><a href="#general">General Changes</a>
   <li><a href="#new_tools">New Tools</a>
   <li><a href="#tool_changes">Tool Changes</a>
   <li><a href="#new_commands">New Commands</a>
   <li><a href="#command_changes">Command Changes</a>
   <li><a href="#bug_fixes">Notable Bug Fixes</a>
   <li><a href="#bugs">Known Bugs</a>
  </ul>

  <blockquote>
   <h4><a name="general">General Changes:</a></h4>
   <ul>
    <li>new <b>Rapid Access</b> interface to frequently used data and tools
    shown automatically upon startup, but can be hidden/shown at any time
    by clicking the lightning-bolt icon, or hidden by opening 3D data
    <li>new fetch type <b>UniProt</b>, protein sequence and annotations
    <li>default fetch directory set to ~/Downloads/Chimera (previously none)
    and option to ignore cached data added to <b>Fetch by ID</b> dialog
    <li>changed default model colors
    <li>labels drawn in front (default, <b>Background</b> preferences)
    <li>ribbon <b>cardinal</b> spline option for closer tracking 
    of backbone atom positions, associated parameters for additional
    smoothing of strand and/or coil, ribbon &ldquo;stiffness&rdquo; 
    (molecule model attributes, see also <b>ribspline</b> command)
    <li>global <b>single-layer transparency</b>, on by default in <b>Effects</b>
    (previously, single-layer transparency was per-surface and unavailable 
    for ribbons and atoms; this individual surface attribute has been removed). 
    Rendering only the topmost transparent layer 
    simplifies the display and effectively de-emphasizes transparent parts.
    <li>the background can be set to a vertical multicolor gradient or to
    an image read from a file
    <li><b>Color Actions</b> dialog includes "more..." buttons to show
    additional background and depth-cueing settings
    <li>PDB chain "molecule" info reported in Reply Log upon input,
    shown in status line when the cursor is paused over the chain
    <li>new residue attribute <b>uniprotIndex</b> 
    (residue number in the corresponding UniProt sequence, determined from 
    PDB header information using a web service provided by the RCSB PDB)
    <li>symmetry matrices read from mmCIF (previously only PDB)
    <li>new input/output type, <b>spheres</b> in the sphgen format used by DOCK
    <li>ViewDock reads <b>Maestro</b> docking output
    <li>new export type, <b>WebGL</b> (experimental feature)
    <li>CASTp &mdash;
      <ul>
      <li>pocket information saved in sessions
      <li>pocket surfaces optionally colored to show hydrophobicity
      </ul>
    <li><b>Save Image</b> dialog includes option to save temporary session file
    <li>maximum image supersampling reduced from 6x6 to 4x4 because high
    levels increase computational demands without further visible improvement
    <li>changing the <b>horizontal field of view</b> camera parameter
    no longer changes the camera Z-position, and changing the width 
    of the graphics window no longer changes either of those
    <li>saved positions include horizontal field of view value
    <li>new camera modes: dome, truncated dome
    <li>BILD/VRML text support added (see BILD .cmov and .font directives)
    <li>scaling with mouse scroll wheel and touchpad scrolling can be turned
    off (<b>Mouse</b> preferences)
   </ul>

   <h4><a name="new_tools">New Tools:</a></h4>
   <ul>
    <li><b>Animation</b> (Utilities) 
    <b>**under development, not fully functional, may change significantly**</b>
    &mdash; save/restore Chimera <b><i>scenes</i></b>
    <li><b>Cage Builder</b> (Higher-Order Structure) &mdash; 
    create polyhedral cages composed of hexagons, pentagons, and other polygons
    <li><b>Change Chain IDs</b> (Structure Editing)
    <li><b>MultiFit</b> (Volume Data) &mdash; 
    simultaneous rigid fitting of multiple structures into density
    (uses RBVI web server running the program MultiFit from the Sali lab)
    <li><b>Palette Editor</b> (Utilities) &mdash; 
    create and choose palettes (series of colors and interpolation method;
    currently used for background color gradients)
    <li><b>Renumber Residues</b> (Structure Editing)
   </ul>

   <h4><a name="tool_changes">Tool Changes:</a></h4>
   <ul>

    <li><b>AddCharge</b> (Structure Analysis, Structure Editing) &mdash; 
    ATP, ADP, GTP, GDP now treated as standard residues
    (charge lookup from Amber parameter database, Bryce Group, Manchester) 

    <li><b>AddH</b> (Structure Editing) &mdash; 
     <ul>
     <li>less common protonation states of Asp, Glu, Lys, Cys can be specified
     <li>new default is to consider each model in isolation, otherwise other
      models in the vicinity may affect hydrogen placement
     </ul>

    <li><b>Adjust Torsions</b> (Structure Editing) &mdash; 
    terminal bonds can be rotated

    <li><b>Blast Protein</b> (Sequence) &mdash; 
      many more columns of information can be shown for pdb hits
     (ligands, resolution, <i>etc.</i>)

    <li><b>Build Structure</b> (Structure Editing) &mdash; 
     <ul>
     <li>easier peptide linking with <b>Join Models</b>
     <li>new <b>Invert</b> section for swapping substituents
     </ul>

    <li><b>Color Key</b> (Utilities) &mdash; 
    numeric color key labels can be spaced in proportion to their values
    (previously, spacing was always equal)

    <li><b>Effects</b> (Viewing Controls) &mdash; 
    global <b>single-layer transparency</b> setting

    <li><b>Ensemble Match</b> (Structure Comparison, MD/Ensemble Analysis) &mdash; 
    RMSD values can be exported collectively to a text file

    <li><b>FindHBond</b> (Structure Analysis, Surface/Binding Analysis) &mdash; 
     added option to label H-bonds with distances

    <li><b>Fit in Map</b> (Volume Data) &mdash; now takes up to 1000 steps
    (formerly 100)

    <li><b>MatchMaker</b> (Structure Comparison) &mdash; 
    can be restricted to selected residues

    <li><b>MD Movie</b> (MD/Ensemble Analysis) &mdash; 
     <ul>
     <li>can plot distances and angles vs. frame number, save values to file
     <li>RMSD map values can be saved to a file 
     <li>trajectory reading speeded up for
	formats that read their coordinate files directly in the Python layer
	by adding memory-mapping support
	(<i>i.e.</i>, all formats except PDB, AMBER, xyz, and particle)
     <li>input option "NAMD (prmtop/DCD)" for those using NAMD with 
     AMBER prmtop files
     </ul>

    <li><b>Metal Geometry</b> (Structure Analysis) &mdash;
      <ul>
      <li>potential coordination geometries now assessed by
      "distance RMSD" from idealized atom positions rather than "angle RMSD," 
      as the former is a better indicator of the correct geometry
      <li>alternate location handling added
      <li>session support added
      <li>idealized geometry for <b>cube</b> corrected
      </ul>

    <li><b>Minimize Structure</b> (Structure Editing) &mdash; 
    allows conjugate gradient minimization (previously just steepest descent)

    <li><b>Model Panel</b> (General Controls) &mdash;
     <ul>
     <li>group/ungroup function to collapse multiple entries into a single row; 
     grouping is hierarchical (groups may contain other groups) and
     multi-model PDBs such as NMR ensembles are grouped by default
     <li>reorganized to show favorites vs. all instead of
    mutually exclusive lists of frequently/infrequently used functions
     <li>save sphgen function to write sphere format for DOCK
    </ul>

    <li><b>Movie Recorder</b> (Utilities) &mdash; 
     several settings moved to preferences, new <b>Movie</b> category

    <li><b>Multalign Viewer</b> (Sequence) &mdash;
    <ul>
     <li>sequences fetched from UniProt (using the <b>PDB/UniProt Info</b> tool,
     <b>Fetch by ID</b>, or Multalign Viewer <b>Edit... Add Sequence</b>) 
     include UniProt feature annotations as regions
     <li>UniProt feature annotations can be mapped as regions
     onto already open sequences using <b>Info... UniProt Annotations</b>
     <li>menu changes (<b>Tools</b> renamed <b>Info</b> and some of its 
     entries moved to the <b>File</b> menu)
     <li>several <b>Region Browser</b> improvements for handling potentially
     large numbers of annotations, including easier coloring, <b>Info</b> button
     to report included residue ranges
     <li>mousing over a region shows the region name in a pop-up "balloon" 
     (can be turned off in the <b>Regions</b> preferences)
     <li>double-clicking a region opens the <b>Region Browser</b>
     <li>calculation of region overall RMSD
     <li><b>Modeller</b> interface allows including water and/or 
     other HETATM residues from the template in the homology models, 
     some loop refinement
     <li>system &ldquo;copy&rdquo; shortcut (<i>e.g.</i>, command-C on Mac)
     works to copy sequence into text buffer
    </ul>

    <li><b>Multiscale Models</b> (Higher-Order Structure) &mdash;
     <ul>
     <li>options to produce packed unit cells
     <li>multiscale model creation can be limited to specific models
     by selecting them
     </ul>

    <li><b>PDB/UniProt Info</b> (Sequence) &mdash; 
      fetches UniProt sequence annotations and loads them as regions 
      in <b>Multalign Viewer</b>

    <li><b>Surface Color</b> (Volume Data, Surface/Binding Analysis) &mdash; 
     added option to report value at mouse position

    <li><b>Unit Cell</b> (Higher-Order Structure) &mdash;
    added option to produce packed unit cells

    <li><b>ViewDock</b> (Surface/Binding Analysis) &mdash;
    reads Maestro/Glide output

    <li><b>Volume Tracer</b> (Volume Data) &mdash; 
    which features are shown in the dialog can be saved/restored

   </ul>

   <h4><a name="new_commands">New Commands:</a></h4>
   <ul>
    <li><b>background</b> &mdash; set background color(s) or image
    <li><b>changechains</b> &mdash; reassign chain IDs
    <li><b>fitmap</b> &mdash; command implementation of <b>Fit in Map</b>,
    **plus features not available in that tool**:
      <ul>
      <li>global search with random initial placement
      <li>sequential fitting of multiple different structures
      <li>symmetric fitting of copies of the same structure 
      </ul>
    <li><b>invert</b> &mdash; swap substituents
    <li><b>resrenumber</b> &mdash; renumber residues
    <li><b>ribclass</b> &mdash; set ribbon residue class: 
    which atoms control ribbon path and which are hidden by ribbon 
    <li><b>ribspline</b> &mdash; set ribbon path method: B-spline or cardinal
    spline; control additional smoothing and &ldquo;stiffness&rdquo;
    of cardinal spline ribbon
    <li><b>scene</b> &mdash; save and restore <b><i>scenes</i></b>, 
    which include positions, display status and styles, labels, etc.
    (see also new tool: <b>Animation</b>)
    <li><b>zonesel</b> &mdash; select atoms and/or surfaces within a cutoff 
    distance of specified atoms and/or surfaces (this command allows defining 
    zones by surfaces, whereas the standard command-line zone operators 
    &lt; and &gt; refer to atoms only)
   </ul>

   <h4><a name="command_changes">Command Changes:</a></h4>
   <ul>
    <li><b>addaa</b> &mdash; can add to N-terminus (previously just C-terminus)
    <li><b>addh</b> &mdash; 
     <ul>
     <li>less common protonation states of Asp, Glu, Lys, Cys can be specified
     <li>new default is to consider each model in isolation, otherwise other
      models in the vicinity may affect hydrogen placement
     </ul>
    <li><b>clip</b> &mdash; 
     can turn global clipping on and off
    <li><b>crystalcontacts</b> &mdash; 
     can report residue minimum contact distance, maximum buried SASA
    <li><b>findhbond</b> (<b>hbonds</b>) &mdash; 
     added option to label H-bonds with distances
    <li><b>ksddssp</b> &mdash; added -v flag for verbose mode, reports
    helix and strand residue ranges and whether interstrand relationships 
    are parallel or antiparallel
    <li><b>labelopt</b> &mdash; significantly enhanced to define a broad
     range of custom labels for both atoms and residues
    <li><b>mask</b> &mdash; input volume can be specified as <b>ones</b>
    (all values 1); previously had to be an existing data set
    <li><b>match</b> &mdash; 
       <ul>
       <li>can move subparts of a model (only certain
       chains, residues, atoms) instead of the whole model
       <li>atom specifications can overlap and/or span multiple models
       <li>options can follow the atom specifications
       </ul>
    <li><b>matchmaker</b> &mdash; can be restricted to specified residues
    <li><b>mcopy</b> &mdash; option to copy coordinates relative to target
    <li><b>measure</b> &mdash; 
     <ul>
     <li><b>area</b>, <b>volume</b> now work on individual surface pieces
     <li><b>center</b> 
     to calculate center of mass of map and/or atoms
     <li><b>contactArea</b> 
    to report the area of one surface within a cutoff distance of another
     <li><b>distance</b> 
    to report the closest distance between one set of atoms and/or surface
    pieces and another set
     <li><b>mapValues</b> 
    to interpolate map values at atom positions, assign as an attribute
    (command-line implementation of <b>Values at Atom Positions</b> tool)
     <li><b>pathLength</b>
    to report the total length of a set of bonds
     <li><b>rotation</b> &mdash; the transformation can be reported in the
     coordinate system of a third model
     <li><b>symmetry</b> to identify map symmetry
     </ul>
    <li><b>minimize</b> &mdash; 
      <ul>
      <li>settings for conjugate gradient minimization
      <li>parts of structures can be ignored
      <li>syntax simplified, allows freezing atoms without making selections
      <li>options to suppress caching and re-prepping
      </ul>
    <li><b>molmap</b> &mdash; 
    symmetry options for making maps corresponding to symmetrical multimers 
    <li><b>movie</b> &mdash; recording <b>size</b> option allows creating
    movies with image dimensions larger than the Chimera window
    <li><b>pause</b> &mdash; <b>end</b> option to exit script immediately
    <li><b>rmsd</b> &mdash; 
       atom specifications can overlap and/or span multiple models
    <li><b>scolor</b> &mdash; 
    <b>zone</b> option (command-line implementation of <b>Color Zone</b> tool)
    <li><b>segment</b> &mdash; more operations on segmentation regions:
     <ul>
     <li><b>directioncolor</b> 
     to color by principal axis orientation
     <li><b>exportmask</b> 
     to export the region index mask as a map file
     <li><b>sliceimage</b> 
     to create montage images of density map slices
     </ul>
    <li><b>set</b> &mdash; added <b>singlelayer</b> 
       (global single-layer transparency)
    <li><b>shape</b> &mdash; added <b>cone</b>
    <li><b>sop</b> &mdash; 
      <ul>
      <li><b>cap</b>
      (command implementation of <b>Surface Capping</b> tool)
      <li><b>clip</b> to spherically clip volume isosurfaces
      <li><b>hideDust</b>
      (command implementation of <b>Hide Dust</b> tool)
      <li><b>zone</b>
      (command implementation of <b>Surface Zone</b> tool)
      </ul>
    <li><b>stereo</b> &mdash; implemented <b>~stereo</b> 
     (equivalent to <b>stereo mono</b>)
    <li><b>swapna</b> &mdash; new default is to preserve existing 
    glycosidic bond length and angle
    <li><b>sym</b> &mdash; 
       <ul>
       <li>can use the symmetry of another model or of polygons within
       a cage model from <b>Cage Builder</b>
       <li>more symmetry groups (tetrahedral, octahedral), products of groups
       <li>changed syntax for helical and translational symmetries 
       <li>the model number of loaded copies can be specified 
       <li>the resolution of low-res surfaces can be specified
       <li>can generate BIOMT matrices for the molecule
       <li><b>occupancy</b> keyword allows filling only a fraction of positions
       </ul>
    <li><b>vop</b> &mdash; 
      <ul>
      <li><b>tile</b> to make a single-plane volume from slices of another
      <li><b>unroll</b> to make a flat slab from a cylindrical slab
      <li><b>zone</b> to mask within or beyond a zone of specified atoms
      </ul>
   </ul>

   <h4><a name="bug_fixes">Notable Bug Fixes:</a></h4>
   <ul>
   <li>Fixed memory leak where closed models were not having their associated memory released. Leak has existed since at least the 1.3 release, but became more widespread in the 1.4 and later releases since presets trigger the condition needed for the leak to occur.
    <li>The inside ribbon color is now exported (<i>e.g.,</i> POV-ray shows it)
    <li>On Windows, non-western user names now work except for compressed TIFF files and the conic command.
    <li>Fixed STL export so the triangle normals match the triangles,
    not one of the vertices.
    <li>Fixed bond rotation bookkeeping errors that occasionally caused
    crashes or rotated the wrong atoms.
   </ul>

   <h4><a name="bugs">Known Bugs:</a></h4>
   <ul>
    <li>surfaces fail for some structures
    <li>POV-Ray incorrectly renders some of the characters in the fonts
    Chimera uses, especially the zero
    <li>Mac OS X problems:
    <ul>
     <li>in Mac OS X 10.6.4,
     the display of large surfaces (&gt; 15 million triangles)
     can cause the machine to freeze requiring a reboot.  This is a
     graphics driver bug affecting both Nvidia and ATI graphics cards.
     <li>in Mac OS X 10.5, opening and then closing IDLE adds 
     the IDLE menus to the Chimera menus, 
     and trying to use the extra menus generates errors
     <li>in sequential stereo, the drag selection outline is only visible
     for a fraction of a second when the mouse button is released
     <li>row-interleaved stereo only works with NVIDIA graphics
    </ul>
    <li>Mac OS X with X Windows problems:
    <ul>
     <li>Observed with 10.5 (Leopard):
     <ul>
      <li>sometimes display does not update
      (can be forced by slightly moving the view contents with the mouse)
      <li> sometimes saved images have a corrupt or "mis-tiled" appearance.
      Workarounds to try include:
      <ul>
       <li>showing the <b>Command Line</b> if it is not shown 
       (or if it is shown, hiding it) and then re-saving the image
       <li>saving the position with command <b>savepos</b>, jiggling
       the view with the mouse, restoring the position with <b>reset</b>,
       and then re-saving the image
       <li>saving a session, then using the native Mac version of Chimera
       (non-X11) to restore the session and re-save the image
      </ul>
     </ul>
     <li>Observed with 10.6 (Snow Leopard):
     <ul>
      <li><b>Side View</b> may come up blank, 
      but will be filled in when structures are moved
      <li>closing the Viewing dialog (<b>Side View</b>, <b>Lighting</b>, etc.)
      causes chimera to crash on next graphics redraw
     </ul>
    </ul>
   </ul>

   <hr>
   <h3><a name="1.6.1">Changes in release 1.6.1</a> relative to 
    <a href="#1.6.0">release (1.6)</a>:</h3>

   <h4>Major Bugs Fixed:</h4>
   <ul>
    <li>Fixed bug that affected recent versions of Linux,
    <i>e.g.</i>, Ubuntu 11.10 and Fedora 16,
    where the graphics would be corrupted or chimera would crash.
	<li>Fixed bug that prevented Planes (of Axes/Planes/Centroids tool) from working.
	<li>Removed inappropriate &quot;file types&quot; (<i>e.g.</i> PDB ID) from the
	file-type list in the main File&rarr;Open dialog.
	<li>Fixed bug that prevented very old session files from restoring on
	64-bit Windows.
   </ul>

   <hr>
   <h3><a name="1.6.2">Changes in release 1.6.2</a> relative to 
    <a href="#1.6.1">release (1.6.1)</a>:</h3>

   <h4>Major Bugs Fixed:</h4>
   <ul>
    <li>Fixed bug where selection of carbon-alpha traces caused session saving
	to fail.
    <li>Fixed bug restoring sessions where the session had been saved with
	a Modeller web-service computation ongoing.
    <li>Sali lab model-evaluation web service changed.  Changed code to account for that.
    <li>Allow chain ids to be swapped in <b>changechains</b> command.
    <li>Fixed two bugs that caused exporting to POV-ray to fail.
    <li>On Linux, don't crash on control-mousewheel events.
    <li>On SELinux, make Python's cryptographic hash functions work.
    <li>Fix saving large images with gradient backgrounds.
   </ul>
  </blockquote>
 </body>
</html>
