<!doctype html public "-//W3C//DTD HTML 4.0//EN">
<!-- vim: sw=1:
>>> UCSF Chimera Copyright
Copyright (c) 2013 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
<<< UCSF Chimera Copyright
-->
<html>
 <head>
  <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
  <link rel="icon" href="../chimera16.png" type="image/png">
  <title>UCSF Chimera Version 1.7 Build 38197 Release Notes</title>
 </head>

 <body>
  <p>
  <img src="../titleChimera.png" width="370" height="55" alt="logo">
  </p>

<font size="-1">
<a href="http://www.cgl.ucsf.edu/chimera/docs/relnotes.html">Release History</a>
</font>

  <h2>UCSF Chimera Version 1.7 Build 38197 Release Notes (30 Jan 2013)</h2>

  <h3>
   Changes since the <a href="1.6.html">previous production release (1.6)</a>:
  </h3>

  <ul>
   <li><a href="#general">General Changes</a>
   <li><a href="#new_tools">New Tools</a>
   <li><a href="#tool_changes">Tool Changes</a>
   <li><a href="#new_commands">New Commands</a>
   <li><a href="#command_changes">Command Changes</a>
   <li><a href="#bug_fixes">Notable Bug Fixes</a>
   <li><a href="#bugs">Known Bugs</a>
  </ul>

  <blockquote>
   <h4><a name="general">General Changes:</a></h4>
   <ul>
    <li>AmberTools updated to version 12 (new force field, <i>etc.</i>)
    <li>cleaner <b>CASTp</b> pocket display, disconnected fragments hidden
    automatically
    <li>&ldquo;show distances to nearby residues&rdquo; function added to 
    atom context menu (raised by doubleclick-select)
    <li>&ldquo;click to center&rdquo; mouse mode added 
    (default Ctrl-3, can reassign in <b>Mouse</b> preferences)
    <li>publication presets no longer change atom/bond display styles
    (except wire &rarr; stick)
    <li>presets can be user-defined (Chimera command or Python file
    locations specified in <b>Presets</b> preferences)
    <li><b>Actions... Color</b> menu works on axis, plane, centroid objects
    <li><b>Image Save</b> dialog reorganized, 
    integrated with output file specification
    <li><b>Fetch by ID</b> allows entering multiple IDs for same database
    <li>new input format: STL
    <li><b>Movie</b> preferences removed, settings put back into various
    recording dialogs
    <li>new movie formats (output): H.264, VP8/WebM, Theora
    <li>default lighting changed
    <li>interactive shadows faster, available on more systems, 
     quality adjustable in <b>Effects</b>
    <li>faster loading of large molecules
    <li>faster graphics for large molecules on many systems
    <li>default maximum graphics frame rate doubled, to 60 fps
    <li>smoother rotation with mouse drag
    <li>Mac retina display full-resolution used for graphics and text
    <li>new pseudobond style: <b>spring</b>
    <li>rounded ribbons look flatter (actually a bugfix; 
    incorrect highlighting had given the illusion of plumpness)
   </ul>

   <h4><a name="new_tools">New Tools:</a></h4>
   <ul>

    <li><b>APBS</b> (Surface/Binding Analysis) &mdash;
    interface to Poisson-Boltzmann electrostatics calculations 
    with APBS, using either a web service provided by the 
    National Biomedical Computation Resource (NBCR) or a locally installed copy

    <li><b>AutoDock Vina</b> (Surface/Binding Analysis) &mdash;
    interface to single-ligand docking with
    AutoDock Vina, using either a web service provided by the 
    National Biomedical Computation Resource (NBCR) or a locally installed copy
   
    <li><b>Model/Refine Loops</b> (Structure Editing) &mdash;
    interface to Modeller for building missing parts or refining existing parts 
    (the former <b>Model Loops</b> tool only performed the latter and
    did not facilitate combining the refined and unchanged parts)

    <li><b>Notepad</b> (Utilities) &mdash; 
    allows entering descriptive text that can be saved along with sessions

    <li><b>PDB2PQR</b> (Structure Editing) &mdash;
    interface to structure cleanup and charge/radius assignment with
    PDB2PQR, using either a web service provided by the 
    National Biomedical Computation Resource (NBCR) or a locally installed copy
   </ul>

   <h4><a name="tool_changes">Tool Changes:</a></h4>
   <ul>

   <li><b>2D Labels</b> (Utilities) &mdash;
     <ul>
     <li>labels can have backgrounds (colored rectangles underneath the text)
     <li>buttons added to hide/show all 2D labels or arrows
     </ul>

   <li><b>Add Charge</b> (Structure Editing) &mdash; 
     force field options changed, new default: Amber <b>ff12SB</b>

   <li><b>Animation</b> (Utilities) &mdash;
    <ul>
    <li>timeline for arranging scenes, playing/recording animation
    </ul>

    <li><b>Blast Protein</b> (Sequence) &mdash;
    results table can be exported as comma-separated values

    <li><b>Build Structure</b> (Structure Editing) &mdash;
    <ul>
     <li>can add atom at specified X,Y,Z coordinates
     <li>bond length adjustment slider and revert button
     <li>new section for adjusting bond (valence) angles
     <li><b>Adjust Torsions</b> moved to this tool
     (was previously part of <b>Structure Measurements</b>)
    </ul>

    <li><b>Effects</b> (Viewing Controls) &mdash;
      <ul>
      <li>transparent background setting moved to <b>Save Image</b> dialog
      <li>local viewer option removed
      <li>new settings related to shadows and transparency
      </ul>

    <li><b>Ensemble Cluster</b> (MD/Ensemble Analysis) &mdash; 
     clustering information can be saved to a text file

    <li><b>Hide Dust</b> (Volume Data) &mdash; 
    surface area, area rank, and volume rank added as metrics

    <li><b>Lighting</b> (Viewing Controls) &mdash; 
      <ul>
      <li>for better control, <b>contrast</b> split into two parameters:
      <b>contrast</b> and <b>key-fill ratio</b>
      <li><b>quality</b> setting removed 
        (former &ldquo;glossy&rdquo; is new default)
      </ul>

    <li><b>MD Movie</b> (MD/Ensemble Analysis) &mdash; 
     <ul>
     <li>can calculate average structures for specified frame ranges or clusters
     <li>can plot RMSD (relative to a single reference frame) vs. frame number
     <li>GROMACS .xtc (compressed) trajectory files supported, faster reading
     of .trr trajectory files courtesy of XTC Library
     <li>input as multiple PDB or XYZ files no longer requires 
     consecutive numbering in filenames (skipped numbers OK)
     <li>occupancy calculation can treat atoms as VDW spheres,
     incrementing all grid cells containing part of an atom
     (previously could only treat atoms as points, the current default)
     <li>clustering information can be saved to a text file
     </ul>

    <li><b>Minimize Structure</b> (Structure Editing) &mdash; 
     uses force field parameters concordant with charges for
     Amber <b>ff12SB</b>, <b>ff03.r1</b>, <b>ff99SB</b>, or <b>ff99bsc0</b>
     (previously always used <b>ff99</b> parameters)

    <li><b>Model Loops</b> removed, replaced with 
    <a href="#new_tools">new tool</a> <b>Model/Refine Loops</b> 

    <li><b>Multalign Viewer</b> (Sequence) &mdash; 
      <ul>
      <li>allows using a subset of sequences as guides for adding a new sequence
      <li>Region Browser has menu for viewing different sets of regions,
       those for the entire alignment or those for an individual sequence
       (now that each sequence can have many UniProt feature annotations)
      <li>Region Browser shows Start, End of individual-sequence regions
      </ul>

    <li><b>PipesAndPlanks</b> (Depiction) &mdash; 
      <ul>
      <li>helix and strand directionality can be shown with arrows
      <li>distinct coloring per secondary structure element can be achieved by
      setting color wells to No Color 
      (uses the ribbon color of the first residue per element)
      <li>significantly improved 
      &ldquo;joins&rdquo; between coil, pipe, and plank representations
      <li>coil subdivision can be adjusted
      <li><b>edge color</b> settings added to outline parts for which standard
      silhouette edges are not shown
      </ul>

    <li><b>ViewDock</b> (Surface/Binding Analysis) &mdash; 
      reads PDBQT format from AutoDock and AutoDock Vina, Mol2 format from GOLD

    <li><b>Volume Tracer</b> (Volume Data) &mdash; 
      <ul>
      <li>changes in mouse mode defaults:
      marker creation on planes and spots (local maxima) both turned on,
      automatic linking turned off
      <li>marker creation on spots applies to all volume displays (not just 
      the active data set) and the slice graph includes all displays
      under the mouseclick
      <li>default marker/link color changed to yellow
      </ul>

    <li><b>Volume Viewer</b> (Volume Data) &mdash; 
      <ul>
      <li>changes in <b>Planes</b> including mouse mode for dragging planes 
      <li><b>Orthogonal planes</b> feature for viewing planes perpendicular 
       to data axes, optionally as box faces
      </ul>

   </ul>

   <h4><a name="new_commands">New Commands:</a></h4>
   <ul>
    <li><b>play</b> &mdash; animate complex motions such as radial
    &ldquo;explosion&rdquo; of multiscale surfaces
    <li><b>transparency</b> &mdash; 
      <ul>
      <li>set transparency of atoms, ribbons, <i>etc.</i> 
       without otherwise changing their colors
      <li>can be applied gradually over a specified number of frames
      </ul>
   </ul>

   <h4><a name="command_changes">Command Changes:</a></h4>
   <ul>
    <li><b>2dlabels</b> &mdash;
      <ul>
      <li>labels can have backgrounds (colored rectangles underneath the text)
      <li>all 2D labels or arrows can be changed with a single command using *
      as name
      </ul>
   <li><b>addcharge</b> &mdash; 
     force field options changed, new default: Amber <b>ff12SB</b>
    <li><b>center</b> &mdash; works on surface pieces (previously only atoms)
    <li><b>color</b> &mdash; 
      <ul>
      <li>can color nonmolecular surface models
      <li>specifiers added for bond labels, bonds-only
      </ul>
    <li><b>coordset</b> &mdash; 
      <ul>
      <li>multiple trajectories can be specified in the same command with 
      a comma-separated list or range of model numbers
      <li>playback <b>loop</b> option
      </ul>
    <li><b>copy</b> &mdash; can specify image dimensions in units of length,
    image dpi 
    <li><b>fitmap</b> &mdash; 
      <ul>
      <li><b>eachModel</b> option to fit multiple models independently
      <li><b>listFits</b> to show results in <b>Fit List</b> dialog
      <li><b>radius</b> to restrict global search radius
      </ul>
    <li><b>lighting</b> &mdash; 
      <ul>
      <li>for better control, <b>contrast</b> split into two parameters:
    <b>contrast</b> and key-fill <b>ratio</b>
      <li>changes in defaults, <b>quality</b> setting removed
      (former &ldquo;glossy&rdquo; is new default)
      </ul>
    <li><b>mcopy</b> &mdash; residue pairing can be user-specified (previously
    pairing required identical residue names, numbers, and chain IDs) 
    <li><b>movie</b> &mdash; 
      <ul>
      <li>several usability improvements 
      (output format can be specified by filename suffix, <i>etc.</i>)
      <li><b>formats</b> option to list the available movie formats
      <li><b>ignore</b> option to start/stop ignoring movie commands,
      useful during script development
      </ul>
    <li><b>perframe</b> &mdash; 
      <ul>
      <li>other commands can be embedded directly, not only as an alias
      <li>variables other than frame count can be substituted
      <li>allows execution at multi-frame intervals
      </ul>
    <li><b>preset</b> &mdash; 
       presets (including custom) can be specified by name
    <li><b>rock, roll, turn</b> &mdash; default rotation per frame halved
    to compensate for doubled default frame rate
    (see <a href="#general">General Changes</a>)
    <li><b>roll</b>, <b>turn</b> &mdash; 
    <b>precessionTilt</b> to specify additionally rotating around a 
    moving axis that is carried along by the main rotation
    <li><b>runscript</b> &mdash; specifying the script as <b>browse</b>
    or <b>browser</b> calls up a dialog for locating/opening the file
    <li><b>set</b> &mdash; 
      <ul>
      <li>keywords can be truncated, underscores removed
      <li><b>fieldOfView</b> to adjust horizontal field of view angle
      <li><b>flatTransparency</b> 
       to make apparent transparency independent of the viewing angle
      <li><b>maxFrameRate</b> to adjust maximum graphics frame rate
      </ul>
    <li><b>shape</b> &mdash; 
      <ul>
      <li><b>boxpath</b> 
      - zigzag &ldquo;beams&rdquo; connecting a series of atoms
      </ul>
    <li><b>sop</b> &mdash; 
      <ul>
      <li><b>hidePieces</b> and <b>showPieces</b> to toggle display
       of specified surface pieces (independently selectable parts)
      <li><b>hideDust</b> metrics added: surface area, area rank, volume rank
      <li><b>transform</b> to scale, rotate, and shift surfaces
      </ul>
    <li><b>surface</b> &mdash; <b>visiblePatches</b> option to display only 
    the <i>N</i> largest-area contiguous patches 
    for cleaner display of pockets and other partial molecular surfaces
    <li><b>stereo</b> &mdash; tilt angle can be specified for <b>dome</b> mode
    <li><b>volume</b> &mdash; 
      <ul>
      <li><b>orthoplanes</b> to display 2-3 orthogonal planes 
       for the same dataset
      <li><b>boxFaces</b> to display orthogonal planes as box faces
      <li><b>expandSinglePlane</b> to expand a single plane along its axis
       to the full thickness of the data
      </ul>
    <li><b>vop</b> &mdash; 
      <ul>
      <li><b>falloff</b> to smooth boundaries of masked density
      <li><b>fourier</b> to generate map of phase values (&ndash;&pi; to &pi;)
      <li><b>localCorrelation</b> 
      to calculate map-map correlation over a sliding box
      <li>new options to <b>cover</b> 
      for using map symmetries and specifying unit cell dimensions
      </ul>
   </ul>

   <h4><a name="bug_fixes">Notable Bug Fixes:</a></h4>
   <ul>
    <li>Chimera-Modeller interface fixed to handle templates containing
    modified mainchain residues
    <li>independent rotation applies to hidden models
    <li>corrected single-layer transparency of VRML models
    (nucleotides, axis/plane/centroid representations, BILD objects)
    <li>shadows work with a single light and are ignored with ambient lighting
   </ul>

   <h4><a name="bugs">Known Bugs:</a></h4>
   <ul>
    <li>surfaces fail for some structures
    <li>Nvidia Quadro graphics on Windows 7 cause ten-second freezes
      when picking atoms (Ctrl-mouseclick) of large models (20000 atoms).
      This is caused by the graphics
      driver "adaptive" power management setting.  Using Nvidia Control
      Panel, Manage 3-D Settings, Global Settings tab to change Power Management
      to "Prefer maximum performance" solves this problem.
      Bug report <a href="http://plato.cgl.ucsf.edu/trac/chimera/ticket/11315">#11315</a>.
    <li>POV-Ray incorrectly renders some of the characters in the fonts
    Chimera uses, especially the zero
    <li>Mac OS X problems:
    <ul>
     <li>in Mac OS X 10.6.4,
     the display of large surfaces (&gt; 15 million triangles)
     can cause the machine to freeze requiring a reboot.  This is a
     graphics driver bug affecting both Nvidia and ATI graphics cards.
     <li>in sequential stereo, the drag selection outline is only visible
     for a fraction of a second when the mouse button is released
     <li>row-interleaved stereo only works with NVIDIA graphics
    </ul>
   </ul>

  </blockquote>
 </body>
</html>
