<!--
--- UCSF Chimera Copyright ---
Copyright (c) 2008-2014 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
--- UCSF Chimera Copyright ---
-->

<html><head>
<title>Superpositions and Alignments Tutorial</title>

<style type="text/css">
A.nounder:link {text-decoration: none}
A.nounder:visited {text-decoration: none}
A.nounder:hover {text-decoration: underline; color: red;}
</style>
</head><body>

<font size="-1">
<a href="../frametut.html" target="_top">Chimera Tutorials Index</a>
</font>

<a name="top"></a>
<h2>Superpositions and Alignments Tutorial</h2>
<p>
In this tutorial,
<a href="../../ContributedSoftware/matchmaker/matchmaker.html"><b>MatchMaker</b></a>
is used to align protein structures (create a superposition),
<a href="../../ContributedSoftware/matchalign/matchalign.html"><b>Match 
-> Align</b></a>
is used to generate a multiple sequence alignment from the structural
superposition, and 
<a href="../../ContributedSoftware/morph/morph.html"><b>Morph 
Conformations</b></a> is used to morph between related structures.
</p><p>
Sequence alignments are displayed in
<a href="../../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a>, which is covered in more detail
in the <a href="super.html">Sequences and Structures tutorial</a>, and
the morphing trajectory is displayed in
<a href="../../ContributedSoftware/movie/framemovie.html"
target="_top"><b>MD Movie</b></a>, which is covered in more detail in the
<a href="ensembles2.html">Trajectory and Ensemble Analysis tutorial</a>.
</p><p>
Internet connectivity is required to fetch the structures used in this
tutorial: <b>1tad</b>, <b>121p</b>, <b>1r2q</b>, <b>1j2j</b>, <b>1puj</b>,
<b>1tnd</b>, <b>1tag</b>
<ul>
<li><a href="#setup">Background and setup</a>
<li><a href="#diffprot">Different but related proteins</a>
  <ul>
  <li><a href="#matchmaker">Superposition</a>
  <li><a href="#matchalign">Structure-based sequence alignment</a>
  </ul>
<li><a href="#sameprot">Different conformations of the same protein</a>
  <ul>
  <li><a href="#morph">Morphing</a>
  </ul>
</ul>
</p>

<h3><a href="#top" class="nounder"><b>&larr;</b></a>
<a name="setup">Background and Setup</a></h3>
<p>
Protein structures are classified within databases such as
<a href="https://scop.berkeley.edu/" target="_blank">SCOP</a>,
<a href="https://www.cathdb.info/" target="_blank">CATH</a>, and
<a href="https://mizuguchilab.org/homstrad/" target="_blank">HOMSTRAD</a>.
Classifications range from groups of highly similar and closely 
related proteins to larger, more diverse sets. For analysis and comparison, 
it is often useful to superimpose related structures.
Although it is not always clear whether proteins with the same fold
are evolutionarily related (homologous), they should still be superimposable.
In general, more closely related proteins are easier to superimpose.
</p><p>
<b><i>G proteins</i></b> (guanine nucleotide-binding proteins)
are used as examples.  G proteins are important in signal transduction.  
They act as molecular switches, changing conformation and interaction
partners depending on whether GTP or GDP is bound.
Many diverse structures are known.  The two main subsets are the
small monomeric G proteins, such as Ras, and the larger heterotrimeric 
G proteins, which act immediately downstream of G-protein-coupled receptors.
The &alpha; subunits of heterotrimeric G proteins 
are homologous to the small G proteins.
</p><p>
Start Chimera by clicking or doubleclicking the <b>Chimera</b> icon
<img src="Chimera-icon.png" width="20px"> (depending on its location).
Typically, this icon will be present on the desktop.
The Chimera executable can also be run from its
installation location (<a href="../startup.html">details...</a>).
</p><p>
A splash screen will appear, to be replaced in a few seconds
by the main Chimera <a href="../chimerawindow.html">graphics window</a>
or <a href="../rapid.html"><b>Rapid Access</b></a> interface (it does not
matter which, the following instructions will work with either).
If you like, resize the Chimera window by dragging its lower right corner.
</p><p>
Show the <a href="../chimerawindow.html#emulator"><b>Command Line</b></a>
(<b>Tools... General Controls... Command Line</b>).
Choose <b>Favorites... Add to Favorites/Toolbar</b> to place some
icons on the toolbar. This opens the <b>Preferences</b>, set to
<b>Category</b>: <a href="../preferences.html#Tools"><b>Tools</b></a>.
In the <b>On Toolbar</b> column, check the boxes for:
<a name="toolicons"></a>
<table align="center" style="margin:8px 8px 8px 8px">
<tr><td valign="center">
<ul>
<li> <a href="../modelpanel.html"><b>Model Panel</b></a> 
(under General Controls)
<li> <a href="../sideview.html#sideview"><b>Side View</b></a> 
(Viewing Controls)
<li> <a href="../../ContributedSoftware/matchmaker/matchmaker.html"><b>MatchMaker</b></a> (Structure Comparison)
<li> <a href="../../ContributedSoftware/matchalign/matchalign.html"><b>Match -> Align</b></a> (Structure Comparison)
</ul>
</td>
<td valign="top"><img src="../modpanicon.png"></td>
<td valign="top"><img src="../svicon.png"></td>
<td valign="top">
<img src="../../ContributedSoftware/matchmaker/mmicon.png"></td>
<td valign="top">
<img src="../../ContributedSoftware/matchalign/matchalign.png"></td></tr>
</table>
You can also specify tools such as the
<a href="../chimerawindow.html#emulator"><b>Command Line</b></a>
to <b>Auto Start</b> (start when Chimera is started).  
If you want these settings to apply to subsequent uses of Chimera, 
click <b>Save</b> before closing the preferences.
</p><p>
Fetch a structure from the <a href="https://www.rcsb.org" 
target="_blank">Protein Data Bank</a>:
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/open.html">open</a>&nbsp;1tad</b>
</blockquote>
The structure contains three copies of the &alpha; subunit of transducin, 
a heterotrimeric G protein.  Delete solvent and two of the copies, 
chains B and C:
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/delete.html">del</a>&nbsp;solvent</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/delete.html">del</a>&nbsp;:.b-c</b>
</blockquote>
Move and scale the structures <a href="../mouse.html">using the mouse</a> and
<a href="../sideview.html#sideview"><b>Side View</b></a> 
as desired throughout the tutorial.
</p>

<h3><a href="#top" class="nounder"><b>&larr;</b></a>
<a name="diffprot">Different but Related Proteins</a></h3>
<p>
We will <a href="#matchmaker">superimpose structures</a> 
of a sample of G proteins,
then <a href="#matchalign">create a sequence alignment</a>
from the superposition.
</p><p>
The &alpha; subunit of the heterotrimeric G protein
transducin was already opened in the <a href="#setup">setup</a>.
Fetch structures for the monomeric G proteins H-Ras, 
Rab5a, and ADP-ribosylation factor 1, respectively:
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/open.html">open</a>&nbsp;121p</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/open.html">open</a>&nbsp;1r2q</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/open.html">open</a>&nbsp;1j2j</b>
</blockquote>
Use the ribbons <a href="../menu.html#menupresets">preset</a>
(which may or may not change the appearance, 
depending on your preference settings):
<blockquote>
<i>Menu</i>:&nbsp;<b>Presets... Interactive 1 (ribbons)</b>
</blockquote>
This preset displays ribbons plus ions, ligands, and nearby sidechains.
</p>

<table align="right" style="margin:8px 8px 8px 8px">
<tr><th><a name="fig1">superimposed G proteins</a></th></tr>
<tr><td><img src="5gees.png"></td></tr>
</table>

<h3><a href="#top" class="nounder"><b>&larr;</b></a>
<a name="matchmaker">Superposition</a></h3>
<p>
The structures need to be superimposed so that they can be compared.
Start <a href="../../ContributedSoftware/matchmaker/matchmaker.html"><b>MatchMaker</b></a>
by clicking its <a href="#toolicons">icon</a>:
<img src="../../ContributedSoftware/matchmaker/mmicon.png" width="25">
</p><p>
<a href="../../ContributedSoftware/matchmaker/matchmaker.html"><b>MatchMaker</b></a>
superimposes structures pairwise by first aligning their sequences
and then fitting the &alpha;-carbons of residues in the same columns
of the sequence alignment.  Usually the fit is iterated so that 
residue pairs aligned in sequence but far apart in space 
are not used in the final 3D match.
</p><p>
Several parameters control the sequence alignment step:
<ul>
<li> algorithm - Needleman-Wunsch (global; default) or Smith-Waterman (local)
<li> scoring function -
  <ul>
  <li> residue similarity matrix (default BLOSUM-62)
  <li> whether secondary structure information should be used
  (default yes) 
  <li> weighting of the secondary structure and residue similarity terms
  (default 30% and 70%, respectively)
  <li> gap penalties
  </ul>
</ul>
Click <b>Reset to defaults</b> (near the bottom of the dialog)
to ensure that the default settings will be used.
All of the structures should already be chosen as the
<b>Structure(s) to match</b>; keep that the same, but choose <b>1tad</b>
as the <b>Reference</b> and click <b>Apply</b> to match all the others to it.
</p><p>
The number of &alpha;-carbon pairs and RMSD in the final iteration of each
pairwise fit are reported in the <b>Reply Log</b> 
(in the menu under <b>Favorites</b>).
However, simple visual inspection of the overall structures
is often the most useful indicator of success.
</p><p>
Another visual indicator is how well similar ligands superimpose. 
Show only residues classified as
<a href="../midas/surface.html#surfcats">ligand</a>, and label them:
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/show.html">show</a>&nbsp;ligand</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/rlabel.html">rlab</a>&nbsp;ligand</b>
</blockquote>
Each of these structures includes GTP or an analog of GTP in the binding site.
However, some other ligands were simply present in the crystallization 
solution and are not biologically relevant.  
GOL is glycerol and can be removed:
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/delete.html">del</a>&nbsp;:gol</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/rlabel.html">~rlab</a></b>
</blockquote>
Try using different reference structures in <b>MatchMaker</b>
(click a line in the <b>Reference structure</b> list, click <b>Apply</b>).
With the default alignment parameters, 
the superposition is similar and basically correct 
no matter which structure is used as the reference.
Detailed examination of the match statistics and guanine nucleotide positions
suggests results may be slightly better with <b>1r2q</b> as the reference.
</p><p>
Next, try a structure that is harder to superimpose, and display its ligand
in the sphere representation:
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/open.html">open</a>&nbsp;1puj</b>
<br>
<i>Menu</i>:&nbsp;<b>Presets... Interactive 1 (ribbons)</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/show.html">show</a>&nbsp;ligand</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/represent.html">repr</a>&nbsp;sphere&nbsp;ligand&nbsp;&amp;&nbsp;#4</b>
</blockquote>
Besides lacking sequence similarity,
this protein is <b><i>circularly permuted</i></b> compared to the others:
its N-terminal part structurally matches the C-terminal part of other G
proteins and vice versa.
</p><p>
In the <b>MatchMaker</b> dialog,
change the <b>Structure to match</b> to only <b>1puj</b> and try the others
in turn as the reference.  
Again, ligand positions can be used to help gauge the match.  
</p><p>
Trials with the default alignment parameters are not
successful.  When proteins are very distantly related, it may be useful
to switch to a lower-number BLOSUM matrix and/or increase 
the proportion of secondary structure scoring.
Usually a range of parameters will give similar results.  
For example, with <b>121p</b> as the reference structure, <b>1puj</b>
can be superimposed as shown in the <a href="#fig1">figure</a> 
with any of BLOSUM 30-75 if secondary structure weighting is raised to 90%
and the Smith-Waterman algorithm is used (leaving other settings as defaults).
Keep in mind that when proteins are very distantly related, 
their backbones may diverge even in the best possible superposition.
</p>

<h3><a href="#top" class="nounder"><b>&larr;</b></a>
<a name="matchalign">Structure-Based Sequence Alignment</a></h3>
<p>
When all five proteins are superimposed to your satisfaction,
<b>Cancel</b> the <b>MatchMaker</b> dialog.
We will generate a structure-based alignment of the five sequences using
<a href="../../ContributedSoftware/matchalign/matchalign.html"><b>Match 
-> Align</b></a>; start that tool by clicking its <a href="#toolicons">icon</a>:
<img src="../../ContributedSoftware/matchalign/matchalign.png" width="25">
</p><p>
<a href="../../ContributedSoftware/matchalign/matchalign.html"><b>Match 
-> Align</b></a> uses only the distances between &alpha;-carbons
to create an alignment.  Residue types and how the structures were superimposed
are not important.
All of the A chains should already be chosen in the dialog; the B chain of
<b>1j2j</b> is an unrelated peptide and should not be chosen.  
Use a cutoff of <b>5.0</b> &Aring;,
specify <b>Residue aligned in column if within cutoff of
[at least one other]</b>, and turn on <b>Allow for circular permutation</b>.
Click <b>OK</b> to start the calculation.
</p>
<table border="1" align="right" style="margin:8px 8px 8px 8px" 
cellspacing="0" cellpadding="0">
<tr><td><img src="5gees-seq.png" alt="structure-based sequence alignment"
width="350"></td></tr>
</table>
<p>
It may take a minute or two to create the alignment; 
progress is reported in the 
<a href="../chimerawindow.html#statusline">status line</a>.  
When the calculation is finished, the new alignment will be displayed
in <a href="../../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a> and can be
<a href="../../ContributedSoftware/multalignviewer/multalignviewer.html#saving">saved to a file</a>
from that tool.
</p><p>
The output multiple sequence alignment 
(example: <a href="5gees.afa" type="text/plain">5gees.afa</a>)
shows that <b>1puj</b> was correctly recognized as a circular permutation
relative to the others.  <b>Match -> Align</b> doubled its sequence to
allow C-terminal residues (in the first copy of the sequence) to appear
before more N-terminal residues (in the second copy) within the alignment.  
The columns with residues from all five structures
are highlighted as a region in light orange with dark orange outline.
Clicking the region will select the corresponding parts of the structures, 
in effect their common cores. 
The <a href="../../ContributedSoftware/multalignviewer/multalignviewer.html#headers">alignment header</a> 
named &ldquo;<b>RMSD: ca</b>&rdquo; shows the spatial variation per column
(&alpha;-carbon root-mean-square deviation) as a histogram.
</p><p>
Keep the sequence alignment, but close most of the structures:
<ol>
<li>start the <a href="../modelpanel.html"><b>Model Panel</b></a>
by clicking its <a href="#toolicons">icon</a>:
<img src="../modpanicon.png" width="25">
<li>in the <b>Model Panel</b>, choose all of the models except 
<b>1tad</b> on the left side and click the <b>close</b> button on the right 
(not at the bottom of the dialog!)
<li><b>Close</b> the <b>Model Panel</b>
</ol>
If <a href="../../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a> (the sequence alignment window)
is hidden behind other windows, it can be resurrected by choosing
<b>MAV - <i>alignment-name</i>... Raise</b> from
near the bottom of the <b>Tools</b> menu.  In <b>Multalign Viewer</b>:
<ul>
<li>choose <b>Preferences... Appearance</b> 
and adjust settings for <b>Multiple alignments</b> as desired
<li>use <b>Info... Percent Identity</b> to compare <b>all sequences</b>
with <b>all sequences</b>, confirming that the pairwise identities are
&lt;30% for these examples
<li>use <b>Edit... Delete Sequences/Gaps</b> to delete the sequence named
<b>2 x 1puj, chain A</b> and any resulting all-gap columns
</ul>
Now the alignment clearly shows the large insertion in &alpha;-transducin 
(<b>1tad</b>) relative to the small monomeric G proteins.  Select and display
residues that are completely conserved in the sequence alignment:
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/select.html#newer">sel</a>&nbsp;:/mavPercentConserved=100</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/display.html">disp</a>&nbsp;sel</b>
</blockquote>
Some of the conserved residues are Gly (no sidechain).
Clear the selection by Ctrl-clicking in an empty area of the graphics window.
</p>

<h3><a href="#top" class="nounder"><b>&larr;</b></a>
<a name="sameprot">Different Conformations of the Same Protein</a></h3>
<p>
(To jump to this section right after performing the <a href="#setup">setup</a>,
open the sequence alignment file 
<a href="4gees.afa" type="text/plain">4gees.afa</a>
included with this tutorial.)
</p>
<a name="fig2"></a>
<table align="right" style="margin:8px 8px 8px 8px">
<tr><th>GTP-binding switch</th></tr>
<tr><td align="center">
(<font color="#3333CC"><b>1tagA</b></font>, 
<font color="#CC0066"><b>1tndA</b></font>, 
<font color="#666699">morph intermediate</font>)
</td></tr><tr><td>
<img src="../../ContributedSoftware/morph/transducin.png" 
alt="transducin morph">
</td></tr>
</table>
<p>
Now we will compare <b>1tad</b> with different structures of the same protein, 
transducin-&alpha;:  
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/open.html">open</a>&nbsp;1tnd</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/open.html">open</a>&nbsp;1tag</b>
</blockquote>
Delete solvent and chains B-C (extra copies in <b>1tag</b>):
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/delete.html">del</a>&nbsp;solvent</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/delete.html">del</a>&nbsp;:.b-c</b>
</blockquote>
If <a href="../../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a> (the sequence alignment window)
is hidden, bring it to the front by choosing <b>MAV - <i>alignment-name</i>... 
Raise</b> from near the bottom of the <b>Tools</b> menu.  
</p><p>
<a href="../../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a> displays lines of information 
called
<a href="../../ContributedSoftware/multalignviewer/multalignviewer.html#headers">headers</a> 
above the sequences in the alignment. Use the <b>Headers</b> menu to hide
<b>Consensus</b> and <b>Conservation</b> and to show <b>RMSD: ca</b>,
if not already shown.
The sequence name <b>1tad, chain A</b> has a dashed green line around it,
indicating that the sequence is associated with multiple structures.
The RMSD header shows the spatial variability of residues
associated with each column (&alpha;-carbon root-mean-square deviation); 
currently, it contains high values everywhere because the structures 
are not all superimposed.  
</p><p>
To superimpose the structures using the sequence alignment, choose 
<b>Structure... Match</b> from the <b>Multalign Viewer</b> menu.  One structure
(it does not matter which) should be designated as the reference, and all three
can be designated as the structures to match.  Check the option to <b>Iterate
by pruning...</b> using a <b>2.0</b>-&Aring; cutoff and click <b>OK</b>.
The RMSD header is automatically recomputed, showing much lower values.
</p><p>
Superposition of proteins with the same or nearly the same sequence is
generally trivial.  We used
<a href="../../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a> since we already had
a sequence alignment, but
<a href="../../ContributedSoftware/matchmaker/matchmaker.html"><b>MatchMaker</b></a>
(or its <a href="../midas/mmaker.html">command equivalent</a>) or the 
command <a href="../midas/match.html"><b>match</b></a> could have been 
used instead.  These other methods are used and discussed in the
<a href="squalene.html#comparison">Structure Analysis and Comparison 
tutorial</a>.
</p><p>
Use the ribbons preset (which may or may not change the
appearance, depending on your preference settings) and focus on the
ligand residues:
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/preset.html">preset</a>&nbsp;apply&nbsp;int&nbsp;1</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/focus.html">focus</a>&nbsp;ligand</b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/rlabel.html">rlab</a>&nbsp;ligand</b>
</blockquote>
Open the <a href="../modelpanel.html"><b>Model Panel</b></a> and use
the <b>S</b>(hown) checkboxes to view the structures individually.
</p><p>
The <b>1tad</b> structure (tan) represents the activated form
of a G protein; even though it includes GDP, 
the GDP and ALF (AlF<sub>4</sub><sup>&ndash;</sup>)
residues together mimic the transition state of GTP hydrolysis. <b>1tnd</b>
(light blue) contains the GTP analog GSP and also represents the activated form.
The third structure, <b>1tag</b> (purplish pink), includes GDP 
and represents the nonactivated form.
</p><p>
Use the <a href="../modelpanel.html"><b>Model Panel</b></a> checkboxes
to show all three structures together. 
Remove the labels and focus on the overall
structures: 
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/rlabel.html">~rlab</a></b>
<br>
<i>Command</i>:&nbsp;<b><a href="../midas/focus.html">focus</a></b>
</blockquote>
Although the structures are mostly similar,
the nonactive conformation (pink) differs from the activated ones
(tan and light blue) in specific areas, termed <b><i>switch regions</i></b>.
</p><p>
In the sequence alignment window,
the three most prominent &ldquo;humps&rdquo; in the RMSD header 
correspond to the known
G protein switch regions at approximately residues 173-183, 195-215, and
227-238 of transducin-&alpha;.  The third switch region is unique to
heterotrimeric G proteins; it is an insertion relative to the monomeric
G proteins.
Placing the cursor over a position in the <b>1tad</b> sequence lists the
associated structure residues near the bottom of the sequence window, and
drawing a box around residues in the sequence alignment (click to start,
drag to expand) <a href="../selection.html">selects</a> the associated parts 
of the structures.
</p><p>
Close <b>1tad</b>:
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/open.html">close</a>&nbsp;0</b>
</blockquote>
The RMSD histogram looks much the same; now it simply
shows the CA-CA distances between the two remaining structures,
<b>1tnd</b> representing the
activated form and <b>1tag</b> representing the nonactivated form.  
</p>

<h3><a href="#top" class="nounder"><b>&larr;</b></a>
<a name="morph">Morphing</a></h3>
<p>
Finally, morph between the two structures.
Morphing involves calculating a series of intermediate structures.
In Chimera, the series of structures is treated as a trajectory 
that can be replayed, saved to a coordinate file, or saved as a movie using
<a href="../../ContributedSoftware/movie/framemovie.html"
target="_top"><b>MD Movie</b></a>.
</p><p>
Start the <a href="../../ContributedSoftware/morph/morph.html">morphing
tool</a>:
<blockquote>
<i>Command</i>:&nbsp;<b><a href="../midas/start.html">start</a>&nbsp;Morph&nbsp;Conformations</b>
</blockquote>
Click <b>Add...</b> and in the resulting list of models, doubleclick
to choose #2, #1, and #2 again, corresponding to a morph trajectory from the
nonactivated structure to the activated and back.  <b>Close</b> the model list.
In the main <b>Morph Conformations</b> dialog, set the <b>Action on Create</b>
to <b>hide Conformations</b>, and then click <b>Create</b>.
</p><p>
The progress of the calculation is reported in the
<a href="../chimerawindow.html#statusline">status line</a>.
When all the intermediate structures have been calculated, the input
structures are hidden, the trajectory is opened as model #0, and the
<a href="../../ContributedSoftware/movie/framemovie.html"
target="_top"><b>MD Movie</b></a> tool appears.  
</p><p>
The trajectory can be played continuously or one step at a time 
using the buttons on the tool.  If the player dialog becomes obscured by 
other windows, it can be resurrected by choosing
<b>MD Movie - <i>trajectory-name</i>... Raise</b> from
near the bottom of the <b>Tools</b> menu.
If you want to see the original structures again, use the <b>S</b>(hown) 
checkboxes in the <a href="../modelpanel.html"><b>Model Panel</b></a>.
</p><p>
When you have finished viewing the morph trajectory, choose <b>File...
Quit</b> from the menu to exit from Chimera.
</p>
<hr>
<address>meng-at-cgl.ucsf.edu / May 2014</address>
</body></html>
