<!--
--- UCSF Chimera Copyright ---
Copyright (c) 2005-2014 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
--- UCSF Chimera Copyright ---
-->

<html><head>
<title>Attributes Tutorial</title>
</head><body>
<font size="-1">
<a href="../frametut.html" target="_top">Chimera Tutorials Index</a>
</font>

<h2>Attributes Tutorial</h2>
<p>
This tutorial demonstrates many uses of
<a href="../../ContributedSoftware/defineattrib/defineattrib.html#attribdef">
attributes</a>, or properties of atoms, residues, and molecule models.
Attributes can be numerical (such as atomic number),
boolean (<i>e.g.</i>, whether a residue is in a helix), 
or string-valued (such as <a href="../idatm.html">atom type</a>).
Attribute values can be rendered visually
and used in <a href="../../UsersGuide/selection.html">selection</a>
and command-line
<a href="../../UsersGuide/midas/frameatom_spec.html" target="_top">atom 
specification</a>.  See also the
examples on the Chimera website:
<a href="https://www.rbvi.ucsf.edu/chimera/data/tutorials/systems/outline.html"
target="_blank">Mapping Sequence Conservation</a>,
<a href="https://www.rbvi.ucsf.edu/chimera/data/nmrcolor-nov2017/nmrcolor.html"
target="_blank">Coloring by NMR Chemical Shift Perturbation</a>
</p><p>
<a href="#part1">Part 1</a> uses a leucine zipper structure, and
<a href="#part2">Part 2</a> uses the structure of a GTP-binding protein.
</p>
<hr>
<a name="part1">
<h3>Attributes, Part 1 - Leucine Zipper</h3></a>

<table align="right" style="margin:8px 8px 8px 8px">
<tr><th align="center">
<a name="fig1">tyrosine colored by charge</a></th></tr>
<tr><td><img src="att1.png"></td></tr>
</table>

<p>
Start Chimera by clicking or doubleclicking the <b>Chimera</b> icon
<img src="Chimera-icon.png" width="20px"> (depending on its location).
Typically, this icon will be present on the desktop.
The Chimera executable can also be run from its
installation location (<a href="../startup.html">details...</a>).
</p><p>
A splash screen will appear, to be replaced in a few seconds
by the main Chimera <a href="../chimerawindow.html">graphics window</a>
or <a href="../rapid.html"><b>Rapid Access</b></a> interface (it does not
matter which, the following instructions will work with either).
If you like, resize the Chimera window by dragging its lower right corner.
Show the <a href="../chimerawindow.html#emulator"><b>Command Line</b></a>
by choosing it from the <b>Favorites</b> menu, and optionally, the
<a href="../sideview.html#sideview"><b>Side View</b></a>  
for clipping and scaling.
</p><p>
If you have internet connectivity, structures can be obtained directly from the
<a href="https://www.rcsb.org" target="_blank">Protein Data Bank</a>.
Choose <b>File... <a href="../fetch.html">Fetch by ID</a></b>
from the Chimera menu and use the resulting dialog to
fetch <b>1zik</b> from the <b>PDB</b>.
If you do not have internet connectivity,
<a href="download.html">download</a> the file <a href="1zik.pdb">1zik.pdb</a>
included with this tutorial and use <b>File... Open</b> to open it.
</p><p>
The structure is a leucine zipper formed by two peptides.
Apply the &ldquo;all atoms&rdquo; <a href="../menu.html#menupresets">preset</a>,
change to stick representation, and hide water:
<blockquote>
<i>Command</i>: <b><a href="../midas/preset.html">preset</a> apply interactive 2</b>
<br><i>Command</i>: <b><a href="../midas/represent.html">repr stick</a></b>
<br><i>Command</i>: <b><a href="../midas/display.html">~disp</a> solvent</b>
</blockquote>
<a href="../mouse.html">Move and scale</a> the structure
as desired throughout the tutorial.  
<p><p>
Open the <a href="../../ContributedSoftware/render/render.html#render">
<b>Render by Attribute</b></a> tool (<b>Tools... Structure Analysis... 
Render by Attribute</b>).
Make sure it is set to show the attributes of <b>atoms</b>.  Look in the
<b>Attribute</b> list to see the attributes available: <b>bfactor</b>
and <b>occupancy</b>, which were read from the input PDB file.
Choose <b>bfactor</b>; a histogram of the values will
appear, with colored vertical bars (or 
<a href="../../ContributedSoftware/render/render.html#thresh">thresholds</a>)
that control how the values will be mapped to color.  
Click <b>Apply</b> to color the atoms by B-factor.  As expected, the
atoms with higher B-factors are on the outside of the structure.
</p><p>
The default coloring scheme is blue to white to red, 
but the color mapping is easy to change.
Thresholds can be dragged along the histogram with the mouse,
and added or deleted with Ctrl-click. 
The <b>Value</b> and <b>Color</b> are shown for the
most recently clicked or moved threshold.
Colors can be changed individually.
For example, drag the middle threshold to a lower value,
click the square <a href="../coloring.html#colorwell">color well</a>,
and use the resulting <a href="../colortool.html"><b>Color Editor</b></a>
to change the color to yellow. <b>Close</b> the editor, and 
in the <b>Render by Attribute</b> dialog, click <b>Apply</b>.
Drag the red threshold to the left of the yellow one and then
<b>Apply</b> again.
</p><p>
The colors can also be changed collectively, by choosing a <b>Palette</b>
in the <b>Render by Attribute</b> dialog.  Choose the <b>Palette</b>
named <b>Cyan-Maroon</b> (cyan to white to maroon), then click
<b>Reverse threshold colors</b> for coloring from maroon to white to
cyan. Move the thresholds if you wish and <b>Apply</b> again.
</p><p>
Note that the histogram includes the B-factor values of the waters
even though they are not displayed.
Display waters with B-factors less than 75:
<blockquote>
<i>Command</i>: <b><a href="../midas/display.html">disp</a> solvent & @/bfactor&lt;75</b>
</blockquote>
Show ribbons:
<blockquote>
<i>Command</i>: <b><a href="../midas/ribbon.html">ribbon</a></b>
</blockquote>
Next, change to the attributes of <b>residues</b>
in the <b>Render by Attribute</b> dialog.  Available residue
attributes include <b>kdHydrophobicity</b> and average <b>bfactor</b>
and <b>occupancy</b>.  The latter two are residue averages over the
atomic values.  Choose <b>kdHydrophobicity</b>, the
<a href="../midas/hydrophob.html">Kyte-Doolittle hydrophobicity scale</a> 
for amino acids.  The values will be displayed in the histogram:
negative for polar residues, positive for hydrophobic residues. 
</p><p>
Hydrophobicity could be shown with color, but for variety,
we will show it with &ldquo;worms,&rdquo; modified ribbons that vary in radius.
In the <b>Render by Attribute</b> dialog, change from <b>Colors</b>
to <b>Worms</b>.
The values are still shown in a histogram, but now the thresholds have
a <b>Worm radius</b> instead of a color.  The thresholds can be added,
deleted, and moved like before, and the <b>Worm radius</b> can be changed.
By default, the more hydrophobic residues will be shown with a fatter worm
(a larger worm radius).  Change the mapping as desired and click <b>Apply</b>.
</p><p>
The worms show that the most hydrophobic residues 
tend to face the interior of the structure. To return to a normal ribbon
instead of a worm, change the <b>Worm style</b> to <b>non-worm</b>
and click <b>OK</b> (which is equivalent to <b>Apply</b> plus <b>Close</b>).
</p><p>
Additional <a href="../midas/hydrophob.html">hydrophobicity scales</a>
are available as
<a href="../../ContributedSoftware/defineattrib/defineattrib.html#attrfile">attribute
assignment files</a>.
</p><p>
Some Chimera tools create new attributes.  For example, the
<a href="../../ContributedSoftware/addcharge/addcharge.html"><b>Add 
Charge</b></a> tool (also implemented as the command
<b><a href="../midas/addcharge.html">addcharge</a></b>)
assigns partial charges as an atom attribute named <b>charge</b>.
Charge values from
<a href="https://ambermd.org/" target="_blank"><b>Amber</b></a> 
are used for standard amino acids and nucleotides.
</p><p>
Use the &ldquo;all atoms&rdquo; preset again:
<blockquote>
<i>Command</i>: <b><a href="../midas/preset.html">preset</a> apply int 2</b>
</blockquote>
Next, add hydrogens and assign atomic partial charges: 
<blockquote>
<i>Command</i>: <b><a href="../midas/addh.html">addh</a></b>
<br><i>Command</i>: <b><a href="../midas/addcharge.html">addcharge</a></b>
</blockquote>
Above, attribute values were shown with color and worms; another approach
is to display the values as atom labels.  Since it would be too confusing to
show all of these labels at once, only show them for the atoms in a
single residue: 
</p>
<blockquote>
<i>Command</i>: <b><a href="../midas/alias.html">alias</a> myres :tyr.a</b>
<br><i>Command</i>: <b><a href="../midas/show.html">show</a> myres</b>
<br><i>Command</i>: <b><a href="../midas/focus.html">focus</a></b>
<br><i>Command</i>: <b><a href="../midas/labelopt.html">labelopt info</a> charge</b>
<br><i>Command</i>: <b><a href="../midas/label.html">label</a> myres</b>
</blockquote>
The number of decimal places and whether plus signs should be used for
positive values can be specified, for example:
<blockquote>
<i>Command</i>: <b><a href="../midas/labelopt.html">labelopt info</a> %(charge)+.2f</b>
<br><i>Command</i>: <b><a href="../midas/label.html">lab</a> myres</b>
</blockquote>
<p>
New attributes can be used in the command line just like
built-in attributes.  For example, show the atoms with <b>charge</b>
less than &ndash;0.4 as balls:
<blockquote>
<i>Command</i>: <b><a href="../midas/represent.html">repr bs</a> &nbsp;@/charge&lt;&ndash;0.4</b>
</blockquote>
Finally, remove the labels, change to ball-and-stick, and
color atoms by their <b>charge</b> values:
<blockquote>
<i>Command</i>: <b><a href="../midas/label.html">~lab</a></b>
<br><i>Command</i>: <b><a href="../midas/represent.html">repr bs</a></b>
<br><i>Command</i>: <b><a href="../midas/rangecolor.html">rangecol</a> &nbsp;charge &nbsp;&ndash;0.8 red 0 white 0.8 blue</b>
</blockquote>
The result should look something like the <a href="#fig1">figure</a>.
<a href="../midas/rangecolor.html"><b>Rangecolor</b></a>
is the command alternative to <b>Render by Attribute</b> for coloring.
</p><p>
Close the model:
<blockquote>
<i>Command</i>: <b><a href="../midas/close.html">close</a> 0</b>
</blockquote>
Go on to <a href="#part2"><b>Part 2</b></a> below, <b>OR</b>
terminate the Chimera session with the following command:
<blockquote>
<i>Command</i>: <b><a href="../midas/stop.html">stop</a></b>
</blockquote>
</p>
<hr>
<a name="part2">
<h3>Attributes, Part 2 - GTP-Binding Protein</h3></a>

<table align="right" style="margin:8px 8px 8px 8px">
<tr><th align="center">
<a name="fig2">121p colored by convexity</a></th></tr>
<tr><td><img src="att2.png"></td></tr>
</table>

<p>
Begin with Chimera started and the
<a href="../chimerawindow.html#emulator"><b>Command Line</b></a>
(and optionally the <a href="../sideview.html#sideview"><b>Side View</b></a>)
opened as described at the beginning of <a href="#part1"><b>Part 1</b></a>.
</p><p>
If you have internet connectivity,
use a command to fetch the PDB structure <b>121p</b>:
<blockquote>
<i>Command</i>: <b><a href="../midas/open.html">open</a> 121p</b>
</blockquote>
If you do not have internet connectivity,
<a href="download.html">download</a> the file <a href="121p.pdb">121p.pdb</a> 
included with this tutorial and use <b>File... Open</b> to open it.
</p><p>
The structure is H-ras, a small GTP-binding protein, along with
a bound GTP analog, a Mg<sup>++</sup> ion, and some water.
<a href="../mouse.html">Move and scale</a> the structure
as desired throughout the tutorial.
</p><p>
Use the ribbons <a href="../menu.html#menupresets">preset</a>: 
<blockquote>
<i>Command</i>: 
<b><a href="../midas/preset.html">preset</a> apply int 1</b>
</blockquote>
This shows a ribbon rainbow-color-coded from blue at the N-terminus 
to red at the C-terminus, plus atoms in or near binding sites.
Delete the water (most of which is hidden)
and label the residues near the ion:
<blockquote>
<i>Command</i>: <b><a href="../midas/delete.html">delete</a> solvent</b>
<br>
<i>Command</i>: <b><a href="../midas/focus.html">focus</a> ions z<3.5</b>
<br>
<i>Command</i>: <b><a href="../midas/rlabel.html">rlabel</a> ions z<3.5</b>
</blockquote>
The active site Mg<sup>++</sup> ion is coordinated by Ser 17, Thr 35, 
and phosphonate oxygens in the GTP analog, which is named GCP.
Color the ligand GCP yellow, remove the residue labels, and zoom back out:
<blockquote>
<i>Command</i>: <b><a href="../midas/color.html">color</a> yellow ligand</b>
<br><i>Command</i>: <b><a href="../midas/rlabel.html">~rlabel</a></b>
<br><i>Command</i>: <b><a href="../midas/focus.html">focus</a></b>
</blockquote>
In <a href="https://mizuguchilab.org/homstrad/"
target="_blank">HOMSTRAD</a>, this protein is classified in the 
<a href="https://mizuguchilab.org/cgi-bin/homstrad/showpage.cgi?family=gtp&disp=str"
target="_blank">GTP-binding protein</a> family.
<a href="download.html">Download</a> the alignment file
<a href="homstrad-gtp.pir">homstrad-gtp.pir</a>
(originally from HOMSTRAD, now included with this tutorial)
and open it with <b>File... Open</b>.
</p><p>
The alignment is shown in a separate
<a href="../../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a> window.
A colored rectangle under the sequence name <b>5p21</b> indicates
association of that sequence with the structure.
If the sequence window later becomes obscured,
it can be raised by choosing <b>MAV - homstrad-gtp.pir... Raise</b>
from near the bottom of the <b>Tools</b> menu.
</p><p>
When a structure is associated with a sequence alignment in
<a href="../../ContributedSoftware/multalignviewer/framemav.html" 
target="_top"><b>Multalign Viewer</b></a>, its residues are assigned
<a href="../../ContributedSoftware/multalignviewer/multalignviewer.html#mavAttributes">
conservation attributes</a>.  From the <b>Multalign Viewer</b>
menu, choose <b>Structure... Render by Conservation</b>.  This opens the same
<a href="../../ContributedSoftware/render/render.html#render"><b>Render 
by Attribute</b></a> tool used in <a href="#part1"><b>Part 1</b></a>, except
that now it shows the attribute of <b>residues</b> named 
<b>mavConservation</b>. The &ldquo;mav&rdquo; part of the name is shorthand for
<b>Multalign Viewer</b>, and &ldquo;Conservation&rdquo; indicates that the 
values correspond to what is shown in the <b>Conservation</b> line of the
<a href="../../ContributedSoftware/multalignviewer/multalignviewer.html#seqwindow">sequence window</a>.
By default, the value for a column is the fraction of sequences with
the most common residue type at that position, for example:
0.7 where 7 of 10 sequences have the same type of residue,
0.1 where each of the 10 has a different type.
(Different calculation methods are available, see 
<a href="../../ContributedSoftware/multalignviewer/multalignviewer.html#mavpref-analysis"><b>Preferences... Headers</b></a>
in the <b>Multalign Viewer</b> menu.)
</p><p>
In <a href="../../ContributedSoftware/render/render.html#render"><b>Render
by Attribute</b></a>, use <b>Colors</b> and
adjust the color mapping as desired before clicking <b>Apply</b>.  
The coloring shows that many positions in or near the binding site
are highly conserved.
</p><p>
The <b>Render by Attribute</b> dialog also lists the residue attribute
<b>mavPercentConserved</b>.  Whereas <b>mavConservation</b> changes
along with the <b>Conservation</b> calculation method,
<b>mavPercentConserved</b> is always the percentage of sequences 
with the most common residue type at that position
(same as the <b>Conservation</b> default, except expressed
as a percentage rather than a fraction).
</p><p>
Select residues within 6 &Aring; of the ligand:
<blockquote>
<i>Command</i>: <b><a href="../midas/select.html#newer">select</a> ligand z&lt;6</b>
</blockquote>
The selection is shown with green boxes on the sequence alignment.
</p><p>
Like other attributes, <b>mavConservation</b> and <b>mavPercentConserved</b>
can be used in commands:
<blockquote>
<i>Command</i>: <b><a href="../midas/select.html#newer">select</a> ligand z&lt;6 & :/mavPercentConserved&gt;80</b>
</blockquote>
This selects a smaller set of residues than before, only those both
within 6 &Aring; of the ligand and meeting the conservation criterion.
Clear the selection (<b>Select... Clear Selection</b> is one way) and
<b>Quit</b> from <b>Multalign Viewer</b>.
</p><p>
Show the molecular surface of the protein:
<blockquote>
<i>Command</i>: <b><a href="../midas/surface.html">surface</a></b>
</blockquote>
By default, the surface colors match the underlying atom colors. It is again 
evident that highly conserved residues line the binding pocket.
Generating a molecular surface automatically creates the atom attributes
<b>areaSAS</b> and <b>areaSES</b>, solvent-accessible
and solvent-excluded surface areas in the context of the structure.  
For example, the values of both attributes are zero for
atoms buried in the protein interior.  What Chimera shows is 
the <b><i>solvent-excluded</i></b> molecular surface,
composed of probe contact, toroidal, and reentrant surface.
The <b><i>solvent-accessible</i></b> surface (not displayed)
is farther out from the molecule, the surface traced out by the probe center.
</p><p>
If <b>Render by Attribute</b> is not already open, open it 
(<b>Tools... Structure Analysis... Render by Attribute</b>) and see 
what attributes are present for <b>atoms</b>.  
If <b>areaSAS</b> and <b>areaSES</b> are not yet listed,
show them with <b>Refresh... Menus</b>.
The histograms show the ranges of values obtained for the structure.
</p><p>
Open the
<a href="../../ContributedSoftware/calculator/calculator.html"><b>Attribute
Calculator</b></a> (<b>Tools... Structure Analysis... Attribute Calculator</b>).
Calculate a new attribute named <b>convexity</b> for <b>atoms</b> 
using the <b>Formula</b>
<blockquote>
atom.areaSAS/atom.areaSES
</blockquote>
Values of <b>convexity</b>
&gt; 1 represent convex areas, while values &lt; 1 represent concave areas.  
Click <b>OK</b> to perform the calculation and assignment.  
A warning message will appear because some atoms have an
<b>areaSES</b> of zero, resulting in a divide-by-zero error. 
However, just close the warning dialog; the attribute 
has been assigned correctly for the atoms with nonzero areas.
</p><p>
Finally,
show the <b>convexity</b> values with color on the protein surface.
In the <b>Render by Attribute</b> dialog, 
make sure that the histogram is showing the
new attribute of <b>atoms</b> named <b>convexity</b>.  In the
<b>Colors</b> section, use three thresholds:
<ul>
<li>drag the lowest-value (leftmost) threshold to the far left and make it
<b>purple</b>
(click the <a href="../coloring.html#colorwell">color well</a> and then
enter the <b>Color name</b> in the <a href="../colortool.html">
<b>Color Editor</b></a>) 
<li>drag the highest-value threshold to the far right and make it <b>yellow</b>
<li>click the middle threshold, make it <b>white</b>, and set its value to 1
by simply entering that number in the <b>Value</b> field and pressing return
</ul>
Click <b>Apply</b>; the result should resemble the <a href="#fig2">figure</a>.
</p><p>
The surface can be smoothed by increasing its vertex density
from the default of 2.0, for example:
<blockquote>
<i>Command</i>: <b><a href="../midas/setattr.html">setattr s</a> density 5</b>
</blockquote>
</p><p>
When finished, end the Chimera session:
<blockquote>
<i>Command</i>: <b><a href="../midas/stop.html">stop really</a></b>
</blockquote>
</p>
<hr>
<address>meng-at-cgl.ucsf.edu / August 2014</address>
</body></html>
