<!--
--- UCSF Chimera Copyright ---
Copyright (c) 2012-2015 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
--- UCSF Chimera Copyright ---
-->

<html><head>
<title> Comparative Modeling Tutorial</title>

<style type="text/css">
A.nounder:link {text-decoration: none}
A.nounder:visited {text-decoration: none}
A.nounder:hover {text-decoration: underline; color: red;}
table.clRight {float: right; clear: right; margin: 8px}
img.clRight {float: right; clear: right; margin: 8px; border: 1px solid }
</style>
</head><body>

<font size="-1">
<a href="../frametut.html" target="_top">Chimera Tutorials Index</a>
</font>

<table class="clRight">
<tr><td align="center"><img src="dor-template-trimmed.png" 
alt="template structure" border="1px"></td>
<td align="center"><img src="dor-5models-trimmed.png" 
alt="5 comparative models" border="1px"></td></tr>
<tr><td align="center"><b>template</b>: &mu;-opioid receptor</td>
<td align="center" valign="bottom"><b>target</b>: &delta;-opioid receptor
<br>(5 comparative models)</td></tr>
</table>

<a name="top"></a>
<h2>Comparative Modeling Tutorial</h2>
<p>
This tutorial includes running 
<a href="../blast.html">Blast sequence search</a> and 
<a href="../../ContributedSoftware/multalignviewer/modeller.html#comparative">Modeller comparative modeling</a>
calculations from Chimera.
Internet connectivity is required to fetch data and to access 
Blast, Modeller, and other web services. Although no software installation
(other than Chimera itself) is needed to follow the tutorial, 
Modeller use requires a license key. Academic users can obtain a license
key free of charge by
<a href="https://www.salilab.org/modeller/registration.html"
target="_blank">registering</a> at the Modeller website.
<ul>
<li><a href="#background">Background and Caveats</a>
<li><a href="#blast">Blast Search for Templates</a>
<li><a href="#alignment">Verifying the Alignment</a>
<li><a href="#modeller">Running Modeller</a>
</ul>
See also: <a href="../../ContributedSoftware/multalignviewer/multalignviewer.html#approaches">ways to generate comparative modeling inputs</a>,
<a href="../midas/mda.html"><b>mda</b></a>
</p>

<h3><a href="#top" class="nounder"><b>&larr;</b></a>
<a name="background">Background and Caveats</a></h3>
<p>
In comparative (homology) modeling,
theoretical models of a protein are built using at least one known related
structure and a sequence alignment of the known and unknown structures.
The protein to be modeled is the <b><i>target</i></b>, and a related
known structure used for modeling is a <b><i>template</i></b>.
</p><p>
The target in this tutorial is the human 
<a href="https://en.wikipedia.org/wiki/Delta_Opioid_receptor"
target="_blank">&delta;-opioid receptor</a>,
a G-protein-coupled receptor (GPCR). GPCRs are transmembrane proteins 
and have been relatively resistant to structure determination.
Although recent advances have allowed solving the structures of
several members of this large and biomedically important class of proteins,
at the time of creating this tutorial (May 2012),
no structure was available for the &delta;-opioid receptor.
<!--
Naturally, soon thereafter a structure of the mouse DOR (4ej4) became 
available, but it is also a lysozyme fusion and thus presents the same issues
as the template used in this tutorial, 4dkl (mouse MOR), except it has 
much better %ID to the target (hDOR vs. mDOR 94%, hDOR vs. mMOR 60%).
-->
</p><p>
Tutorial caveats:
</p>
<ul>
<li>As database contents and web services are updated, the 
results of calculations are likely to differ from what is described here.  
However, this tutorial is meant to illustrate the general process 
rather than any specific result. 
<li>This tutorial is not meant to indicate the optimal parameter settings for 
comparative modeling, as these will vary depending on the system of interest 
and the information available at the time.
<li>This tutorial describes only one of several equally valid approaches.  
For example, the sequence of the target could be read from
a FASTA file instead of fetched from the UniProt database.
<li>Modeller quality scores were developed for globular (soluble) proteins
and may be less effective indicators of model quality 
for transmembrane proteins such as GPCRs.
</ul>

<img class="clRight" src="dor-seq.png" alt="dOR sequence">
<img class="clRight" src="dor-seqregions.png" alt="dOR annotations">
<img class="clRight" src="dor-blast.png" alt="dOR annotations">

<h3><a href="#top" class="nounder"><b>&larr;</b></a>
<a name="blast">Blast Search for Templates</a></h3>
<p>
<a href="../startup.html">Start Chimera</a>.
A splash screen will appear, to be replaced in a few seconds
by the main Chimera <a href="../chimerawindow.html">graphics window</a>
or <a href="../rapid.html"><b>Rapid Access</b></a> interface (it does not
matter which, the following instructions will work with either).
</p><p>
Choose <b>File... Fetch by ID</b> from the menu and use the resulting dialog
to fetch the sequence of the target, the human &delta;-opioid receptor: 
<b>UniProt</b> ID <b>oprd_human</b>.
If you want to verify the ID before fetching, click the <b>Web Page</b> button
on the fetch dialog to see the corresponding page at 
<a href="https://www.uniprot.org" target="_blank">UniProt</a>.
(One way to determine the ID in the first place is by searching at
the <a href="https://www.uniprot.org" target="_blank">UniProt</a> site.)
</p><p>
The sequence is displayed in
<a href="../../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a>, and its UniProt feature
annotations listed in the
<a href="../../ContributedSoftware/multalignviewer/multalignviewer.html#regionbrowser"><b>Region Browser</b></a>.
The <b>S</b> column checkboxes in the <b>Region Browser</b> can be used
to show feature annotations as colored boxes in the sequence window.
Close the <b>Region Browser</b>; it can be accessed any time
from the sequence window <b>Info</b> menu.
</p><p>
The next step is to find a known protein structure suitable for use
as a modeling template. We will use Chimera's
<a href="../blast.html"><b>Blast Protein</b></a> tool to search the
<a href="https://www.rcsb.org" target="_blank">Protein Data Bank</a> 
(PDB; a database of known structures) for sequences similar to the target.
From the sequence window menu, choose <b>Info... Blast Protein</b>, click
<b>OK</b> to use <b>oprd_human</b> as the query, and <b>OK</b> again to
perform the search using default settings, including <b>pdb</b>
as the database to search. 
Searching the <b>pdb</b> sequences should take only a few seconds.
Searching the <b>nr</b> database, which also contains a huge number of 
sequences without known structures, would take much longer.
</p><p>
In the Blast results dialog, the hits are listed from best to worst.
Click the <b>Columns</b> button to reveal several checkboxes 
for controlling which columns of information are shown.  
Hide (uncheck) <b>Description</b>, then show <b>Resolution</b> and 
<b>Chain names</b>.  As shown in the figure, the two best hits 
are opioid receptors, followed by several other types of GPCRs.
One technique for GPCR structure determination is to express the receptor
as a fusion with some other protein that favors crystallization.
The two best hits, PDB entries
<a href="https://www.rcsb.org/structure/4DKL"
target="_blank">4DKL</a> and
<a href="https://www.rcsb.org/structure/4DJH"
target="_blank">4DJH</a>,
contain structures of opioid receptors fused with lysozyme.  
</p><p>
It is possible to use multiple templates, but we will use just 4DKL_A
(PDB entry 4DKL, chain A).
The <b>Chain names</b> information says to &ldquo;SEE REMARK 999,&rdquo;
which we will do after opening the structure.
<a name="blastresults">
In the Blast results dialog, click to highlight the corresponding row, 
then at the bottom of the dialog:
</a>
</p>
<ol>
<li>click <b>Show in MAV</b> to display the query-hit sequence alignment
from Blast in another
<a href="../../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a> (MAV) window 
<li>click <b>Load Structure</b> to fetch 4DKL from the PDB and open it in
Chimera
<li>click <b>Quit</b> to dismiss the Blast results dialog
</ol>
<a name="remark999">
To see the &ldquo;REMARK 999&rdquo; lines in the PDB file of the structure:
</a>
  <ul>
  <li>choose <b>Favorites... Model Panel</b> from the Chimera menu
  <li>in the <a href="modelpanel.html"><b>Model Panel</b></a>,
  click <b>attributes...</b> to show the attributes of model <b>4DKL</b>
  <li>at the bottom of the attributes dialog, click <b>PDB Headers...</b>
  <li>in the resulting dialog, scroll down to see the <b>REMARK 999</b> lines:
<pre>
REMARK 999                                                                      
REMARK 999 SEQUENCE                                                             
REMARK 999 CHAIN A IS AN INTERNAL FUSION OF LYSOZYME (RESIDUES 2-161 OF UNP     
REMARK 999 P00720) BETWEEN RESIDUES 52-263 AND RESIDUES 270-352 OF MU-TYPE      
REMARK 999 OPIOID RECEPTOR (UNP P42866). AN OFFSET OF 1000 HAS BEEN ADDED TO    
REMARK 999 LYSOZYME RESIDUE NUMBERS WITHIN THE COORDINATES TO DISTINGUISH THAT  
REMARK 999 PORTION OF CHAIN A. LYSOZYME RESIDUES ARE THEREFORE NUMBERED 1002-   
REMARK 999 1161.     
</pre>
  From this, we note:
    <ul>
    <li>the &mu;-opioid receptor portion has UniProt ID
    <a href="https://www.uniprot.org/uniprot/p42866" 
    target="_blank"><b>p42866</b></a> (equivalent to <b>oprm_mouse</b>)
    <li>the lysozyme residues are numbered 1002-1161
    </ul>
    <br>
  <li>Close the headers dialog, attributes dialog, and <b>Model Panel</b>
  </ul>
<img class="clRight" src="dor-template.png" alt="template structure 4DKL_A">
<p>
Show the Chimera 
<a href="../chimerawindow.html#emulator"><b>Command Line</b></a>
(for example, with <b>Favorites... Command Line</b>), hide atoms, 
rainbow-color the ribbon, and make the lysozyme portion dark:
</p>
<blockquote>
<i>Command</i>: <b><a href="../midas/display.html">~display</a></b>
<br>
<i>Command</i>: <b><a href="../midas/rainbow.html">rainbow</a></b>
<br>
<i>Command</i>: <b><a href="../midas/alias.html">alias</a> lyso :1002-1161</b>
<br>
<i>Command</i>: <b><a href="../midas/color.html">color</a> dim gray lyso</b>
</blockquote>
<p>
GPCRs have an extracellular N-terminus,
seven transmembrane helices, and an intracellular C-terminus.
Visually tracing the structure from N-terminus (blue) to C-terminus (red)
reveals that lysozyme is inserted between the fifth and sixth transmembrane
helices, in the third intracellular loop.
</p>

<h3><a href="#top" class="nounder"><b>&larr;</b></a>
<a name="alignment">Verifying the Alignment</a></h3>
<p>
Comparative modeling requires a template structure and a target-template
sequence alignment.  The sequence alignment is important;
it controls which residues in the template are used to model 
which residues in the target, and any inaccuracies in the alignment will
result in the application of incorrect constraints during 3D modeling.
Regardless of how the sequence alignment was obtained, 
it should be examined and adjusted as needed before initiating 
the more computationally intensive 3D modeling calculations.
</p>
<a href="dor-badalign.png">
<img class="clRight" src="dor-badalign.png" width="480" 
title="Blast alignment; click to see full size..."
alt="Blast alignment"></a>
<p>
An alignment was generated <a href="#blastresults">above</a>
as a by-product of the Blast similarity search.  
However, Blast is meant to identify local similarities quickly
rather than to give accurate full-length alignments.
View the Blast alignment to see if it looks reasonable.
</p><p>
In general, if a Chimera window or dialog is obscured by other windows, 
it can be accessed using its <a href="../instances.html">instance</a> 
near the bottom of the <b>Tools</b> menu, or 
from the Active Dialogs section of the 
<a href="../rapid.html"><b>Rapid Access</b></a> interface (itself shown by 
clicking the lightning bolt icon near the bottom of the main Chimera window).
</p><p>
Use the sequence window <b>Headers</b> menu to hide the <b>Consensus</b>
and <b>Conservation</b> lines, then scroll or resize the window to show
the whole alignment.  A large part of the query (target) 
sequence is not aligned. To understand what happened, try coloring
the structure sequence to match the ribbon display:
from the sequence window menu, choose <b>Preferences... Appearance</b>
and in the resulting dialog, change the multiple alignments <b>Color scheme</b>
to <b>ribbon</b>. As shown in the figure,
the coloring reveals that the alignment includes the first five
transmembrane helices but cuts off in the lysozyme insertion.
The rest of the structure sequence is omitted, 
leaving the rest of the query unaligned. 
</p><p>
Thus the alignment from Blast is not adequate for modeling purposes, 
and the target-template sequence alignment must be generated in some other way.
Click <b>Quit</b> to close the sequence alignment from Blast. 
</p><p>
To generate the target-template sequence alignment, we will
return to the original <b>oprd_human</b> (target) sequence and 
use the Needleman-Wunsch global alignment algorithm to
add the sequence of the &mu;-opioid receptor (template).
If the sequence window was closed, not to worry,
the target sequence can be fetched again as described 
<a href="#blast">above</a> or using a command:
<blockquote>
<i>Command</i>:
<b><a href="../midas/open.html">open</a> &nbsp;uniprot:oprd_human</b>
</blockquote>
From the sequence window menu, choose <b>Edit... Add Sequence</b>.
The resulting dialog contains tabs for different ways of obtaining the
sequence.
</p><p>
In similar situations, it would usually be fine to add the
template sequence <b>From Structure</b>.  However, in this special
case of a chimeric template protein, the structure sequence includes 
residues from another protein (lysozyme) that should not be in the alignment.
Even deleting the residues from the structure, as will be done below, 
does not remove them from the structure sequence.
</p><p>
Instead, use the <b>From UniProt</b> tab and add ID <b>p42866</b>
(noted <a href="#remark999">above</a> for the &mu;-opioid receptor
part of the structure) using default alignment parameters.
The <b>Region Browser</b> will appear and can be closed.
</p>
<a href="dor-goodalign.png">
<img class="clRight" src="dor-goodalign.png" width="480" 
title="Needleman-Wunsch alignment; click to see full size..."
alt="Needleman-Wunsch alignment"></a>
<p>
Delete the lysozyme part of the template structure since only the
&mu;-opioid receptor part is useful for modeling the &delta;-opioid receptor:
<blockquote>
<i>Command</i>: <b><a href="../midas/delete.html">delete</a> lyso</b>
</blockquote>
The template structure still needs to be associated 
with the corresponding sequence in the alignment.
Often sequences and structures will associate automatically,
but in this case it is necessary to do it manually:
from the sequence window menu, choose
<b>Structure... Associations</b> and associate the structure <b>4DKL</b>
with the sequence <b>p42866</b>. 
</p><p>
In the sequence window, the association is indicated with a tan box 
around the sequence name (tan is the default color of the structure).
Red outline boxes enclose residues that are in the sequence but not
in the associated structure. 
There are quite a few missing residues: stretches at both ends and
a few in the third intracellular loop, which had been partly
replaced by lysozyme. However, the residues that are present in the 
template structure, including all seven transmembrane helices, 
are well-aligned with the target.
</p><p>
If you find the sequence coloring difficult to view, it can be changed to
some other scheme (such as all black) using <b>Preferences... Appearance</b>
in the sequence window menu.  This also allows changing sequence wrapping,
font size, <i>etc.</i>  
Coloring the sequence to match the structure ribbon is only one of several
approaches for sequence-structure mapping.  For example:
<ul>
<li>highlighting residues in the sequence with the mouse
<a href="../selection.html">selects</a> them in the structure
<li><a href="../selection.html">selecting</a> residues in the structure 
highlights them in the sequence (green boxes)
<li>structure helix and strand assignments can be shown on the sequence 
with <b>Structure... Secondary Structure... show actual</b>
</ul>
</p>

<h3><a href="#top" class="nounder"><b>&larr;</b></a>
<a name="modeller">Running Modeller</a></h3>

<table class="clRight" cellpadding="0" cellspacing="0">
<tr><td><img src="dor-5models.png" 
alt="5 comparative models" border="1px"></td></tr>
<tr><td><img src="dor-modellist1.png" width="175"
alt="Model List dialog" border="1px"></td></tr>
</table>
<p>
From the sequence window menu, choose <b>Structure... Modeller (homology)</b>
to open the Chimera interface to 
<a href="../../ContributedSoftware/multalignviewer/modeller.html#comparative">comparative modeling with Modeller</a>.
The target should be set to <b>oprd_human</b>.
Click <b>p42866</b> in the dialog to choose it as the template.
</p><p>
Click the <b>Advanced Options</b> button to reveal additional settings.
<b>Run Modeller via web service</b> indicates using a web service hosted
by the <a href="https://www.rbvi.ucsf.edu" target="_blank">UCSF RBVI</a>.
No local installation is required to run the web service,
but it is necessary to enter a <b>Modeller license key</b>,
available free of charge to academic users upon
<a href="https://www.salilab.org/modeller/registration.html"
target="_blank">registration</a> at the Modeller website.  
After entering the license key, click <b>OK</b> 
to launch the calculation with default settings.
Five comparative models will be generated.
</p><p>
The Modeller run may take several minutes and is handled as a
background task.  Clicking the information icon <img src="../taskicon.png">
near the bottom of the Chimera window will bring up the 
<a href="../../ContributedSoftware/idle/task.html"><b>Task Panel</b></a>,
in which the job can be canceled if desired.
</p><p>
When the five models have been generated, they will be
opened in Chimera and their evaluation scores shown in
a <b>Model List</b> dialog.  The models can be viewed individually
or collectively by choosing rows in the dialog with the mouse.
The different scores from Modeller use different criteria and will not
necessarily agree on which models are best:
<ul>
<li>
<a href="https://modbase.compbio.ucsf.edu/modbase-cgi/display.cgi?type=help&server=modbase#model_score"
target="_blank"><b>GA341</b></a> -
model score derived from statistical potentials;
a value &gt; 0.7 generally indicates a reliable model,
&gt;95% probability of having the correct fold
<li>
<a href="https://modbase.compbio.ucsf.edu/modbase-cgi/display.cgi?type=help&server=modbase#dope"
target="_blank"><b>zDOPE</b></a>
- normalized Discrete Optimized Protein Energy (DOPE), an
atomic distance-dependent statistical score;
negative values indicate better models
</ul>
Although Modeller scores were developed for globular proteins and thus have 
limited applicability to the transmembrane protein in this tutorial,
another reason for the poor (positive) zDOPE scores is that the termini 
extend far beyond the template structure and cannot be modeled reliably.
Displaying all the models at once shows little conformational variability
except in the termini, and to a lesser extent, the untemplated part
of the third intracellular loop. This conclusion is reinforced by the
<b>RMSD</b> histogram in the sequence window, where bar heights indicate 
root-mean-square distances among the &alpha;-carbons of the residues 
associated with a column.
</p>
<table class="clRight" cellpadding="0" cellspacing="0">
<tr><td><img src="dor-modellist2-best.png" width="175"
alt="Model List dialog" border="1px"></td></tr>
<tr><td><img src="dor-bestmodel-hydrophob.png" 
alt="model #1.2, best est RMSD"></td></tr>
</table>
<p>
Although there is also a Chimera interface to Modeller for
<a href="../../ContributedSoftware/multalignviewer/modeller.html#building">untemplated building and refinement</a>,
in this tutorial we will simply remove the termini and rescore the models.
</p><p>
Click into the sequence window, then move the cursor over the residues to 
see the corresponding structure residue numbers near the bottom of the
window. In the comparative models (#1.1-5), Leu-46 is aligned with the first 
residue in the template structure and Arg-334 is aligned with the last 
residue in the template structure. 
Delete the termini in the comparative models that extend beyond the template:
</p>
<blockquote>
<i>Command</i>: 
<b><a href="../midas/delete.html">del</a> #1:start-45,335-end</b>
</blockquote>
<p>
To rescore the models,
choose <b>Fetch Scores... zDOPE and Estimated RMSD/Overlap</b> from the
<b>Model List</b> menu.
Rescoring uses a web service provided by the 
<a href="https://www.salilab.org/" target="_blank">Sali lab</a> at UCSF.  
After a minute or few, more favorable zDOPE values are obtained, 
along with the additional scores:
</p>
<ul>
<li><a href="https://modbase.compbio.ucsf.edu/modbase-cgi/display.cgi?type=help&server=modbase#tsvmod"
target="_blank"><b>Estimated RMSD</b></a>
- TSVMod-predicted C&alpha; root-mean-square deviation (RMSD)
of the model from the native structure
<li><a href="https://modbase.compbio.ucsf.edu/modbase-cgi/display.cgi?type=help&server=modbase#tsvmod"
target="_blank"><b>Estimated Overlap (3.5 &Aring;)</b></a>
- TSVMod-predicted native overlap (3.5 &Aring;),
fraction of C&alpha; atoms in the model within 3.5 &Aring;
of the corresponding atoms in the native structure after rigid-body
superposition 
</ul>
The comparative models are atomically detailed and
can be subjected to various analyses in Chimera. In the figure, the
model with the best (lowest) estimated RMSD score is shown as spheres 
colored by <a href="../midas/hydrophob.html">amino acid hydrophobicity</a>, 
from <b>dodger blue</b> for the most hydrophilic to <b>white</b>
to <b>orange red</b> for the most hydrophobic 
(see <a href="../colortables.html#allcolors">color names</a>). 
The <b>Model List</b> dialog was used to show only this model of the five,
then the following commands were used to hide the template and adjust the 
model's appearance:
<blockquote>
<i>Command</i>: <b><a href="../midas/modeldisplay.html">~modeldisp</a> #0</b>
<br><i>Command</i>: <b><a href="../midas/display.html">disp</a></b>
<br><i>Command</i>: <b><a href="../midas/ribbon.html">~ribbon</a></b>
<br><i>Command</i>:&nbsp;<b><a href="../midas/rangecolor.html">rangecol</a>&nbsp;kdHydrophobicity&nbsp;min&nbsp;dodger&nbsp;blue&nbsp;mid&nbsp;white&nbsp;max&nbsp;orange&nbsp;red</b>
<br><i>Command</i>: <b><a href="../midas/preset.html">preset</a> apply pub 1</b>
<br><i>Command</i>: <b><a href="../midas/represent.html">repr</a> sphere</b>
</blockquote>
<hr>
<address>meng-at-cgl.ucsf.edu / July 2015</address>
</body></html>
