<!--
--- UCSF Chimera Copyright ---
Copyright (c) 2000-2017 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
--- UCSF Chimera Copyright ---
-->

<html><head>
<title>open, close</title>
</head><body>
<font size="-1">
<a href="../framecommand.html" target="_top">Chimera Commands Index</a>
</font>

<h3><a href="usageconventions.html">Usage</a>: 
<BR><b>open</b> [ noprefs ] [ nowildcard ]
[ <i>model-ID</i> ] [ <i>filename1</i> <i>filename2</i> ... ]</h3>
<h3><a href="usageconventions.html">Usage</a>: 
<BR>( <b>close</b> | <b>~open</b> ) 
( <i>model-spec</i> | <b>all</b> | <b>session</b> )</h3>
<p>
The command <b>open</b> reads <a href="#local">local files</a> and
fetches <a href="#fetch">data from the Web</a>.
Chimera <a href="../filetypes.html">input files</a> may contain structures 
to be displayed, commands or code to be executed, or other data;
the filename(s), if any, should be the last arguments in the command.
If no filename is supplied, a <a href="../opensave.html">dialog</a>
for browsing the local filesystem for input will appear.
Any resulting models in Chimera are assigned the specified <i>model-ID</i>
number, or if none is specified, the lowest number(s) available.
When multiple models are opened with a single command, <i>model-ID</i>
should not be specified unless there is some reason to assign all of the
models the same number.
</p><p>
Various attributes of newly opened molecules can be customized
in the <a href="../preferences.html#New Molecules">
<b>New Molecules</b> preferences</a>
(see also <b><A href="set.html#autocolor">set autoColor</a></b>).
The <b>noprefs</b> keyword indicates that a user's
<a href="../preferences.html#New Molecules"><b>New Molecules</b> preferences</a>
should not be applied.  This prevents inconsistent behavior of
<a href="../indexcommand.html#cmdfile">command files</a> and
<a href="../../ContributedSoftware/demos/demos.html">demos</a>
potentially caused by the different preferences settings of different users.
</p><p>
If a model is closed and another model opened with the same model number,
none of the transformations applied to the previous model are applied to
the newly opened model.  The transformation matrix of one model can be applied
to another model using <b><A href="matrixcopy.html">matrixcopy</A></b>.</p>
</p><p>
The command <b>close</b> or <b>~open</b> removes structures and other data:
model(s) specified by number, or <b>all</b> models (everything listed in the 
<a href="../modelpanel.html"><b>Model Panel</b></a>), 
or the <b>session</b> (all models, plus data without associated model numbers,
such as <a href="2dlabels.html">2D labels</a> and 
<a href="../filetypes.html#alignment">sequence alignments</a>).
Closing a session shows the <a href="../rapid.html"><b>Rapid Access</b></a>
interface.
</p><p>
See also:
<b><A href="read.html">read</A></b>,
<b><A href="runscript.html">runscript</A></b>,
<b><A href="split.html">split</A></b>,
<b><A href="start.html">start</A></b>,
<a href="../rapid.html"><b>Rapid Access</b></a>,
<b><A href="../fetch.html">Fetch by ID</A></b>,
<a href="../filetypes.html">Chimera input file types</a>
</p>

<a name="local">
<h3>Local Files</h3></a>
<p>
A filename is generally a pathname to a local file, either absolute 
or relative to the current working directory.  The working
directory can be changed with <b><A href="cd.html">cd</A></b>.
A pathname may start with "~" to indicate the user's home directory.
Multiple files can be specified by including * (wild card), 
? (single-character wild card), and/or square-bracketed character ranges
in a filename.  Such filename expansion or <b><i>globbing</i></b> 
is on by default, but can be turned off with the <b>nowildcard</b> keyword.
If no filename is supplied, a <a href="../opensave.html">dialog</a>
for browsing the filesystem will appear.
</p><p>
Any of the <a href="../filetypes.html">registered file types</a>
can be opened.
</p><p>
A file type can be specified
by a <a href="../filetypes.html#suffix">suffix</a>
that is part of the filename or by
<b><i><a href="../filetypes.html#prefix">prefix</a>:filename</i></b>,
where <a href="../filetypes.html#prefix">prefix</a>
is not part of the filename.
If a prefix and a suffix are both given, the prefix overrides the suffix.
Filenames, prefixes, and suffixes are case-sensitive.
Unrecognized prefixes are assumed to be part of the filename.
For many of the <a href="../filetypes.html">registered file types</a>,
files that are gzipped (as indicated by <b>.gz</b> 
following the regular filename) are recognized and opened.
Similarly, compressed files (as indicated by <b>.Z</b>) 
can be recognized and opened
for many input types if <i>gzip</i> is on the user's execution path
(can be run by entering <i>gzip</i> at the system command line).
<a href="../tutorials/framepdbintro.html" target="_top">PDB format</a>
is the default input type.
</p>

<a name="fetch">
<h3>Files from the Web</h3></a>
<p>
Other than a local file, a filename can specify data to be retrieved 
over the Web.
Fetched files can be cached locally and reused as needed depending on the
<a href="../preferences.html#Fetch"><b>Fetch</b> preferences</a>
and <a href="../preferences.html#PDB"><b>PDB</b> preferences</a>.
<ul>
<li> many <a href="../filetypes.html">registered types</a> can be opened 
by specifying an URL of the form <b>http://[...]</b> as the filename.
Prefix and suffix rules are as described above for 
<a href="#local">local files</a>, but filename
globbing and reading compressed or gzipped files are not supported.
<li> the prefix <b>pdbID:</b> indicates a PDB ID
(4-character Protein Data Bank identifier)
to be used to fetch a PDB-format file from the Protein Data Bank:
  <ol>
  <li>Chimera will first attempt to find the file within a
  local installation of the Protein Data Bank.
  The default places to look for a local installation
  are <b>/usr/mol/pdb/</b> and then <b>/mol/pdb/</b>
  &mdash; these can be changed by editing the Python file
  <b>share/chimera/pdbDir</b> within a Chimera installation.
  <li> Next, Chimera will look for the file in any
  personal PDB directories designated in the
  <a href="../preferences.html#PDB"><b>PDB</b> preferences</a>.
  <li> Next, the PDB subdirectory of the 
  <a href="../preferences.html#Fetch">fetch download directory</a> will be 
  checked (unless the use of previously fetched files has been disallowed 
  in the <a href="../preferences.html#Fetch"><b>Fetch</b> preferences</a>).
  <li> Finally, if not found locally, the file will be fetched from the
  <a href="https://www.rcsb.org" target="_blank">Protein Data Bank website</a>
  (unless fetching has been disallowed in the
  <a href="../preferences.html#PDB"><b>PDB</b> preferences</a>).
  </ol>
<li> any input specified as PDB format (which is the default) will be
fetched in the same way if the filename resembles a
PDB ID and is not found in the current working directory
or any <a href="../preferences.html#PDB">personal PDB directories</a>
<li> the prefix <b>cifID:</b> indicates a PDB ID
to be used to fetch an mmCIF-format file
from the <a href="https://www.rcsb.org" target="_blank">Protein Data Bank</a>
<li> the prefix <b>ndb:</b> indicates an
<a href="http://ndbserver.rutgers.edu/index.html" target="_blank">NDB</a>
ID to be translated into a PDB ID and used to fetch the corresponding structure
from the <a href="https://www.rcsb.org" target="_blank">Protein Data Bank</a>
<li> the prefix <b>biounitID:</b> indicates a PDB ID 
to be used to fetch 
<a href="https://pdb101.rcsb.org/learn/guide-to-understanding-pdb-data/biological-assemblies#Anchor-BioUnit"
target="_blank">biological assemblies</a> from the
<a href="https://www.rcsb.org" target="_blank">Protein Data Bank</a>;
there may be multiple assemblies for a given entry, and a specific assembly
can be designated by appending &ldquo;.N&rdquo; to the PDB ID.
For example, <b>1dok</b> specifies all biological assemblies of entry
<a href="https://www.rcsb.org/structure/1DOK"
target="_blank">1dok</a>, <b>1dok.2</b> specifies biological assembly 2,
and <b>1dok.1.2</b> specifies biological assemblies 1 and 2.
Each assembly will be opened as a separate main model number, 
possibly containing <a href="../modelpanel.html#multimodel">submodels</a>.
<li> the prefix <b>pqsID:</b> indicates a 
PDB ID to be used to fetch the predicted biological unit 
from the <a href="http://pqs.ebi.ac.uk/pqs-doc.shtml" target="_blank">Protein 
Quaternary Structure server</a>
(predictions are not available for all PDB entries; some entries have
multiple predictions, and for those, multiple files will be retrieved)
<li> the prefix <b>viperID:</b> indicates a PDB ID to be used to fetch an
entry from the <a href="http://viperdb.scripps.edu/" target="_blank">Virus
Particle Explorer database</a> of icosahedral virus capsid structures
(the capsid will be constructed automatically with
<a href="../../ContributedSoftware/multiscale/framemulti.html"
target="_top"><b>Multiscale Models</b></a>)
<li> the prefix <b>scop:</b> indicates a
<a href="../domains.html">SCOP domain ID</a>
to be used to fetch a domain structure from the
<a href="https://scop.berkeley.edu/about/" target="_blank">ASTRAL</a> database
<li> the prefix <b>castp:</b> or <b>CASTp:</b> indicates a
PDB ID (with chain ID optionally appended, for example <b>2gsh.A</b>)
to be used to fetch a structure and its
precomputed pocket measurements from the 
<a href="http://sts.bioe.uic.edu/castp/" target="_blank">
Computed Atlas of Surface Topography of proteins</a>
(does not include results for all PDB entries; measurements
will be displayed in a <a href="../castp.html">pocket list</a>)
<li> the prefix <b>modbase:</b> indicates a 
SwissProt, TrEMBL, GenPept or PIR accession code for which to fetch 
PDB-format comparative models from
<a href="https://modbase.compbio.ucsf.edu/modbase-cgi/index.cgi"
target="_blank">ModBase</a> (associated information will be shown 
in a <a href="../modbase.html">list</a>).
If multiple modeled structures are available for the sequence,
they will be opened as <a href="atom_spec.html#subcats">submodels</a>
of a single model number.  
<li><a name="smiles">the prefix <b>smiles:</b> or <b>SMILES:</b></a> 
indicates a 
<a href="https://www.daylight.com/dayhtml/doc/theory/theory.smiles.html"
target="_blank">SMILES</a> string to be converted to a 3D structure using the
<a href="https://cactus.nci.nih.gov/translate/" target="_blank">SMILES
translator</a> provided by the 
<a href="https://cactus.nci.nih.gov/" target="_blank">National Cancer Institute
CADD group</a>, or if that does not produce a structure, the
<i>smi23d</i> web service provided by the
cheminformatics group at Indiana University.
The <i>smi23d</i> service deploys the same procedure 
as described for <b>Pub3D</b> in
<a href="https://www.biomedcentral.com/1471-2105/8/487"
target="_blank">Willighagen <i>et al.</i></a>, <i>BMC Bioinformatics</i>
<b>8</b>:487 (2007).
<li><a name="pubchem">the prefix <b>pubchem:</b> or <b>PubChem:</b></a> 
indicates a 
<a href="https://pubchem.ncbi.nlm.nih.gov/" target="_blank">PubChem</a>
compound identifier (CID) for which to fetch a modeled 3D structure from
<a href="https://pubchemdocs.ncbi.nlm.nih.gov/pubchem3d"
target="_blank"><b>PubChem3D</b></a>
(this resource includes most but not all PubChem Compound entries, 
<i>e.g.</i>, only the parent forms of salts, and subject to limits 
on compound size, flexibility, and constituent atom types)
<li> the prefix <b>cellpackID:</b> indicates a 
<a href="http://www.cellpack.org" target="_blank">cellPACK</a> model identifier
(see <a href="http://www.cellpack.org/use/" target="_blank">cellPACK
Chimera instructions</a> for available identifiers)
<li> the prefix <b>edsID:</b> indicates a
PDB ID to be used to fetch an electron density map from the
<a href="https://www.ebi.ac.uk/pdbe/eds/" target="_blank">Protein Data Bank
in Europe</a> (not all PDB entries have maps available;
<a href="../../ContributedSoftware/volumeviewer/framevolumeviewer.html"
target="_top"><b>Volume Viewer</b></a> will be used to show the map)
<li> the prefix <b>edsdiffID:</b> indicates a
PDB ID to be used to fetch an electron density 
difference map (fo-fc) from the
<a href="https://www.ebi.ac.uk/pdbe/eds/" target="_blank">Protein Data Bank
in Europe</a> (not all PDB entries have maps available;
<a href="../../ContributedSoftware/volumeviewer/framevolumeviewer.html"
target="_top"><b>Volume Viewer</b></a> will be used to show the map)
<li> the prefix <b>emdbID:</b> indicates a
numerical identifier to be used to fetch an
electron density map from the
<a href="https://www.ebi.ac.uk/pdbe/emdb/" target="_blank">Electron 
Microscopy Data Bank</a> (or Chinese mirror site, for .cn hosts)
<li> the prefix <b>emdbfitID:</b> indicates a
numerical identifier to be used to fetch an
electron density map from the
<a href="https://www.ebi.ac.uk/pdbe/emdb/" target="_blank">Electron 
Microscopy Data Bank</a> (or Chinese mirror site, for .cn hosts),
along with any corresponding
<a href="https://www.rcsb.org" target="_blank">Protein Data Bank</a>
entries, which may or may not be in the fit positions relative to the map.
PDB entries only available as mmCIF will be fetched in that format
instead of in PDB format.
<li>the prefix <b>uniprot:</b> indicates a
<a href="https://www.uniprot.org/" target="_blank">UniProt</a>
accession code or identifier to be used to fetch a protein sequence 
and its annotations and show them in
<a href="../../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a> (much as described for
<a href="../../ContributedSoftware/uniprot/uniprot.html#seqanno"><b>PDB/UniProt
Info</b></a>, except independent of structure).  
This type of fetch is not saved locally even if a
<a href="../preferences.html#Fetch">download directory</a> has been specified.
UniProt's <a href="https://www.uniprot.org/uploadlists/" target="_blank">ID 
mapping service</a>
can be used to obtain UniProt identifiers from other sequence
database identifiers.
</ul>
</body></html>
