<!--
--- UCSF Chimera Copyright ---
Copyright (c) 2009-2014 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
--- UCSF Chimera Copyright ---
-->

<html><head>
<title>Blast Protein</title>
</head><body>
<h3>Blast Protein <img src="blasticon.png" alt="Blast Protein icon"></h3>
<p>
<b>Blast Protein</b> performs protein sequence searches using 
a <a href="#notes">BLAST</a> web service hosted by the
<a href="https://www.cgl.ucsf.edu/Overview/index.html" target="_blank">UCSF
Resource for Biocomputing, Visualization, and Informatics (RBVI)</a>.
Corresponding structures can be retrieved and automatically superimposed,
and the <a href="#notes">pseudo-multiple alignment</a> 
from BLAST can be shown in
<a href="../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a>.
The search results are included in saved <a href="sessions.html">sessions</a>.
See also: <a href="midas/mda.html"><b>mda</b></a>
</p><p>
There are <a href="extension.html">several ways to start</a>
<b>Blast Protein</b>, a tool in the <b>Sequence</b> category.
When it is started from the 
<a href="../ContributedSoftware/multalignviewer/multalignviewer.html#mavmenu-tools"><b>Info</b> menu</a> in
<a href="../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a>,
any of the sequences in the alignment can be specified as the query.
Otherwise, the query sequence can be:
<ul>
<li> <b>From Structure</b>
- a protein chain from a structure open in Chimera
<li> <a name="addplain"><b>Plain Text</b></a> 
- a protein sequence entered or pasted in the <b>Sequence</b> field 
as plain text
</ul>
Search parameters:
<ul>
<li><b>Program</b>
  <ul>
  <li><b>blast</b> (default) - protein-protein search 
  <li><b>psiblast</b> - iterated protein-protein search, where each pass 
  after the first uses a PSSM (position-specific scoring matrix) 
  based on the results of the previous pass
  </ul>
<li><b>Database</b>
  <ul>
  <li><b>pdb</b> (default) - sequences of structures in the
  <a href="https://www.rcsb.org" target="_blank">Protein Data Bank (PDB)</a>.
  The <b>List only...</b> option
  controls whether multiple hits with the same PDB identifier (<i>i.e.</i>
  different chains) or only the best-matching one should be included in the results.
  <li><b>nr</b> - all non-redundant GenBank CDS translations + 
  RefSeq Proteins + PDB + SwissProt + PIR + PRF.  This database is much
  larger than <b>pdb</b> and will take longer to search.
  </ul>
<li><b>E-value (1e-X)</b> - significance cutoff; 
  only matches with E-values &le; 10<sup>-X</sup> will be returned 
  (X=<b>3</b> by default) 
<li><b>Matrix</b> (BLOSUM45/<b>BLOSUM62</b>/BLOSUM80/PAM30/PAM70) 
  - amino acid substitution matrix to use for alignment scoring
<li><b>Passes</b> (default <b>1</b>) 
  - number of <b>psiblast</b> iterations; 1 pass is equivalent to <b>blast</b>,
  whereas multiple passes will find more distantly related sequences
<li><b>List only best-matching chain per PDB entry</b> (default)
- whether multiple hits with the same PDB identifier 
  (matches to different chains in that PDB entry) or only 
  the best-matching one should be included in the results
</ul>
Clicking <b>OK</b> launches the search and dismisses the dialog.
<b>Apply</b> launches the search without dismissing the dialog,
<b>Cancel</b> simply dismisses the dialog, 
and <b>Help</b> opens this manual page in a browser window.
</p><p>
The search is a background task.  
Clicking the information icon <img src="taskicon.png"> in the Chimera
<a href="chimerawindow.html#statusline">status line</a>
will bring up the
<a href="../ContributedSoftware/idle/task.html"><b>Task Panel</b></a>,
in which the job can be canceled if desired.
</p>

<a name="results">
<h3>Blast Results</h3></a>
<p>
Results are returned as a table of sequences with several columns of 
information. Which columns are displayed can be set in the
<a href="#configure"><b>Column</b> configuration</a> panel.
</p><p>
Some columns of information are available for <b>nr</b> sequences 
regardless of whether they are in the <b>pdb</b> subset:
</p>
<ul>
<li><b>GI</b> - sequence GI number
  (for the query, <b>model name</b> is shown instead) 
<li><b>Evalue</b> - significance value
<li><b>Score</b> - alignment score
<li><b>Description</b> - text description (truncated)
</ul>
<p>
Additional columns are available for <b>pdb</b> sequences:
</p>
<ul>
<li><b>PDB</b> - corresponding PDB identifier, including chain identifier
<li><b>Authors</b> - structure authors
<li><b>Chain names</b> 
- chain identifiers and descriptions of polymer chains in the structure
<li><b>Copies</b> - number of copies of the hit chain in the structure
<li><b>Date</b> - structure deposition date
<li><b>Ligand formulas</b> - chemical formulae of ligand chemical components
<li><b>Ligand names</b> - names of ligand chemical components
<li><b>Ligand smiles</b> - 
<a href="https://www.daylight.com/dayhtml/doc/theory/theory.smiles.html"
target="_blank"><b>SMILES</b></a> strings of ligand chemical components
<li><b>Ligand symbols</b> - residue names of ligand chemical components
<li><b>Ligand weights</b> - molecular weights of ligand chemical components
<li><b>Method</b> - method of structure determination
<li><b>Polymers</b> - number of different polymer chains in the structure
(not counting multiple copies of the same sequence)
<li><b>PubMed</b> - <a href="https://www.ncbi.nlm.nih.gov/pubmed/"
target="_blank">PubMed</a> identifier of literature reference, if any
<li><b>Residues</b> - number of residues in the hit chain
<li><b>Resolution</b> - crystallographic resolution
<li><b>Species</b> - scientific name of source organism
<li><b>Title</b> - structure title
<li><b>Total atoms</b> 
- total number of atoms in the structure (all chains)
<li><b>Total residues</b> 
- total number of residues in the structure (all chains)
<li><b>UniProt</b> 
- <a href="https://www.uniprot.org/" target="_blank">UniProt</a> identifier,
if any, for hit chain
<li><b>Weight</b> - molecular weight of hit chain
</ul>
<p>
Other than <b>PDB</b> identifier, the information in these columns is
retrieved using a web service provided by the 
<a href="https://www.rcsb.org" target="_blank">RCSB PDB</a>.
</p><p>
One or more of the hits (lines in the table) can be chosen 
with the left mouse button.
A block of lines can be chosen by dragging, or by
clicking on the first (or last) line in the desired block
and then <b>Shift</b>-clicking on its last (or first) line.
<b>Ctrl</b>-click toggles the state (chosen or not) of single line.
</p><p>
Buttons at the bottom of the results dialog:
</p>
<ul>
<li><b>Show in MAV</b> - display the 
<a href="#notes">pseudo-multiple alignment</a> 
of the chosen sequences in 
<a href="../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a>
<li><b>Load Structure</b> 
- retrieve and open the PDB structures corresponding
to the chosen sequences (as described for
<a href="fetch.html"><b>Fetch by ID</b></a>); they will be superimposed
automatically according to the <a href="#notes">pseudo-multiple alignment</a>.
Entire PDB entries are retrieved, not just the matching chains.
<li><b>PDB Web Site</b> 
- show 
<a href="https://www.rcsb.org" target="_blank">RCSB PDB</a>
entries for the chosen sequences in a browser window
<li><a name="configure"><b>Columns</b></a>
- show/hide the <b>Column</b> configuration panel, which contains checkboxes 
to set which <a href="#results">columns</a> are displayed. 
A maximum column width (default <b>40</b> characters) can be specified.
Buttons within the panel: 
  <ul>
  <li><b>All</b> - show all columns in the table
  <li><b>Default</b> - restore the previously saved configuration
  <li><b>Standard</b> - use the factory default configuration
  <li><b>Set Default</b> - save the current configuration to the Chimera
  <a href="preferences.html#preffile"><b>preferences</b> file</a>
  </ul>
<li><b>Export...</b> save the chosen rows, in the current sort order 
and including the currently displayed <a href="#results">columns</a>,
as a file in CSV (comma-separated values) format. 
If no rows are chosen, all will be exported. Note that some columns 
may contain commas or newline characters (multiple lines), but these 
are handled according to CSV standard by enclosing the values in quotes.
Spreadsheet software and other programs built to work with
the CSV standard will interpret the exported files correctly, 
but user scripts need to account for the potential complexities,
and viewing the file contents directly may be confusing.
<li><b>Hide</b> - hide the table without deleting the results;
it can be shown again using the <b>Raise</b> option of its 
<a href="instances.html">instance in the <b>Tools</b> menu</a>.
This is also useful when the table has become obscured by other windows.
<li><b>Quit</b> - delete the results
<li><b>Help</b> - show this manual page in a browser window
</ul>

<a name="notes">
<h3>Notes</h3></a>
<p>
<b>Basic Local Alignment Search Tool (BLAST)</b>.
The <a href="https://blast.ncbi.nlm.nih.gov/" target="_blank">BLAST</a>
software is provided by the 
<a href="https://www.ncbi.nlm.nih.gov/" target="_blank">NCBI</a>
and described in the following:
<blockquote>
<a href="https://nar.oxfordjournals.org/content/25/17/3389.full"
target="_blank">Gapped BLAST and PSI-BLAST: 
a new generation of protein database search programs.</a>
Altschul SF, Madden TL, Sch&auml;ffer AA, Zhang J, Zhang Z, Miller W, Lipman DJ.
<i>Nucleic Acids Res.</i> 1997 Sep 1;25(17):3389-402.
<br><br>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/2231712" target="_blank">
Basic local alignment search tool.</a>
Altschul SF, Gish W, Miller W, Myers EW, Lipman DJ.
<i>J Mol Biol.</i> 1990 Oct 5;215(3):403-10.
</blockquote>
</p><p>
<b>Pseudo-multiple alignment</b>.
The pseudo-multiple alignment from BLAST is not a true multiple alignment,
but a consolidation of the pairwise alignments of individual hits to the
query.  This output corresponds to the BLAST formatting option
(<a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp#alignment_view"
target="_blank">alignment view</a>)
<!-- 
better descriptions 
<a href="https://www.ebi.ac.uk/Tools/sss/ncbiblast/help/index-vectors.html#align"
target="_blank">here</a>
-->
"flat query-anchored without identities,"
in which identities and conservative changes relative to the query
are shown in upper case, nonconservative changes in lower case.
</p>
<hr>
<address>UCSF Computer Graphics Laboratory / May 2012</address>
</body></html>
