<!--
--- UCSF Chimera Copyright ---
Copyright (c) 2002-2020 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
--- UCSF Chimera Copyright ---
-->

<html><head>
<title>Input File Types</title>
</head><body>
<h2>Input File Types</h2>
<p>
Ways to open registered file types in Chimera:
<ul>
<li> with <a href="opensave.html#opendialog"><b>File... Open</b></a>
(local files), <a href="fetch.html"><b>Fetch by ID</b></a> 
(files to be fetched from databases)
<li> from the Chimera 
<a href="chimerawindow.html#emulator"><b>Command Line</b></a>,
using the command <a href="midas/open.html"><b>open</b></a> (default type 
<a href="tutorials/framepdbintro.html" target="_top">PDB</a>)
<li> from the system command line at the time of 
<a href="startup.html">Chimera startup</a>
</ul>
In the Chimera and system command lines, the file type can be specified by a 
<b><i><a name="suffix">suffix</a></i></b> that is part of the filename or by 
<b><i><a name="prefix">prefix</a>:filename</i></b>,
where <b><i>prefix</i></b> is not part of the filename.
Suffixes (filename extensions) also control which files are listed
in the <a href="opensave.html#opendialog"><b>Open File</b> dialog</a>
when the <b>File type</b> is set to a specific type.
<p><p>
The file types are grouped by data type below:
<ul>
<li> <a href="#molstruct">Molecular Structures and Related Data</a>
(see also
<a href="../ContributedSoftware/movie/framemovie.html" target="_top"><b>MD 
Movie</b></a> and 
<a href="../ContributedSoftware/viewdock/framevd.html" 
target="_top"><b>ViewDock</b></a>)
<li> <a href="#esp">Electrostatic Potential</a> (associated tool:
<a href="../ContributedSoftware/surfcolor/surfcolor.html"><b>Electrostatic
Surface Coloring</b></a>)
<li> <a href="#volume">Volume Data</a> (associated tool:
<a href="../ContributedSoftware/volumeviewer/framevolumeviewer.html"
target="_top"><b>Volume Viewer</b></a>)
<li> <a href="#alignment">Sequence</a> (associated tool:
<a href="../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a>)
<li> <a href="#command">Command Scripts</a>
<li> <a href="#object">Other 3D Objects</a>
</ul>
<a name="associated">
When a tool-specific type of file is opened, 
the associated tool will execute or start.</a>
</p>

<table border width="98%" cellpadding="2" cellspacing="0">
<caption><a name="molstruct"><b>Molecular Structures 
and Related Data</b></a></caption>
<tr><th>file type</th>
<th><a href="#prefix">prefix</a></th>
<th><a href="#suffix">suffix</a></th>
<th>contents</th>
</tr><tr>
<td align="center"><a href="castp.html">CASTp</a> ID fetch</td>
<td align="center">CASTp:<br>castp:</td>
<td align="center"></td>
<td align="center">structure and pocket measurements
<br>(data to fetch from the
from the <a href="http://sts.bioe.uic.edu/castp/"
target="_blank">Computed Atlas of Surface Topography of proteins (CASTp) 
database</a> specified by 4-character PDB ID
with chain ID optionally appended, for example <b>2gsh.A</b>; 
not all PDB entries are in the database;
measurements will be displayed in a <a href="castp.html#pocketlist">pocket list</a>)</td>
</tr><tr>
<td align="center"><a href="castp.html">CASTp</a> local</td>
<td align="center"></td>
<td align="center">.poc</td>
<td align="center">structure and pocket measurements
<br>(previously saved results from the
<a href="http://sts.bioe.uic.edu/castp/calculation.html"
target="_blank">CASTp server</a> specified by <i>name</i>.poc, 
which will also read <i>name</i>.pdb, <i>name</i>.pocInfo, <i>name</i>.mouth, 
and <i>name</i>.mouthInfo from the same location and display
the measurements in a <a href="castp.html#pocketlist">pocket list</a>)</td>
</tr><tr>
<td align="center"><a href="https://mmcif.wwpdb.org/docs/faqs/pdbx-mmcif-faq-general.html"
target="_blank">mmCIF</a></td>
<td align="center">cif:<br>mmcif:</td>
<td align="center">.cif</td>
<td align="center">molecular structure</td>
</tr><tr>
<td align="center">CIFID</td>
<td align="center">cifID:</td>
<td align="center"></td>
<td align="center">molecular structure
<br>(mmCIF format; file to fetch specified by 4-character PDB ID)</td>
</tr><tr>
<td align="center"><a href="evalscore.html">EvalScore<br>
summary file</a></td>
<td align="center">score:</td>
<td align="center">.score</td>
<td align="center">
model scores and associated filenames from protein structure prediction
(used in CASP assessment)</td>
</tr><tr>
<td align="center"><a href="gaussian.html">Gaussian<br>
formatted checkpoint</a></td>
<td align="center">fchk:<br>gaussian:</td>
<td align="center">.fchk</td>
<td align="center">molecular structure</td>
</tr><tr>
<td align="center">GRASP surface</td>
<td align="center">graspsurf:</td>
<td align="center">.srf</td>
<td align="center">molecular surface generated by
<a href="http://compbio.clemson.edu/delphi" target="_blank">DelPhi</a>
(the academic version; see also
<a href="../ContributedSoftware/delphicontroller/delphicontroller.html">
<b>DelPhiController</b></a>) or
<a href="https://honig.c2b2.columbia.edu/grasp/" target="_blank">GRASP</a>
</td>
</tr><tr>
<td align="center"><a href="http://gromos.net/"
target="_blank">Gromos87</a></td>
<td align="center">gro:</td>
<td align="center">.gro</td>
<td align="center">molecular structure
<br>(not read as a trajectory)</td>
</tr><tr>
<td align="center"><a href="https://www.schrodinger.com/maestro"
target="_blank">Maestro</a> file</td>
<td align="center">maestro:<br>mae:</td>
<td align="center">.mae</td>
<td align="center">molecular structure</td>
</tr><tr>
<td align="center">
<a href="../ContributedSoftware/movie/movie.html#metafile">MD Movie 
metafile</a></td>
<td align="center">md:<br>movie:</td>
<td align="center"></td>
<td align="center">specification of 
<a href="../ContributedSoftware/movie/movie.html#movieinput">trajectory format</a>
and filenames
<br>(<a href="#associated">associated tool</a>: 
<a href="../ContributedSoftware/movie/framemovie.html" target="_top"><b>MD 
Movie</b></a>)</td>
</tr><tr>
<td align="center">
<a href="https://en.wikipedia.org/wiki/Chemical_table_file"
target="_blank">MDL MOL/SDF</a></td>
<td align="center">mol:<br>sdf:</td>
<td align="center">.mol<br>.sdf</td>
<td align="center">molecular structure</td>
</tr><tr>
<td align="center"><a href="modbase.html">ModBase</a></td>
<td align="center">modbase:</td>
<td align="center"></td>
<td align="center">modeled protein structure
<br>(comparative models in PDB format to fetch from 
<a href="https://modbase.compbio.ucsf.edu/modbase-cgi/index.cgi"
target="_blank">ModBase</a> specified by
SwissProt, TrEMBL, GenPept or PIR accession code; associated information
will be shown in a <a href="modbase.html">list</a>)</td>
</tr><tr>
<td align="center">Mol2</td>
<td align="center">mol2:</td>
<td align="center">.mol2</td>
<td align="center">molecular structure</td>
</tr><tr>
<td align="center"><a href="dms.html">MS surface</a></td>
<td align="center">dms:<br>ms:</td>
<td align="center">.dms<br>.ms</td>
<td align="center">dot molecular surface</td>
</tr><tr>
<td align="center">NDB</td>
<td align="center">ndb:</td>
<td align="center"></td>
<td align="center">nucleic acid structure
<br>(PDB format; file to fetch specified by
<a href="http://ndbserver.rutgers.edu/index.html" target="_blank">NDB</a> 
ID)</td>
</tr><tr>
<td align="center"><a href="tutorials/framepdbintro.html"
target="_top">PDB</a>
<br>(see also <a href="tutorials/pdbintro.html#PQR">PQR</a>)</td>
<td align="center">pdb:</td>
<td align="center">.pdb<br>.pdb1<br>.ent<br>.pqr</td>
<td align="center">molecular structure</td>
</tr><tr>
<td align="center">PDBID</td>
<td align="center">pdbID:</td>
<td align="center"></td>
<td align="center">molecular structure
<br>(PDB format; file to fetch specified by 4-character PDB ID)</td>
</tr><tr>
<td align="center">PDB biounit</td>
<td align="center">biounitID:</td>
<td align="center"></td>
<td align="center">known or predicted
<a href="https://pdb101.rcsb.org/learn/guide-to-understanding-pdb-data/biological-assemblies#Anchor-BioUnit"
target="_blank">biological assembly</a>
<br>
(file(s) to fetch specified by 4-character PDB ID, optionally with 
&ldquo;.N&rdquo; appended to specify assembly N; otherwise, if the entry 
has multiple assemblies, multiple files will be retrieved)</td>
</tr><tr>
<td align="center">PQSID</td>
<td align="center">pqsID:</td>
<td align="center"></td>
<td align="center">predicted biological unit
<br>(file to fetch from the
from the <a href="http://pqs.ebi.ac.uk/pqs-doc.shtml" target="_blank">Protein
Quaternary Structure server</a> specified by 4-character PDB ID;
predictions are not available for all PDB entries; some entries have 
multiple predictions, and for those, multiple files will be retrieved)</td>
</tr><tr>
<td align="center">
<a href="https://pubchem.ncbi.nlm.nih.gov/" target="_blank">PubChem</a>
CID (input)</td> 
<td align="center">PubChem:<br>pubchem:</td>
<td align="center"></td>
<td align="center">small molecule structure
<br>(3D structure SDF fetched from the
<a href="https://pubchemdocs.ncbi.nlm.nih.gov/pubchem3d"
target="_blank"><b>PubChem3D</b></a> database of modeled structures)
</td>
</tr><tr>
<td align="center"><a href="https://integrativemodeling.org/rmf/format.html"
target="_blank">Rich Molecular Format</a>
<br>(based on the hierarchical data format
<a href="https://www.hdfgroup.org/HDF5/" target="_blank">HDF5</a>)</td>
<td align="center">rmf:</td>
<td align="center">.rmf<br>.rmf2info</td>
<td align="center">hierarchical molecular structure, feature information,
geometric markup<br>(associated data and controls will be shown in an
<a href="rmfviewer.html"><b>RMF Viewer</b></a> dialog)</td>
</tr><tr>
<td align="center">SCOP</td>
<td align="center">scop:</td>
<td align="center"></td>
<td align="center">protein domain structure
<br>(PDB format; file to fetch specified by 
<a href="domains.html#scop">SCOP domain ID</a>)</td>
</tr><tr>
<td align="center">
<a href="https://www.daylight.com/dayhtml/doc/theory/theory.smiles.html"
target="_blank">SMILES</a> (input)</td> 
<td align="center">SMILES:<br>smiles:</td>
<td align="center"></td>
<td align="center">small molecule structure
<br>(SMILES string converted to 3D structure SDF using the NCI 
<a href="https://cactus.nci.nih.gov/translate/" target="_blank">SMILES
translator</a>, or if that does not produce a structure, the <i>smi23d</i> 
web service provided by the cheminformatics group at Indiana University)
</td>
</tr><tr>
<td align="center">VIPERID</td>
<td align="center">viperID:</td>
<td align="center"></td>
<td align="center">icosahedral virus capsid structure
<br>(PDB format; file to fetch specified by PDB ID;
capsid automatically constructed with
<a href="../ContributedSoftware/multiscale/framemulti.html"
target="_top"><b>Multiscale Models</b></a>)</td>
</tr><tr>
<td align="center">
<a href="xyz.html">XYZ coordinate</a></td>
<td align="center">xyz:</td>
<td align="center">.xyz</td>
<td align="center">molecular structure</td>
</tr>
</table>

<br><br>
<table border width="98%" cellpadding="2" cellspacing="0">
<caption><a name="esp"><b>Electrostatic Potential</b></a> 
(can also be handled as <a href="#volume">volume data</a>)
<br><a href="#associated">Associated tool</a>:
<a href="../ContributedSoftware/surfcolor/surfcolor.html"><b>Electrostatic
Surface Coloring</b></a></caption>
<tr><th>file type</th>
<th><a href="#prefix">prefix</a></th>
<th><a href="#suffix">suffix</a></th>
<th>contents</th>
</tr><tr>
<td align="center">APBS potential</td>
<td align="center">apbs:</td>
<td align="center">.dx</td>
<td align="center">
<a href="https://apbs.readthedocs.io/en/latest/formats/opendx.html"
target="_blank">electrostatic potential map</a> from
<a href="https://www.poissonboltzmann.org/" target="_blank">Adaptive 
Poisson-Boltzmann Solver</a> (APBS); 
<br>see also the 
<a href="../ContributedSoftware/apbs/apbs.html"><b>APBS</b></a> tool in Chimera
</td>
</tr><tr>
<td align="center">DelPhi or GRASP potential</td>
<td align="center">delphi:</td>
<td align="center">.phi</td>
<td align="center">electrostatic potential map from 
<a href="http://compbio.clemson.edu/delphi" target="_blank">DelPhi</a>
(the academic version; see also
<a href="../ContributedSoftware/delphicontroller/delphicontroller.html">
<b>DelPhiController</b></a>) or
<a href="https://honig.c2b2.columbia.edu/grasp/" target="_blank">GRASP</a>
</td>
</tr><tr>
<td align="center">UHBD grid, binary</td>
<td align="center">uhbd:</td>
<td align="center">.grd</td>
<td align="center">electrostatic potential map from
University of Houston Brownian Dynamics (UHBD)</td>
</tr></table>
<sup>*</sup>Electrostatic potential maps can also be generated by using
<a href="../ContributedSoftware/coulombic/coulombic.html"><b>Coulombic 
Surface Coloring</b></a> with the
<a href="../ContributedSoftware/coulombic/coulombic.html#computegrid">compute
grid</a> option
<br>
<sup>**</sup>Electrostatic potential isosurfaces are not displayed
automatically, but can be shown with
<a href="../ContributedSoftware/volumeviewer/framevolumeviewer.html"
target="_top"><b>Volume Viewer</b></a>
or the <a href="midas/volume.html"><b>volume</b></a> command

<br><br>
<table border width="98%" cellpadding="2" cellspacing="0">
<caption><a name="volume"><b>Volume Data</b></a>
(see also <a href="#esp">electrostatic potential</a>)
<br><a href="#associated">Associated tool</a>:
<a href="../ContributedSoftware/volumeviewer/framevolumeviewer.html"
target="_top"><b>Volume Viewer</b></a></caption>
<tr><th>file type</th>
<th><a href="#prefix">prefix</a></th>
<th><a href="#suffix">suffix</a></th>
<th>contents</th>
</tr><tr>
<td align="center">
<a href="https://www.fei.com/software/amira-for-life-sciences/"
target="_blank">Amira</a> mesh</td>
<td align="center">amira:</td>
<td align="center">.am</td>
<td align="center">3D data</td>
</tr><tr>
<td align="center">BRIX or DSN6 map</td>
<td align="center">dsn6:</td>
<td align="center">.brix<br>.omap</td>
<td align="center">electron density map</td>
</tr><tr>
<td align="center">
<a href="https://en.wikipedia.org/wiki/CCP4_(file_format)" 
target="_blank">CCP4</a> map</td>
<td align="center">ccp4:</td>
<td align="center">.ccp4<br>.map</td>
<td align="center">electron density map</td>
</tr><tr>
<td align="center"><a href="midas/volume.html#chimap">Chimera map</a></td>
<td align="center">cmap:</td>
<td align="center">.cmap<br>.cmp</td>
<td align="center">volume data from Chimera/ChimeraX
<br>(based on the hierarchical data format
<a href="https://www.hdfgroup.org/HDF5/" target="_blank">HDF5</a>)</td>
</tr><tr>
<td align="center">CNS or XPLOR map</td>
<td align="center">xplor:</td>
<td align="center">.cns<br>.xplor</td>
<td align="center">unformatted ASCII density map</td>
</tr><tr>
<td align="center">DOCK scoring grid</td>
<td align="center">dock:</td>
<td align="center">.bmp<br>.cnt<br>.nrg</td>
<td align="center">
<a href="http://dock.compbio.ucsf.edu/index.html"
target="_blank">DOCK</a> (versions 4, 5, 6) 
bump, contact, and energy scoring grids<br>
(suffixes not interchangeable; <i>gridname</i>.bmp required for reading
<i>gridname</i>.cnt and/or <i>gridname</i>.nrg)</td>
</tr><tr>
<td align="center">EDSID</td>
<td align="center">edsID:</td>
<td align="center"></td>
<td align="center">electron density map
<br>(file to fetch from the
<a href="https://eds.bmc.uu.se/eds/index.html" target="_blank">Electron 
Density Server</a> specified by 4-character PDB ID;
not all PDB entries have maps available)</td>
</tr><tr>
<td align="center">EDSDIFFID</td>
<td align="center">edsdiffID:</td>
<td align="center"></td>
<td align="center">electron density difference map (fo-fc)
<br>(file to fetch from the
<a href="https://www.ebi.ac.uk/pdbe/eds/" target="_blank">Electron 
Density Server</a> specified by 4-character PDB ID;
not all PDB entries have maps available)</td>
</tr><tr>
<td align="center">EMAN HDF map</td>
<td align="center">emanhdf:</td>
<td align="center">.hdf<br>.h5</td>
<td align="center">electron density map
<br>(based on the hierarchical data format
<a href="https://www.hdfgroup.org/HDF5/" target="_blank">HDF5</a>)</td>
</tr><tr>
<td align="center">EMDBID</td>
<td align="center">emdbID:</td>
<td align="center"></td>
<td align="center">electron density map
<br>(entry to fetch from the
<a href="https://www.ebi.ac.uk/pdbe/emdb/" target="_blank">Electron 
Microscopy Data Bank</a>; prefix emdbfitID will additionally fetch
any corresponding PDB entries, but they may or may not be in the fit
positions relative to the map)</td>
</tr><tr>
<td align="center">Gaussian cube grid</td>
<td align="center">cube:</td>
<td align="center">.cube<br>.cub</td>
<td align="center">orbitals, electron densities, other</td>
</tr><tr>
<td align="center"><a href="https://research.csc.fi/-/gopenmol_download"
target="_blank">gOpenMol</a> grid</td>
<td align="center">gopenmol:</td>
<td align="center">.plt</td>
<td align="center">orbitals, electron densities, other</td>
</tr><tr>
<td align="center">
<a href="../ContributedSoftware/volumeviewer/imagestack.html">
Image stack</a></td>
<td align="center">images:</td>
<td align="center">.tif<br>.tiff<br>.png<br>(<i>etc.</i>)</td>
<td align="center">image stack
<br>(grayscale image series in multiple files or a single multipage file,
in formats supported by
<a href="https://en.wikipedia.org/wiki/Python_Imaging_Library"
target="_blank">PIL</a>)</td>
</tr><tr>
<td align="center">
<a href="https://www.imagescience.de/formats.html"
target="_blank">IMAGIC map</a>
<br>(file reader courtesy of Ralf Schmidt and Michael Schatz, 
Image Science Software)</td>
<td align="center">imagic:</td>
<td align="center">.hed<br>.img</td>
<td align="center">density map
<br>(the format consists of two files, <i>filename</i>.hed with
header information and <i>filename</i>.img with the map values;
<br>the format allows for multiple maps, but only the first will be read)</td>
</tr><tr>
<td align="center">
<a href="https://bio3d.colorado.edu/imod/doc/mrc_format.txt"
target="_blank">IMOD</a> map</td>
<td align="center">imodmap:</td>
<td align="center">.rec</td>
<td align="center">electron density map
<br>(MRC map with signed 8-bit mode interpreted as unsigned)</td>
</tr><tr>
<td align="center">
<a href="https://brettbode.github.io/wxmacmolplt/MacMolPlt_Manual.html"
target="_blank">MacMolPlt</a> grid</td>
<td align="center">macmolplt:</td>
<td align="center">.mmp</td>
<td align="center">
<a href="https://brettbode.github.io/wxmacmolplt/Manual_pages/MacMolPlt_Files.html#SurfExport"
target="_blank">3D surfaces</a></td>
</tr><tr>
<td align="center">
<a href="https://www2.mrc-lmb.cam.ac.uk/research/locally-developed-software/image-processing-software/#image" 
target="_blank">MRC</a> map</td>
<td align="center">mrc:</td>
<td align="center">.mrc</td>
<td align="center">electron density map</td>
</tr><tr>
<td align="center"><a href="https://www.unidata.ucar.edu/software/netcdf/"
target="_blank">NetCDF</a> generic array</td>
<td align="center">netcdf:</td>
<td align="center">.nc</td>
<td align="center">3D data</td>
</tr><tr>
<td align="center">Priism data</td>
<td align="center">priism:</td>
<td align="center">.xyzw</td>
<td align="center">3D light or EM data</td>
</tr><tr>
<td align="center">Priism time series</td>
<td align="center">priism_t:</td>
<td align="center">.xyzt</td>
<td align="center"><a name="volumetime">time series of 3D light or EM data</a>
<br>(also starts <a href="../ContributedSoftware/volseries/volseries.html">
<b>Volume Series</b></a>)</td>
</tr><tr>
<td align="center">PROFEC free energy grid</td>
<td align="center">profec:</td>
<td align="center">.profec</td>
<td align="center">interaction free energy grid from PROFEC
(in <a href="https://ambermd.org/" target="_blank"><b>Amber</b></a> 
versions 6 and 7)</td>
</tr><tr>
<td align="center">Purdue image format</td>
<td align="center">pif:</td>
<td align="center">.pif</td>
<td align="center">electron density map</td>
</tr><tr>
<td align="center"><a href="https://situs.biomachina.org/index.html" 
target="_blank">Situs</a> map</td>
<td align="center">situs:</td>
<td align="center">.situs<br>.sit</td>
<td align="center">electron density map</td>
</tr><tr>
<td align="center">
<a href="https://spider.wadsworth.org/spider_doc/spider/docs/spider.html"
target="_blank">SPIDER</a> map</td>
<td align="center">spider:</td>
<td align="center">.spi<br>.vol</td>
<td align="center">electron density map</td>
</tr><tr>
<td align="center"><a href="https://www.biochem.mpg.de/6348566/tom_e"
target="_blank">TOM toolbox</a> map</td>
<td align="center">tom_em:</td>
<td align="center">.em</td>
<td align="center">electron density map</td>
</tr><tr>
<td align="center">VTK PolyData</td>
<td align="center">vtk:</td>
<td align="center">.vtk</td>
<td align="center">VTK ascii structure points</td>
</tr>
</table>

<br><br>
<table border width="98%" cellpadding="2" cellspacing="0">
<caption><a name="alignment"><b>Sequence</b></a>
<br><a href="#associated">Associated tool</a>:
<a href="../ContributedSoftware/multalignviewer/framemav.html"
target="_top"><b>Multalign Viewer</b></a></caption>
<tr><th>file type</th>
<th><a href="#prefix">prefix</a></th>
<th><a href="#suffix">suffix</a></th>
<th>contents</th>
</tr><tr>
<td align="center">
<a href="../ContributedSoftware/multalignviewer/afasta.html">Aligned FASTA</a>
</td>
<td align="center">afasta:</td>
<td align="center">.afasta<br>.afa<br>.fasta<br>.fa</td>
<td align="center">sequence alignment</td>
</tr><tr>
<td align="center">
<a href="../ContributedSoftware/multalignviewer/pir.html">
Aligned NBRF/PIR</a></td>
<td align="center">pir:</td>
<td align="center">.ali<br>.pir</td>
<td align="center">sequence alignment</td>
</tr><tr>
<td align="center">
Clustal ALN</td>
<td align="center">aln:</td>
<td align="center">.aln<br>.clustal<br>.clustalw<br>.clustalx</td>
<td align="center">sequence alignment</td>
</tr><tr>
<td align="center">GCG RSF</td>
<td align="center">rsf:</td>
<td align="center">.rsf</td>
<td align="center">sequence alignment</td>
</tr><tr>
<td align="center">
<a href="https://nar.oxfordjournals.org/content/25/1/226.full"
target="_blank">HSSP</a><br>(read-only)
</td>
<td align="center">hssp:</td>
<td align="center">.hssp</td>
<td align="center">sequence alignment
<br>(other information not read; see
<a href="https://swift.cmbi.ru.nl/gv/hssp/" target="_blank">HSSP database</a>)
</td>
</tr><tr>
<td align="center">MSF</td>
<td align="center">msf:</td>
<td align="center">.msf</td>
<td align="center">sequence alignment</td>
</tr><tr>
<td align="center">Selex</td>
<td align="center">selex:<br>pfam:</td>
<td align="center">.selex<br>.slx<br>.pfam</td>
<td align="center">sequence alignment</td>
</tr><tr>
<td align="center">
<a href="https://en.wikipedia.org/wiki/Stockholm_format"
target="_blank">Stockholm</a></td>
<td align="center">sth:<br>hmmer:</td>
<td align="center">.sth<br>.sto</td>
<td align="center">sequence alignment</td>
</tr><tr>
<td align="center">
<a href="https://www.uniprot.org" target="_blank">UniProt</a> 
accession or ID (input)</td>
<td align="center">uniprot:</td>
<td align="center"></td>
<td align="center">protein sequence with feature annotations</td>
</tr>
</table>

<br><br>
<table border width="98%" cellpadding="2" cellspacing="0">
<caption><a name="command"><b>Command Scripts</b></a></caption>
<tr><th>file type</th>
<th><a href="#prefix">prefix</a></th>
<th><a href="#suffix">suffix</a></th>
<th>contents</th>
</tr><tr>
<td align="center"><a href="indexcommand.html#cmdfile">
Chimera commands</a></td>
<td align="center">com:<br>cmd:</td>
<td align="center">.com<br>.cmd</td>
<td align="center"><a href="indexcommand.html#cmdfile">Chimera commands</a>
<br>(locations of files opened by a command file can be specified relative 
to the command file's location, or with absolute pathnames)</td>
</tr><tr>
<td align="center">
<a href="../ContributedSoftware/demos/demos.html#demosrc">
Chimera demo</a></td>
<td align="center">demo:</td>
<td align="center">.src</td>
<td align="center">instructions to Chimera and
explanatory text for the demo viewer 
<br>(<a href="#associated">associated tool</a>:
<a href="../ContributedSoftware/demos/demos.html"><b>Demos</b></a>)</td>
</tr><tr>
<td align="center">
<a href="../ContributedSoftware/webdata/chimerax.html">Chimera web data</a></td>
<td align="center">chimerax:</td>
<td align="center">.chimerax</td>
<td align="center">instructions on data files to open,
commands and code to execute (can have a
<a href="../ContributedSoftware/demos/demos.html">Chimera demo</a> 
embedded)</td>
</tr><tr>
<td align="center">Python</td>
<td align="center">python:<br>py:<br>chimera:</td>
<td align="center">.py<br>.pyc<br>.pyo<br>.pyw</td>
<td align="center">Python code
<br>(locations of files opened by a Python script can be specified relative 
to the script's location, or with absolute pathnames;
<br>for scripts with arguments, 
<br>see also <b><A href="midas/runscript.html">runscript</A></b>,
startup option <b><A href="options.html#script">--script</A></b>)
</td>
</tr>
</table>

<br><br>
<table border width="98%" cellpadding="2" cellspacing="0">
<caption><a name="object"><b>Other 3D Objects</b></a></caption>
<tr><th>file type</th>
<th><a href="#prefix">prefix</a></th>
<th><a href="#suffix">suffix</a></th>
<th>contents</th>
</tr><tr>
<td align="center"><a href="bild.html">Bild</a></td>
<td align="center">bild:</td>
<td align="center">.bild<br>.bld</td>
<td align="center">graphical objects</td>
</tr><tr>
<td align="center"><a href="../ContributedSoftware/volumepathtracer/volumepathtracer.html#markerfiles">
Chimera markers</a></td>
<td align="center">markers:</td>
<td align="center">.cmm</td>
<td align="center">markers placed in 3D
<br>(<a href="#associated">associated tool</a>:
<a href="../ContributedSoftware/volumepathtracer/framevolpath.html"
target="_top"><b>Volume Tracer</b></a>)</td>
</tr><tr>
<td align="center"><a href="https://www.khronos.org/collada/"
target="_blank">COLLADA</a></td>
<td align="center">collada:</td>
<td align="center">.dae</td>
<td align="center">graphical objects<br>(geometry nodes define Chimera
<a href="surfattrib.html#surfpiecedef">surface pieces</a>)</td>
</tr><tr>
<td align="center">
<a href="../ContributedSoftware/volumeviewer/imod.html">IMOD 
segmentation</a></td>
<td align="center">imod:</td>
<td align="center">.imod<br>.mod</td>
<td align="center">EM segmentation meshes and contours</td>
</tr><tr>
<td align="center">Neuron trace
<br>(<a href="http://www.neuronland.org/NLMorphologyConverter/MorphologyFormats/SWC/Spec.html" target="_blank">SWC</a>, see also the
<a href="http://neuromorpho.org/myfaq.jsp#QS3"
target="_blank">NeuroMorpho FAQ</a>)</td>
<td align="center">swc:</td>
<td align="center">.swc</td>
<td align="center">neuron reconstruction, such as from 
<a href="http://neuromorpho.org/" target="_blank">NeuroMorpho.Org</a></td>
</tr><tr>
<td align="center">
<a href="../ContributedSoftware/segger/segment.html">Segger</a>
segmentation</td>
<td align="center">segger:</td>
<td align="center">.seg</td>
<td align="center">segmentation surfaces and grouping hierarchy,
pointer to corresponding <a href="#volume">volume data</a> file;
based on the hierarchical data format
<a href="https://www.hdfgroup.org/HDF5/" target="_blank">HDF5</a>
<br>(<a href="#associated">associated tool</a>:
<a href="../ContributedSoftware/segger/segment.html"><b>Segger</b></a>)</td>
</tr><tr>
<td align="center">
<a href="sphgen.html">Sphgen spheres</a></td> 
<td align="center">sph:</td>
<td align="center">.sph</td>
<td align="center">spheres from the 
<a href="http://dock.compbio.ucsf.edu/index.html"
target="_blank">DOCK</a> accessory program 
<a href="http://dock.compbio.ucsf.edu/DOCK_6/dock6_manual.htm#Sphgen"
target="_blank"><i>sphgen</i></a></td>
</tr><tr>
<td align="center"><a href="https://en.wikipedia.org/wiki/STL_%28file_format%29#Binary_STL"
target="_blank">STereoLithography</a> (binary)</td>
<td align="center">stl:</td>
<td align="center">.stl</td>
<td align="center">triangle-based format
native to stereolithography CAD software</td>
</tr><tr>
<td align="center">VIPERdb</td>
<td align="center">viper:</td>
<td align="center">.vdb</td>
<td align="center"><a name="viperdata">icosahedral virus capsid structure</a>
<br>(PDB format, in the
<a href="http://viperdb.scripps.edu/" target="_blank">Virus
Particle Explorer database</a> coordinate system; 
capsid automatically constructed with
<a href="../ContributedSoftware/multiscale/framemulti.html"
target="_top"><b>Multiscale Models</b></a>)</td>
</tr><tr>
<td align="center">VRML</td>
<td align="center">vrml:</td>
<td align="center">.vrml<br>.wrl</td>
<td align="center">graphical objects
<br>(described in VRML geometry nodes)</td>
</tr>
</table>

<hr>
<address>UCSF Computer Graphics Laboratory / March 2021</address>
</body></html>
