<!--
--- UCSF Chimera Copyright ---
Copyright (c) 2001-2018 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
--- UCSF Chimera Copyright ---
-->

<html><head>
<title>Commands Index</title>
</head><body>
<h2>Commands Index</h2>
<p>
Commands are entered into Chimera's
<a href="chimerawindow.html#emulator"><b>Command Line</b></a>.
Past commands can be accessed from the <a href="history.html">
<b>Command History</b></a>, and commands can be placed in an
executable <a href="#cmdfile">command file</a>.
Chimera commands are listed below and in the
<a href="quickref.pdf">Quick Reference (PDF)</a>.
</p><p>
All commands can be truncated to unique strings, but their keywords
cannot be truncated except as noted in the documentation.
Entering the contents of the
<a href="chimerawindow.html#emulator"><b>Command Line</b></a>
(pressing return) executes the contents and updates the display.
To hide intermediate stages of processing,
multiple commands can be combined into one line with semicolon separators.
Certain <a href="midas/usageconventions.html">text conventions</a> are used
to describe usage.
</p><p>
<A href="midas/2dlabels.html" target="Content">2dlabels</A>
- create labels with text, symbols, and arrows in 2D
<br>
<A href="midas/ac.html">ac</A>
- use accelerators (<a 
href="../ContributedSoftware/accelerators/alist.html">keyboard shortcuts</a>)
<br>
<A href="midas/addaa.html">addaa</A>
- add an amino acid to a peptide N- or C-terminus
<br>
<A href="midas/addcharge.html">addcharge</A>
- assign Amber partial charges and atom types
<br>
<A href="midas/addh.html">addh</A>
- add hydrogens
<br>
<A href="midas/adjust.html">adjust</A>
- change bond angle or bond length
<br>
<A href="midas/alias.html">alias</A>
- create an alias or list the existing aliases
<br>
<A href="midas/align.html">align</A>
- align two atoms or sets of atoms along the line of sight
<br>
<A href="midas/angle.html">angle</A>
- measure angles formed by atoms or by axes and planes
<br>
<A href="midas/aniso.html">aniso</A>
- show thermal ellipsoids
<br>
<A href="midas/apbs.html">apbs</A>
- interface to electrostatics calculations with
<a href="https://www.poissonboltzmann.org/" target="_blank">APBS</a>
(Adaptive Poisson-Boltzmann Solver)
<br>
<A href="midas/aromatic.html">aromatic</A>
- show ring aromaticity
<br>
<A href="midas/background.html">background</A>
- set background color, gradient, or image
<br>
<A href="midas/bond.html">bond</A>
- add/delete bonds
<br>
<A href="midas/bondcolor.html">bondcolor</A>
- color bonds independently from atoms
<br>
<A href="midas/bonddisplay.html">bonddisplay</A>
- control how bond display depends on atom display
<br>
<A href="midas/bondrepr.html">bondrepr</A>
- control bond style (wire or stick)
<br>
<A href="midas/bondzone.html">bondzone</A>
- make zoning tools use points along bonds
<br>
<A href="midas/cd.html">cd</A>
- change the working directory
<br>
<A href="midas/center.html">center</A>
- center the view on specified atoms
<br>
<A href="midas/chain.html">chain</A>
- chain specified atoms, undisplay the others
<br>
<A href="midas/changechains.html">changechains</A>
- reassign chain identifiers
<br>
<A href="midas/chirality.html">chirality</A>
- report the R/S configuration of a chiral center
<br>
<A href="midas/clip.html">clip</A>
- move global clipping planes
<br>
<A href="midas/open.html">close</A>
- close a model
<br>
<A href="midas/cofr.html">cofr</A>
- report or change the center of rotation
<br>
<A href="midas/color.html">color</A>
- color atoms/bonds, ribbons, labels, and surfaces
<br>
<A href="midas/colordef.html">colordef</A>
- define a new color
<br>
<A href="midas/colorkey.html">colorkey</A>
- create a color key
<br>
<a href="midas/combine.html">combine</a> 
- combine molecule models into a single model
or create another copy of a molecule model
<br>
<A href="midas/conic.html">conic</A>
- create a shadowed space-filling image
<br>
<A href="midas/coordset.html">coordset</A>
- play through frames of a trajectory
<br>
<A href="midas/copy.html">copy</A>
- save image files
<br>
<A href="midas/coulombic.html">coulombic</A>
- color surfaces by Coulombic electrostatic potential
<br>
<A href="midas/crystalcontacts.html">crystalcontacts</A>
- identify clashes between PDB symmetry copies
<br>
<A href="midas/defattr.html" target="Content">defattr</A>
- assign attribute values to atoms, residues, or models
<br>
<A href="midas/define.html">define</A>
- calculate axes, planes, centroids for sets of atoms
<br>
<A href="midas/delete.html">delete</A>
- delete atoms and bonds
<br>
<A href="midas/display.html">display</A>
- display and undisplay atoms
<br>
<A href="midas/distance.html">distance</A>
- measure distances between atoms, axes, planes, and/or centroids
<br>
<A href="midas/echo.html">echo</A>
- send text to the status line and Reply Log
<br>
<A href="midas/export.html">export</A>
- save the graphical scene 
<br>
<A href="midas/fillring.html">fillring</A>
- show rings as filled
<br>
<A href="midas/findclash.html">findclash</A>
- identify clashes and contacts
<br>
<A href="midas/findhbond.html">findhbond</A>
- identify hydrogen bonds
<br>
<A href="midas/fitmap.html">fitmap</A>
- fit atoms or map into map 
<br>
<A href="midas/fly.html">fly</A>
- smoothly traverse a series of saved positions
<br>
<A href="midas/focus.html">focus</A>
- adjust the view and center of rotation
<br>
<A href="midas/freeze.html">freeze</A>
- stop all motion
<br>
<A href="midas/getcrd.html">getcrd</A>
- report coordinates
<br>
<A href="midas/hbonds.html">hbonds</A>
- identify hydrogen bonds
<br>
<A href="midas/help.html">help</A>
- display the manual page for a command
<br>
<A href="midas/hkcage.html">hkcage</A>
- create a hexagon/pentagon mesh that covers an icosahedron
<br>
<A href="midas/intersurf.html">intersurf</A>
- generate and display interface surfaces
<br>
<A href="midas/invert.html">invert</A>
- swap substituents of an atom
<br>
<A href="midas/ksdssp.html">ksdssp</A>
- determine secondary structure from protein coordinates 
<br>
<A href="midas/label.html">label</A>
- display atom labels
<br>
<a href="midas/labelopt.html">labelopt</a>
- control the information in atom labels
<br>
<a href="midas/leap.html">leap</a>
- use a <a href="https://www.leapmotion.com/product" 
target="_blank">Leap Motion Controller</a> with Chimera
<br>
<a href="midas/lighting.html">lighting</a>
- adjust lighting and shininess
<br>
<a href="midas/linewidth.html">linewidth</a>
- control the width of wire bonds
<br>
<a href="midas/listen.html">list</a>
- report attribute information
<br>
<a href="midas/listen.html">listen</a>
- report Chimera status
<br>
<A href="midas/longbond.html">longbond</A>
- show/hide pseudobonds representing missing segments
<br>
<A href="midas/mask.html">mask</A>
- extract <a href="filetypes.html#volume">volume data</a> bounded by surfaces
<br>
<A href="midas/match.html">match</A>
- perform least-squares fitting of specified atoms
<br>
<A href="midas/matchmaker.html">matchmaker</A>
- align models in sequence, then in 3D
<br>
<A href="midas/matrixcopy.html">matrixcopy</A>
- apply the transformation matrix of one model to another
<br>
<A href="midas/matrixset.html">matrixget</A>
- write the current transformation matrices to a file
<br>
<A href="midas/matrixset.html">matrixset</A>
- read and apply transformation matrices from a file
<br>
<A href="midas/mclip.html">mclip</A>
- control <a href="../ContributedSoftware/per-model/per-model.html">per-model
clipping</a>
<br>
<A href="midas/mcopy.html">mcopy</A>
- copy settings from one molecule model to another
<br>
<A href="midas/mda.html">mda</A>
- MultiDomain Assembler: 
perform several steps toward homology-modeling a multidomain protein
<br>
<A href="midas/measure.html">measure</A>
- perform various calculations on structures, surfaces, maps
<br>
<A href="midas/meshmol.html">meshmol</A>
- create a &ldquo;molecule&rdquo; to show surface mesh as sticks
<br>
<A href="midas/minimize.html">minimize</A>
- energy-minimize structures
<br>
<A href="midas/mmaker.html">mmaker</A>
- align models in sequence, then in 3D
<br>
<a href="midas/modelcolor.html">modelcolor</a>
- set color at the model level 
<br>
<a href="midas/modeldisplay.html">modeldisplay</a>
- set display at the model level
<br>
<A href="midas/molmap.html">molmap</A>
- create a density map from atomic coordinates
<br>
<A href="midas/morph.html">morph</A>
- morph (interpolate) between different structures
<br>
<A href="midas/move.html">move</A>
- translate models
<br>
<A href="midas/movie.html">movie</A>
- capture image frames and assemble them into a movie file
<br>
<a href="midas/msc.html">msc</a>
- color <a href="../ContributedSoftware/multiscale/framemulti.html"
target="_top"><b>Multiscale Models</b></a> surfaces to match atoms 
<br>
<a href="midas/msms.html">msms</a>
- (see <a href="#surfstuff">surfcat/surfrepr</a>)
<br>
<A href="midas/namesel.html">namesel</A>
- save and name the current <a href="selection.html">selection</a>
<br>
<A href="midas/neon.html">neon</A>
- create a shadowed stick/tube image
(<b>not available on Windows systems</b>)
<br>
<A href="midas/nucleotides.html">nucleotides</A>
- create special nucleotide representations
<br>
<A href="midas/objdisplay.html">objdisplay</A>
- display and undisplay VRML models
<br>
<A href="midas/open.html">open</A>
- read local files or fetch by ID
<br>
<A href="midas/pause.html">pause</A>
- pause script execution until the user presses a key
<br>
<A href="midas/pdb2pqr.html">pdb2pqr</A>
- deprecated: use <a href="https://server.poissonboltzmann.org/"
target="_blank">PDB2PQR web service</a> directly instead
<br>
<A href="midas/pdbrun.html">pdbrun</A>
- send an annotated PDB file to the system shell 
<br>
<A href="midas/perframe.html">perframe</A>
- specify commands to be executed at each display frame
<br>
<A href="midas/pipes.html">pipes</A>
- create pipes-and-planks representations of proteins
<br>
<A href="midas/play.html">play</A>
- script various complex motions 
<br>
<A href="midas/preset.html">preset</A>
- apply a predefined combination of display settings
<br>
<A href="midas/rainbow.html">rainbow</A>
- color residues, chains, or models over a range
<br>
<A href="midas/ramachandran.html">ramachandran</A>
- show peptide &phi;,&psi; distribution
(<a href="../ContributedSoftware/ramachandran/ramachandran.html">Ramachandran plot</a>)
<br>
<A href="midas/rangecolor.html">rangecolor</A>
- color over a range according to attribute values
<br>
<A href="midas/read.html">read</A>
- execute a command file, updating the display at the end
<br>
<a href="midas/represent.html">represent</a>
- control atom/bond display style
  (wire, stick, ball-and-stick, or sphere)
<br>
<A href="midas/reset.html">reset</A>
- restore default or saved orientations
<br>
<A href="midas/resrenumber.html">resrenumber</A>
- reassign residue numbers
<br>
<a href="midas/ribbackbone.html">ribbackbone</a>
- allow display of both ribbon and backbone atoms
<br>
<a href="midas/ribbon.html">ribbon</a>
- display ribbon
<br>
<a href="midas/ribclass.html">ribclass</a>
- set ribbon residue class:
which atoms control ribbon path and which are hidden by ribbon
<br>
<a href="midas/ribcolor.html">ribcolor</a>
- color ribbons
<br>
<a href="midas/ribinsidecolor.html">ribinsidecolor</a>
- set a separate color for the insides of protein helix ribbons
<br>
<a href="midas/ribrepr.html">ribrepr</a>
- control ribbon style (flat, edged, or rounded)
<br>
<a href="midas/ribscale.html">ribscale</a>
- control ribbon secondary-structure-specific dimensions 
(Chimera default or licorice)
<br>
<a href="midas/ribspline.html">ribspline</a>
- control ribbon path method (B-spline or cardinal spline)
<br>
<A href="midas/rlabel.html">rlabel</A>
- display residue labels
<br>
<A href="midas/rmfalias.html">rmfalias</A>
- create command-line specifiers for sets of nodes in
<a href="rmfviewer.html"><b>RMF Viewer</b></a>
<br>
<A href="midas/rmsd.html">rmsd</A>
- evaluate the RMSD between specified sets of atoms
<br>
<A href="midas/rna.html">rna</A>
- build rough but potentially large-scale models of RNA and DNA
<br>
<A href="midas/rock.html">rock</A>
- rotate models back and forth (oscillate)
<br>
<A href="midas/roll.html">roll</A>
- rotate models
<br>
<A href="midas/rotation.html">rotation</A>
- rotate bonds
<br>
<A href="midas/runscript.html">runscript</A>
- run Python script with command-line arguments
<br>
<A href="midas/save.html">save</A>
- save the current Chimera session
<br>
<A href="midas/savepos.html">savepos</A>
- save model positions
<br>
<A href="midas/scale.html">scale</A>
- scale the view
<br>
<A href="midas/scene.html">scene</A>
- save and restore scenes (positions plus styles, colors, labels, <i>etc.</i>)
<br>
<A href="midas/scolor.html">scolor</A>
- color surfaces a single color or by 
<a href="filetypes.html#volume">volume data</a> or geometry
<br>
<A href="midas/section.html">section</A>
- move global clipping planes in parallel
<br>
<A href="midas/segment.html">segment</A>
- act on
<a href="../ContributedSoftware/segger/segment.html">segmentation</a> models
<br>
<A href="midas/select.html">select</A>
- <a href="selection.html">select</a> atoms or
<a href="mouse.html#activedef">activate</a> models for motion
<br>
<a href="midas/listen.html#sequence">sequence</a>
- show
<a href="../ContributedSoftware/multalignviewer/framemav.html" 
target="_top"><b>Sequence</b></a> for specified chains
<br>
<A href="midas/set.html">set/unset</A>
- set visual effects, make models rotate about individual centers
<br>
<A href="midas/setattr.html">setattr</A>
- set an attribute to a specified value
<br>
<A href="midas/shape.html">shape</A>
- create a surface of a specified geometric shape
<br>
<A href="midas/show.html">show</A>
- display specified atoms, undisplay the others
<br>
<A href="midas/sleep.html">sleep</A>
- pause script execution for a specified length of time
<br>
<a href="midas/solvate.html">solvate</a> - add solvent using 
<a href="https://ambermd.org/AmberTools.php" target="_blank">AmberTools</a>
<br>
<A href="midas/sop.html" target="Content">sop</A>
- adjust capping, edit surface models
<br>
<A href="midas/split.html">split</A>
- partition a molecule model into separate submodels
<br>
<a href="midas/start.html">start</a>
- start Chimera tools by name
<br>
<A href="midas/stereo.html">stereo</A>
- switch amongst stereo options and mono viewing
<br>
<A href="midas/stop.html">stop</A>
- exit from Chimera
<br>
<A href="midas/struts.html">struts</A>
- add pseudobonds to a molecule to strengthen it for 3D printing
<br>
<A href="midas/surface.html">surface</A>
- calculate and display molecular surfaces
<br>
<a href="midas/msms.html">surfcat/surfrepr</a>
<a name="surfstuff">
- create molecular surface categories and control surface style
</a>
<br>
<a href="midas/surfcolor.html">surfcolor</a>
- set molecular surface color source
<br>
<A href="midas/surftransparency.html">surftransparency</A>
- adjust surface transparency
<br>
<A href="midas/swapaa.html">swapaa</A>
- mutate amino acids or swap rotamers
<br>
<A href="midas/swapna.html">swapna</A>
- mutate nucleic acid residues
<br>
<A href="midas/sym.html">sym</A>
- generate symmetry-related copies of a structure
<br>
<A href="midas/system.html">system</A>
- send a command to the system shell
<br>
<a href="midas/tcolor.html">tcolor</a>
- color atoms/bonds, labels, and surfaces with a texture color
<br>
<a href="midas/texture.html">texture</a>
- define texture maps and associated colors
<br>
<A href="midas/thickness.html">thickness</A>
- move global clipping planes in opposite directions
<br>
<a href="midas/tile.html">tile</a>
- arrange models in a plane
<br>
<a href="midas/topography.html">topography</a>
- plot values in a volume data plane as heights in a surface
<br>
<a href="midas/transparency.html">transparency</a>
- make atoms/bonds, ribbons, and surfaces transparent
<br>
<A href="midas/turn.html">turn</A>
- rotate models
<br>
<A href="midas/vdw.html">vdw</A>
- display van der Waals (VDW) dot surface
<br>
<a href="midas/vdwdefine.html">vdwdefine</a>
- set VDW radii
<br>
<a href="midas/vdwdensity.html">vdwdensity</a>
- set VDW surface dot density
<br>
<A href="midas/version.html">version</A>
- show copyright information and Chimera version
<br>
<A href="midas/viewdock.html">viewdock</A>
- start <a href="../ContributedSoftware/viewdock/framevd.html" 
target="_top"><b>ViewDock</b></a> and load docking results
<br>
<A href="midas/vina.html">vina</A>
- requires installing Autodock Vina locally (Apr 2020)
<br>
<A href="midas/volume.html" target="Content">volume</A>
- display <a href="filetypes.html#volume">volume data</a>
(3D numerical data such as electron density)
<br>
<A href="midas/vop.html" target="Content">vop</A>
- edit <a href="filetypes.html#volume">volume data</a> 
<br>
<A href="midas/vseries.html" target="Content">vseries</A>
- display an ordered sequence of
<a href="filetypes.html#volume">volume data</a> sets;
can also be used to process and save the data
<br>
<A href="midas/wait.html">wait</A>
- suspend command processing a specified number of frames
or until motion has stopped
<br>
<A href="midas/window.html">window</A>
- adjust the view to contain the displayed and specified atoms
<br>
<A href="midas/windoworigin.html">windoworigin</A>
- set graphics window location
<br>
<A href="midas/windowsize.html">windowsize</A>
- adjust the dimensions of the graphics window
<br>
<A href="midas/write.html">write</A>
- save atomic coordinates (pdb, mol2)
<br>
<A href="midas/writesel.html">writesel</A>
- write a list of the currently 
<a href="selection.html">selected</a> (or unselected) items
<br>
<A href="midas/zonesel.html">zonesel</A>
- <a href="selection.html">select</a> atoms and/or surfaces within a cutoff
distance of specified atoms and/or surfaces
</p>

<h3><a name="cmdfile">Chimera Command Files</a></h3>
<p>
Chimera command files (scripts) are simply text files containing the
same commands that could be entered at the
<a href="chimerawindow.html#emulator"><b>Command Line</b></a>.
Only <b>plain text</b> is accepted,
not rich text format or Microsoft Word format.
Example files:
<ul>
<li><a href="../ContributedSoftware/defineattrib/rescol.com" 
type="text/plain">rescol.com</a> for coloring amino acid residues
<li><a href="tutorials/setup.com" type="text/plain">setup.com</a>
from the <a href="tutorials/vdtut.html">ViewDock tutorial</a>
<li><a href="tutorials/convergent.com" type="text/plain">convergent.com</a>
from the <a href="tutorials/convergent.html">Similar Binding Sites</a>
image tutorial
<li> <a href="movies.html#examples">movie content examples</a>
</ul>
Chimera command files can be created manually or by saving
the <a href="history.html"><b>Command History</b></a>.
<a name="cmdtips">
Rules and tips</a>:
<ul>
<li> Lines that begin with a pound sign (<b>#</b>) are interpreted as comments.
<li> Multiple commands separated by semicolons (<b>;</b>) can be combined
into a single line. The last semicolon in a line can precede either 
a command or a pound sign and in-line comment.
<li> During execution, a single-frame display update
(implicit <a href="midas/wait.html"><b>wait</b></a> <b>1</b>)
will be added at the end of each line that:
  <ul>
  <li> contains one or more commands that could change the display
  <br>&ndash; and &ndash;
  <li> does not already end with an explicit 
  <a href="midas/wait.html"><b>wait</b></a> of any length
  </ul>
(Exception: display updates are not added when a command file is executed
with <a href="midas/read.html"><b>read</b></a>.)
Multiple commands can be combined into a single line as described above
to suppress the display of intermediate states.
This allows multiple graphical changes to appear simultaneously
instead of one by one.
<li> Several of the <a href="movies.html#moviecommands">movie-related
commands</a> execute over multiple frames to generate gradual changes such
as continuous motion.  A multiple-frame command should be followed by 
a multiple-frame <a href="midas/wait.html"><b>wait</b></a>
(not the implicit single-frame wait mentioned above) 
to prevent subsequent commands from executing until it is finished.
<li> Even when multiple actions occur within a single image update frame,
their order is important, and one command may need to precede or follow
another to give the desired result. In general, the order of execution
is the order in which the commands are given.
For example, if a script plays a molecular dynamics trajectory and at each
frame shows a molecular surface and colors it by electrostatic potential,
the coloring should come after the surface calculation.
<li> The locations of files opened by a command file can be specified
relative to the command file's location, or with absolute pathnames.  
For example, if the files are in the same directory as the command file,
only their names need to be given.
<li> Structure files can be opened with <a href="midas/open.html"><b>open 
noprefs</b></a> to circumvent any
<a href="preferences.html#New Molecules"><b>New Molecules</b> preferences</a>;
this prevents inconsistent behavior of command files and
<a href="../ContributedSoftware/demos/demos.html">demos</a>
potentially caused by the different preferences settings of different users.
</ul>
Ways to execute a command file:
<ul>
<li>by simply <a href="filetypes.html">opening</a> the file,
which updates the display after each line as needed.
The <a href="filetypes.html">file type</a> is specified
with a suffix (.cmd or .com, part of the filename)
or prefix (cmd: or com:, not part of the filename).
<li>with the command <a href="midas/read.html"><b>read</b></a> 
<li>automatically at
<a href="chimerawindow.html#emulator"><b>Command Line</b></a> startup,
as specified in the 
<a href="preferences.html#Command Line"><b>Command Line</b> preferences</a>
</ul>
Command script execution can be aborted by pressing the <b>Esc</b> (escape) key,
or paused/resumed with <b>Shift</b>-<b>Esc</b>.  See also:
<A href="midas/pause.html"><b>pause</b></A>
</p><p>
A <b>midasrc</b> file is a command file that is executed automatically 
when the <a href="chimerawindow.html#emulator"><b>Command Line</b></a>
is started (see the <a href="preferences.html#Command Line"><b>Command Line</b> 
preferences</a>).
Placing <a href="midas/alias.html">aliases</a> and
<a href="midas/colordef.html">color definitions</a> in a <b>midasrc</b> file 
is a convenient way to apply them each time Chimera is used.
</p><p>
<a href="../ContributedSoftware/demos/demos.html#demodef">Chimera demos</a>
are largely
<a href="../ContributedSoftware/demos/demos.html#editor">constructed</a>
from commands.
</p><p>
Chimera commands do not provide a way to loop through multiple
residues or models, but this can be done by combining Chimera commands
with Python code or a shell script.  The Chimera Programmer's Guide includes
a <a href="../ProgrammersGuide/basicPrimer.html">primer on looping</a>.
</p>
<hr>
<address>UCSF Computer Graphics Laboratory / September 2018</address>
</body></html>
