<!--
--- UCSF Chimera Copyright ---
Copyright (c) 2004-2018 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
--- UCSF Chimera Copyright ---
-->

<html><head>
<title>Building Structures, Modifying and Saving Data</title>
</head><body>
<h2>Building Structures, Modifying and Saving Data</h2>
<p>
The state of Chimera can be saved to a
<a href="sessions.html">session</a> file and later restored.
</p>
<ul>
<li>Atomic coordinates can be modified or generated in Chimera
in many ways, including
 <ul>
 <li><a href="../ContributedSoftware/editing/editing.html#adjust">bond rotation</a>
 <li>amino acid <a href="../ContributedSoftware/rotamers/framerot.html"
 target="_top">mutation and rotamer swapping</a>
 <li><a href="../ContributedSoftware/addh/addh.html">adding hydrogens</a>
 <li><a href="../ContributedSoftware/editing/editing.html#deletion">deleting 
 atoms</a>
 <li><a href="../ContributedSoftware/editing/editing.html">structure
 building</a>
 <li><a href="../ContributedSoftware/multalignviewer/modeller.html">Modeller
 calculations</a>: comparative (homology) modeling, building missing segments,
 refinement
 <li><a href="../ContributedSoftware/minimize/minimize.html">minimization</a>
 <li>generating multimers (<a href="midas/sym.html"><b>sym</b></a>,
 <a href="../ContributedSoftware/unitcell/unitcell.html"><b>Unit Cell</b></a>,
 <a href="../ContributedSoftware/multiscale/framemulti.html" target="_top">
 <b>Multiscale Models</b></a>)
 <li>repositioning by
 <a href="superposition.html">superimposing structures</a> or
 <a href="../ContributedSoftware/fitmaps/fitmaps.html">fitting</a> into maps
 <li><a href="../ContributedSoftware/editing/renumber.html">renumbering
 residues</a> and
 <a href="../ContributedSoftware/editing/changechains.html">changing 
 chain IDs</a>
 </ul>
 and saved using the command <a href="midas/write.html"><b>write</b></a> 
 or the dialogs described below for 
 <a href="#pdb">PDB</a> and <a href="#mol2">Mol2</a>.
<br><br>
<li><a href="filetypes.html#alignment">Sequence alignments</a> can be
generated in Chimera
 <ul>
 <li>by sequence search and retrieval of hits with
<a href="blast.html"><b>Blast Protein</b></a> (pseudo-multiple)
 <li>by <a href="../ContributedSoftware/matchalign/matchalign.html"><b>Match -> Align</b></a> from the spatial overlap of superimposed structures (multiple)
 <li>as a by-product of superposition with
<a href="../ContributedSoftware/matchmaker/matchmaker.html"><b>MatchMaker</b></a> (pairwise)
 </ul>
 and <a href="../ContributedSoftware/multalignviewer/multalignviewer.html#editing">edited and saved</a> 
 using <a href="../ContributedSoftware/multalignviewer/framemav.html"
 target="_top"><b>Multalign Viewer</b></a>.
<br><br>
<li><a href="filetypes.html#volume">Volume data</a> 
(<a name="maps">maps</a>) can be modified or generated in Chimera by
 <ul>
 <li><a href="../ContributedSoftware/volumeviewer/framevolumeviewer.html"
 target="_top"><b>Volume Viewer</b></a> (see
 <a href="../ContributedSoftware/volumeviewer/volumeviewer.html#coordinates">
 Coordinates</a>,
 <a href="../ContributedSoftware/volumeviewer/volumeviewer.html#bounds">
 Region bounds</a>,
 <a href="../ContributedSoftware/volumeviewer/volumeviewer.html#mouseselect">
 Subregion selection</a>, and
 <a href="../ContributedSoftware/volumeviewer/volumeviewer.html#zone">Zone</a>)
 <li><a href="../ContributedSoftware/voleraser/voleraser.html">
 <b>Volume Eraser</b></a>
 <li><a href="../ContributedSoftware/volumeviewer/gaussian.html"><b>Volume
 Filter</b></a>
 <li><a href="midas/molmap.html"><b>molmap</b></a> to simulate a map from
 atomic coordinates
 <li>commands <a href="midas/volume.html"><b>volume</b></a>,
 <a href="midas/vop.html"><b>vop</b></a>,
 <a href="midas/mask.html"><b>mask</b></a>
 <li><a href="../ContributedSoftware/accelerators/accelerators.html">keyboard
 shortcuts</a> for
 <a href="../ContributedSoftware/accelerators/alist.html#voledit">volume
 editing</a>
 <li><a href="../ContributedSoftware/fitmaps/fitmaps.html">fitting</a>
 (but see note on <a href="#afterfitting">saving after fitting</a>)
 </ul>
 and saved using 
 <a href="../ContributedSoftware/volumeviewer/framevolumeviewer.html"
 target="_top"><b>Volume Viewer</b></a> (the 
 <a href="../ContributedSoftware/volumeviewer/volumeviewer.html#organization">
 File menu</a>)
 or the command <a href="midas/volume.html"><b>volume</b></a>. 
 Related data:
 <ul>
  <li>markers/paths can be created and saved with
  <a href="../ContributedSoftware/volumepathtracer/framevolpath.html" 
  target="_top"><b>Volume Tracer</b></a>
  <li>segmentations can be created and saved with
  <a href="../ContributedSoftware/segger/segment.html"><b>Segger</b></a>
 </ul>
</ul>
<p>
See also:
<a href="export.html">exporting a scene</a>,
<a href="dms.html#writedms"><b>Write DMS</b></a>,
<a href="../ContributedSoftware/addions/writeprmtop.html"><b>Write 
Prmtop</b></a>
</p>

<h3><a name="pdb">Saving PDB Files</a></h3>
<p>
The dialog for saving <a href="tutorials/framepdbintro.html"
target="_top">PDB</a> files can be invoked with
<b>File... Save PDB</b> or <b>Actions... Write PDB</b>
in the <a href="menu.html">Chimera menu</a>, the <b>Write PDB...</b>
button on the <a href="inspection.html"><b>Selection Inspector</b></a>, or
<a href="modelpanel.html#writepdb"><b>write PDB</b></a> 
in the <a href="modelpanel.html"><b>Model Panel</b></a>.
The dialog resembles other <a href="opensave.html">open/save dialogs</a>,
but has additional specific options.
See also: <a href="midas/write.html"><b>write</b></a> 
</p><p>
<a name="chosen">Individual models or blocks of models can be chosen</a> 
from the <b>Save models</b> list with the left mouse button.  
Ctrl-click toggles the status of an individual model.
To choose a block of models without dragging,
click on the first (or last) and then Shift-click 
on the last (or first) in the desired block.
Only molecule models are listed.
</p><p>
Options:
<ul>
<li> <b>Save displayed atoms only</b> - only write coordinates of atoms 
that are displayed
<li> <b>Save selected atoms only</b> - only write coordinates of atoms 
that are <a href="selection.html">selected</a>
</ul>
Option when only one model is present:
<ul>
<li> <b>Use untransformed coordinates</b> - write coordinates without
applying any rotations and translations that have been performed in Chimera
</ul>
Option when more than one model is present:
<ul>
<li> <b>Save relative to model [<i>model</i>]</b> -
save coordinates relative to the untransformed coordinates of a
specified reference <i>model</i> 
(otherwise, transformed coordinates will be saved).
This is useful for preserving the spatial relationship between models.  
For example, if model 1 has been matched or docked to model 0, 
saving 1 relative to 0 results in the models being matched 
or docked in the same way when the model 0 file (original) and 
the model 1 file (saved relative to 0) are reopened.
</ul>
Option when more than one model is <a href="#chosen">chosen</a>:
<ul>
<li> <b>Save multiple models in</b>
  <ul>
  <li> <b>a single file</b> 
  - one PDB file containing all chosen models, with
  <tt>MODEL/ENDMDL</tt> records delimiting the coordinates of each.
  Technically, this treatment is only correct for models with identical 
  sets of atoms, but may be convenient in the more general case of disparate
  models.  In the case of multiple models containing different sets of atoms,
  the <tt>CONECT</tt> records should be removed before the file is 
  read back into Chimera, as there is no mechanism for handling the
  different <tt>CONECT</tt> records for the different models.
  <li> <b>multiple files [file name must contain $name or $number]</b> 
  - a different PDB file for each model, named by substituting
  the model name for <b>$name</b> and/or the model ID number 
  (<b><i>model</i></b> or <b><i>model.submodel</i></b>) for <b>$number</b> 
  in the specified file name;
  if this substitution does not produce unique file names, 
  each file will contain only the last appropriate model
  </ul>
</ul>
If one or more of the chosen models contains a trajectory, there will also be
an option to <b>Save</b> the <b>current frame</b> or <b>all frames</b>.
In this context, "all frames" means all frames whose coordinate sets have been
<a href="../ContributedSoftware/movie/movie.html#framesread">read in</a>
with <a href="../ContributedSoftware/movie/framemovie.html"
target="_top"><b>MD Movie</b></a>.
A model will not be recognized as a trajectory until more than one
set of coordinates has been read.
</p><p>
Besides atomic coordinates, 
<a href="tutorials/pdbintro.html#secondary">HELIX and SHEET records</a>
reflecting the current protein <a href="midas/ksdssp.html">secondary structure 
assignments</a> are written, along with any other header lines 
read from PDB input.  Even if protein helix/strand assignments have not been
changed in Chimera, the output HELIX and SHEET records may differ from 
the input because helices are written assuming the right-handed &alpha; type,
and strands are written as if each were a separate sheet.
</p>
<ul>
<li><b>Large-structure serial numbering</b> 
indicates how to accommodate structures so large 
that the numbering fields in standard PDB format would overflow
(output for smaller structures is not affected):
<ul>
<li><b>Amber</b> handles atomic models
with &gt; 99,999 atoms by &ldquo;stealing&rdquo; a column to the
left of the atom numbering field in standard PDB format
<li><b>hybrid-36</b> (default) handles atomic models
with &gt; 99,999 atoms (up to 87,440,031)
and/or &gt; 9999 residues in a chain (up to 2,436,111)
by incorporating alphabetical characters
(<a href="https://cci.lbl.gov/hybrid_36/" target="_blank">details...</a>)
</ul>
</ul>

<h3><a name="mol2">Saving Mol2 Files</a></h3>
<p>
<a name="mol2info"></a>
Mol2 format
includes atom types and (optionally) partial charges.
When a structure is read from a Mol2 file, the
atom types and any partial charges are stored as the atom
<a href="../ContributedSoftware/defineattrib/defineattrib.html#attribdef">attributes</a> 
<b>mol2type</b> and <b>charge</b>.
The input bond orders are also stored.
Unless the user specifies <a href="#writegaff">writing Amber/GAFF types
instead</a>, output Mol2 files include Sybyl atom types.
The Sybyl types read from input will be used, if available; 
otherwise, <a href="idatm.html">Chimera atom types</a> will be translated
into Sybyl atom types.
As there is not a simple one-to-one relationship between Chimera and Sybyl
atom types, <b><i>users may wish to check the Sybyl type assignments</i></b>,
particularly for nonstandard residues.  
Similarly, if bond orders were not read from Mol2 input, they will be
guessed upon output.  
Values for the <b>charge</b>
<a href="../ContributedSoftware/defineattrib/defineattrib.html#attribdef">attribute</a>
can be assigned within Chimera, for example using
<a href="../ContributedSoftware/addcharge/addcharge.html"><b>Add 
Charge</b></a>.  If no <b>charge</b> values have been read or assigned, 
the partial charges will be written as zero.
</p><p>
The dialog for saving <a href="#mol2info">Mol2</a> files can be invoked with
<b>File... Save Mol2</b> in the <a href="menu.html">Chimera menu</a>.
The dialog resembles other <a href="opensave.html">open/save dialogs</a>,
but has additional specific options.
See also: <a href="midas/write.html"><b>write</b></a> 
</p><p>
<a name="chosen2">Individual models or blocks of models can be chosen</a> 
from the <b>Save models</b> list with the left mouse button.  
Ctrl-click toggles the status of an individual model.
To choose a block of models without dragging,
click on the first (or last) and then Shift-click 
on the last (or first) in the desired block.
Only molecule models are listed.
</p><p>
Option when only one model is present:
<ul>
<li> <b>Use untransformed coordinates</b> - write coordinates without
applying any rotations and translations that have been performed in Chimera
</ul>
Option when more than one model is present:
<ul>
<li> <b>Save relative to model [<i>model</i>]</b> - save coordinates
relative to the untransformed coordinates of a specified 
reference <i>model</i> (otherwise, transformed coordinates will be saved)
</ul>
Option when more than one model is <a href="#chosen2">chosen</a>:
<ul>
<li> <b>Save multiple models in</b>
  <ul>
  <li> <b>a single file [individual @MOLECULE sections]</b> 
  <li> <b>multiple files [file name must contain $name or $number]</b> 
  - a different Mol2 file for each model, named by substituting
  the model name for <b>$name</b> and/or the model ID number 
  (<b><i>model</i></b> or <b><i>model.submodel</i></b>) for <b>$number</b> 
  in the specified file name;
  if this substitution does not produce unique file names,
  each file will contain only the last appropriate model
  <li> <b>a single file [combined @MOLECULE section]</b> 
  - if the models are all single-residue and the model names are unique 
  but the residue types are not, the model names will be used as 
  substructure names, and the option to include residue numbers in
  substructure names (below) will be ignored
  </ul>
</ul>
Further options:
<ul>
<li><b>Use Sybyl-style hydrogen naming (e.g. HE12 rather than 2HE1)</b>
- move leading numerals to the ends of hydrogen names
<li><b>Include residue sequence numbers in substructure names</b>
- generate substructure names with embedded residue numbers,
for example, ALA85 rather than ALA
<li><a name="writegaff">
<b>Write Amber/GAFF atom types instead of Sybyl atom types</b></a>
- include the Amber/GAFF atom types assigned by
<a href="../ContributedSoftware/addcharge/addcharge.html"><b>Add Charge</b></a>
(or command <a href="midas/addcharge.html"><b>addcharge</b></a>) instead of
Sybyl atom types; if any atom in the <a href="#chosen">chosen</a> models
lacks an Amber/GAFF type assignment, file-writing will halt and
an error message will be displayed.  <b><i>Mol2 files with Amber/GAFF types
are not suitable for reading back into Chimera, as the types are
likely to be misinterpreted.</i></b>
<li><b>Write current selection to @SET section of file</b>
- make the currently 
<a href="selection.html">selected</a> atoms a SET 
(as used to specify the rigid portion of a ligand in
<a href="http://dock.compbio.ucsf.edu/index.html" target="_blank">DOCK</a>)
</ul>
</p>

<h3><a name="afterfitting">Saving Maps after Fitting</a></h3>
<p>
Simply saving a map to a file after 
<a href="../ContributedSoftware/fitmaps/fitmaps.html">fitting</a>
will not save its new position, because the map file formats 
do not include rotation information.
One way to save the new position is to save a Chimera
<a href="sessions.html">session</a>, which preserves
the translations and orientations of all models.
However, if data files are needed for use in another program:
<ul>
<li>When a map has been fit to an atomic structure,
<a href="#pdb">save the PDB file</a> relative to the map 
rather than <i>vice versa</i>.  
<li>When a map has been fit to another map, a new map can be created
in the new orientation by resampling the map that was fit, on the grid
of the map to which it was fit.  Resampling can be performed with
the command <a href="midas/vop.html#resample"><b>vop resample</b></a>.
Resampling uses interpolation and may cause some loss of resolution.
The new, resampled map can be saved as described <a href="#maps">above</a>.
(Even if the map was not fit to another map, however,
a separate zero-valued map to serve as a grid for resampling
can be created with <a href="midas/vop.html#new"><b>vop new</b></a>.)
</ul>
If the goal is to save a map repositioned with translation only 
(not rotation), the map origin can be changed in the
<a href="../ContributedSoftware/volumeviewer/volumeviewer.html#coordinates">
Coordinates</a> section of
<a href="../ContributedSoftware/volumeviewer/framevolumeviewer.html"
target="_top"><b>Volume Viewer</b></a> 
or with the <a href="midas/volume.html"><b>volume</b></a> command
(see <a href="midas/volume.html#dimensions">Dimension and Scale Options</a>)
before saving.
However, not all programs will use the origin specified in the map file,
so it may still be necessary to resample as described above.
</p>
<hr>
<address>UCSF Computer Graphics Laboratory / April 2018</address>
</body></html>
