<!--
--- UCSF Chimera Copyright ---
Copyright (c) 2006-2018 Regents of the University of California.
All rights reserved.  This software provided pursuant to a
license agreement containing restrictions on its disclosure,
duplication and use.  This notice must be embedded in or
attached to all copies, including partial copies, of the
software or any revisions or derivations thereof.
--- UCSF Chimera Copyright ---
-->

<html>
<head>
<TITLE>Add Charge</TITLE>
</head>
<body>
<h3>Add Charge <img src="addchargeicon.png" alt="Add Charge icon"></h3>

<table border="1" align="right" style="margin:8px 8px 8px 8px"
cellpadding="0" cellspacing="0">
<tr><td align="center"><img src="charges.png"></td></tr>
<tr><td align="center"><font size="-1">
Atoms can be <a href="#chargelabels">labeled</a> to show charge values.
</font></td></tr>
</table>

<p>
<b>Add Charge</b> assigns atomic partial charges and Amber/GAFF atom types
as the <a href="../defineattrib/defineattrib.html#attribdef">attributes</a>
<b>charge</b> and <b>gaffType</b>, respectively. 
<ul>
<li>Atoms in <b><i>standard residues</i></b>
(water, standard amino acids, standard nucleic acids, 
and a few common variants and capping groups) 
are assigned charges and types from
<a href="https://ambermd.org/" target="_blank"><b>Amber</b></a>
(<a href="#standard">details...</a>).
<li>Except for <b><i>monatomic ions</i></b>,
which are simply assigned their net charges,
atoms in <b><i>nonstandard residues</i></b>
are assigned charges and types determined by
<a href="https://ambermd.org/antechamber/antechamber.html" 
target="_blank"><b>Antechamber</b></a>, which is included with Chimera
(<a href="#antechamber">details...</a>).
</ul>
Atomic partial charges (and optionally, Amber/GAFF atom types) are included 
in saved <a href="../../UsersGuide/savemodel.html#mol2">Mol2 files</a>.
The charges and Amber/GAFF atom types are used by other Chimera tools such as
<a href="../minimize/minimize.html"><b>Minimize Structure</b></a>.
</p><p>
Charge assignment requires explicit hydrogens, 
which can be added beforehand with <a href="../addh/addh.html"><b>AddH</b></a>.
Structures often have problems
such as truncated sidechains that can lead to non-integral net charges.
The <a href="../dockprep/dockprep.html"><b>Dock Prep</b></a> tool
includes options for fixing such problems prior to running
<a href="../addh/addh.html"><b>AddH</b></a> and <b>Add Charge</b>.
See also:
<a href="../apbs/pdb2pqr.html"><b>PDB2PQR</b></a>,
<a href="../coulombic/coulombic.html"><b>Coulombic Surface Coloring</b></a>,
<a href="../apbs/apbs.html"><b>APBS</b></a>,
<a href="../addions/addions.html"><b>Add Ions</b></a>,
<a href="../addions/writeprmtop.html"><b>Write Prmtop</b></a>,
<a href="#input">reading charges from files</a>
</p><p>
There are <a href="../../UsersGuide/extension.html">several ways to start</a>
<b>Add Charge</b>, a tool in the <b>Structure Analysis</b> and
<b>Structure Editing</b> categories.
<b>Add Charge</b> is also implemented as the command
<a href="../../UsersGuide/midas/addcharge.html"><b>addcharge</b></a>.
</p><p>
Models to process can be chosen from the list with the left mouse button.  
Ctrl-click toggles the status of an individual model.
To choose a block of models without dragging,
click on the first (or last) and then Shift-click 
on the last (or first) in the desired block.
</p><p>
<b>Standard residues</b> can be assigned parameters from any of the following
force fields (<a href="#standard">details</a>):
</p>
<ul>
<li><b>AMBER ff14SB</b> (default)
<li><b>AMBER ff03ua</b>
- treats amino acid sidechain aliphatic hydrogens as if collapsed onto the 
corresponding carbons; these &ldquo;extra&rdquo; hydrogens will be removed
(however, the united-atom approach is not supported by
<a href="../minimize/minimize.html"><b>Minimize Structure</b></a>,
see <a href="#note2">note</a>)
<li><b>AMBER ff03.r1</b>
<li><b>AMBER ff02pol.r1</b> 
(however, polarizability is not supported by
<a href="../minimize/minimize.html"><b>Minimize Structure</b></a>,
see <a href="#note">note</a>)
<li><b>AMBER ff99SB</b> (deprecated)
<li><b>AMBER ff99bsc0</b> (deprecated)
</ul>
<p>
Monatomic ions are simply assigned their net charges.
<b>Other residues</b> can be handled with either of the following
<a name="antechamber-method">charge calculation methods</a>
(<a href="#antechamber">details</a>):
<ul>
<li> <b>AM1-BCC</b> - semi-empirical (AM1) with bond charge correction (BCC)
<li> <b>Gasteiger</b> - faster method based on atom types and connectivity
</ul>
<a name="chargelabels">
The partial atomic charges can be shown as labels in
<b>nonstandard residues</b> and/or <b>standard residues</b>.</a>
Labels will be shown for displayed atoms only.
If charge labels are not shown with <b>Add Charge</b>,
they can still be shown at some later point
with <a href="../../UsersGuide/menu.html#actlabother"><b>Actions...
Label... other</b></a> (label with attribute <b>charge</b>) or commands:
<blockquote>
<i>Command</i>: <b><a href="../../UsersGuide/midas/labelopt.html">labelopt</a> 
info charge</b>
<br>
<i>Command</i>: <b><a href="../../UsersGuide/midas/label.html">label</a></b>
</blockquote>
Some subset of the atoms could be specified with the
<b><a href="../../UsersGuide/midas/label.html">label</a></b> command,
whereas the above would label all displayed atoms.
</p><p>
<b>OK</b> initiates processing and dismisses the dialog, <b>Apply</b>
initiates processing without dismissing the dialog, and <b>Close</b>
simply dismisses the dialog.  <b>Help</b> opens this manual page in a
browser window.
</p><p>
<a href="../addh/addh.html"><b>AddH</b></a> will be called as needed to
add hydrogens.  
<a name="protonation">
Potentially ambiguous or rare (shifted-pKa) protonation states,
especially in binding sites and nonstandard residues, 
should be verified and corrected before charges are assigned.  
</a>
For example,
extra hydrogens can be <a href="../editing/editing.html#deletion">deleted</a>,
and <a href="../../UsersGuide/idatm.html">atom types</a> can be edited 
(before hydrogen addition)
with <a href="../../UsersGuide/midas/setattr.html"><b>setattr</b></a>
or <a href="../editing/editing.html"><b>Build Structure</b></a>.
</p><p>
If there are any nonstandard residues,
a dialog will appear for specifying their net charges.
Names of the form RES1+RES2 indicate covalently bonded residues
that will be treated as a single unit, and the net charge is for the
combined entity.
The <a href="#antechamber-method">charge calculation method</a>
can be changed at this point, if desired. 
</p><p>
<a name="terminalP">
If a nucleic acid chain has a 5' terminal phosphate</a>,
the user will be asked whether this group should be deleted;
otherwise, its atoms will be assigned charges of zero
(the charge sets lack parameters for 5' phosphates).
</p><p>
<a name="warnings">
A warning will appear</a> if the name of any atom in a standard residue
is not recognized or a model's net charge is not an integer; 
details will be reported in the 
<a href="../../UsersGuide/reply.html"><b>Reply Log</b></a>.
Cases of unrecognized atoms in standard residues and/or incorrect net charges
should be examined and <a href="#resolved">resolved</a>.
Note that chain-terminal nucleotide residues will normally
have non-integral charges, but the 5' and 3' charges sum to an integer.
</p>

<a name="standard">
<h3>Details for Standard Residues</h3></a>
<p>
Atoms in standard residues (water, standard amino acids, 
standard nucleic acids, and a few common variants and capping groups)
are assigned charges and atom types taken from Amber parameter files. 
Precalculated charges are also used for
<b>ATP</b>, <b>ADP</b>, <b>GTP</b>, <b>GDP</b>, <b>NAD</b>, and <b>NDP</b>,
as described <a href="#antechamber">below</a>.
</p><p>
<b>Charge model</b> corresponds to force field version
(see the <a href="https://ambermd.org/" 
target="_blank"><b>Amber</b></a> documentation for further details):
<ul>
<li><b>AMBER ff14SB</b> 
(default; used when <b>Add Charge</b> is called from 
<a href="../coulombic/coulombic.html"><b>Coulombic Surface Coloring</b></a> 
or by the command
<a href="../../UsersGuide/midas/minimize.html"><b>minimize</b></a> with
<b>nogui true</b>)
- charges from amino12.lib, amino12nt.lib, amino12ct.lib, nucleic12.lib
<blockquote>
D.A. Case, V. Babin, J.T. Berryman, R.M. Betz, Q. Cai, D.S. Cerutti, 
T.E. Cheatham, III, T.A. Darden, R.E. Duke, H. Gohlke, A.W. Goetz, S. Gusarov, 
N. Homeyer, P. Janowski, J. Kaus, I. Kolossv&aacute;ry, A.  Kovalenko,
T.S. Lee, S. LeGrand, T. Luchko, R. Luo, B. Madej, K.M. Merz, F. Paesani, 
D.R. Roe, A. Roitberg, C. Sagui, R. Salomon-Ferrer, G. Seabra, C.L. Simmerling,
W. Smith, J. Swails, R.C. Walker, J. Wang, R.M. Wolf, X. Wu, and P.A. Kollman 
(2014), AMBER 14, University of California, San Francisco.
</blockquote>
Improvements in <b>ff14SB</b> relative to
<a href="#deprecated"><b>ff99</b> versions</a>
include dihedral corrections for proteins and
updated parameters for nucleic acids.
<!--
p. 32 of the Amber 14 manual suggests that different modifications may be 
optimal for DNA, but it's unclear which frcmod files would be used
-->
<br><br>
<li><b>AMBER ff03ua</b>
- charges from
uni_amino03.lib, uni_aminont03.lib,
uni_aminoct03.lib, all_nucleic94.lib
<blockquote>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/16805629" target="_blank">
New-generation amber united-atom force field.</a>
Yang L, Tan CH, Hsieh MJ, Wang J, Duan Y, Cieplak P, 
Caldwell J, Kollman PA, Luo R.
<i>J Phys Chem B.</i> 2006 Jul 6;110(26):13166-76.
</blockquote>
This &ldquo;united-atom&rdquo; force field treats amino acid sidechain 
aliphatic hydrogens as if they were collapsed onto the corresponding carbons.  
All nucleic acid hydrogens and amino acid backbone, polar, and aromatic 
hydrogens are still explicit.
<b>Add Charge</b> will remove the &ldquo;extra&rdquo; hydrogens 
(those which are to be treated implicitly).
<a name="note2"></a>
** <a href="../minimize/minimize.html"><b>Minimize Structure</b></a> 
does not support the united-atom force field; this option is mainly useful
for assigning charges and types before 
<a href="../addions/writeprmtop.html">writing a prmtop file</a>
for <a href="https://www.ambermd.org/" target="_blank">Amber</a>. **
<br><br>
<li><b>AMBER ff03.r1</b>
- charges from all_amino03.lib, all_aminont03.lib, all_aminoct03.lib,
all_nucleic94.lib
(<b>r1</b> refers to updated charge sets for N- and C-terminal amino acids;
the original <b>ff03</b> used all_aminont94.lib and all_aminoct94.lib).
Original <b>ff03</b> reference:
<blockquote>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/14531054" target="_blank">
A point-charge force field for molecular mechanics simulations of proteins 
based on condensed-phase quantum mechanical calculations.</a>
Duan Y, Wu C, Chowdhury S, Lee MC, Xiong G, Zhang W, Yang R, Cieplak P, 
Luo R, Lee T, Caldwell J, Wang J, Kollman P.
<i>J Comput Chem.</i> 2003 Dec;24(16):1999-2012.
</blockquote>
<a name="note"></a>
<li><b>AMBER ff02pol.r1</b>
- charges from
all_amino02.lib, all_aminont02.lib,
all_aminoct02.lib, all_nucleic02.lib
(<b>r1</b> uses atom type H0 in glycine).
** <a href="../minimize/minimize.html"><b>Minimize Structure</b></a> 
does not support the polarizable force field; this option is mainly useful
for assigning charges and types before 
<a href="../addions/writeprmtop.html">writing a prmtop file</a>
for <a href="https://www.ambermd.org/" target="_blank">Amber</a>. **
<blockquote>
Molecular mechanical models for organic and biological systems going
beyond the atom centered two body additive approximation.
Cieplak P, Caldwell J, Kollman P.
<i>J Comput Chem.</i> 2001 Jul 30;22(10):1048-57.
<br><br>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/16526038" target="_blank">
Strike a balance: optimization of backbone torsion parameters of 
AMBER polarizable force field for simulations of proteins and peptides.</a>
Wang ZX, Zhang W, Wu C, Lei H, Cieplak P, Duan Y.
<i>J Comput Chem.</i> 2006 Apr 30;27(6):781-90. 
</blockquote>
<a name="deprecated"></a>
<li><b>AMBER ff99SB</b> (deprecated)
- charges from all_amino94.lib, all_aminont94.lib,
all_aminoct94.lib, all_nucleic94.lib, same as the original <b>ff99</b>:
<blockquote>
A second generation force field for the simulation of proteins, nucleic
acids, and organic molecules.
Cornell WD, Cieplak P, Bayly CI, Gould IR, Merz Jr KM,
Ferguson DM, Spellmeyer DC, Fox T, Caldwell JW, Kollman PA.
<i>J Am Chem Soc.</i> 1995 May;117(19):5179-97.
</blockquote>
<li><b>AMBER ff99bsc0</b> (deprecated;
same charges as above, but uses atom type CI in nucleic acids)
<blockquote>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/17351000" target="_blank">
Refinement of the AMBER force field for nucleic acids:
improving the description of &alpha;/&gamma; conformers.</a>
P&eacute;rez A, March&aacute;n I, Svozil D, Sponer J, Cheatham TE 3rd, 
Laughton CA, Orozco M.
<i>Biophys J.</i> 2007 Jun 1;92(11):3817-29.
</blockquote>
</ul>
<a name="alterstates">
The parameters include alternative sidechain protonation states
for some standard residues.</a>  By default,
<a href="../addh/addh.html"><b>AddH</b></a> adds hydrogens to 
generate the states shown in <b>bold</b> below, although it does not remove
hydrogens already present.  The nondefault states can be attained by
<a href="../editing/editing.html#deletion">deleting</a> hydrogens
and/or editing <a href="../../UsersGuide/idatm.html">atom types</a> 
(with <a href="../../UsersGuide/midas/setattr.html"><b>setattr</b></a>
or <a href="../editing/editing.html"><b>Build Structure</b></a>)
before adding hydrogens:
  <ul>
  <li> aspartic acid <b>negative</b> or neutral
     <ul>
     <li> charges for the neutral ASH residue are assigned if one of the
     carboxylic acid oxygens (preferably OD2 for the proper charge distribution)
     has <a href="../../UsersGuide/idatm.html">atom type</a> <b>O3</b> 
     and an attached hydrogen
     </ul>
  <li> glutamic acid <b>negative</b> or neutral
     <ul>
     <li> charges for the neutral GLH residue are assigned if one of the
    carboxylic acid oxygens (preferably OE2 for the proper charge distribution)
    has <a href="../../UsersGuide/idatm.html">atom type</a> <b>O3</b> 
    and an attached hydrogen
     </ul>
  <li> cysteine <b>neutral reduced</b>, thiolate (negatively charged), or
  <b>half-cystine</b> (if there is a disulfide bond)
     <ul>
     <li> charges for the negative CYM residue are assigned if there is
     no disulfide bond and no atom named HG
     </ul>
  <li> lysine <b>positive</b> or neutral 
     <ul>
     <li> charges for the neutral LYN residue are assigned if the atom named 
     NZ has <a href="../../UsersGuide/idatm.html">atom type</a> <b>N3</b> 
     and thus two attached hydrogens rather than three
     </ul>
  <li> histidine protonated at &delta;-nitrogen (neutral), 
  &epsilon;-nitrogen (neutral), or both (positive)
     <ul>
     <li>in <a href="../addh/addh.html"><b>AddH</b></a>, the
      <a href="../addh/addh.html#hisprot">histidine protonation states</a>
      can be user-specified or guessed based on the local environment,
      and in <a href="../coulombic/coulombic.html"><b>Coulombic Surface 
      Coloring</b></a>, they can be user-specified;
      <b>Add Charge</b> recognizes the state from the presence of hydrogen
      atom HD1, HE2, or both
     </ul>
  </ul>
Water charges correspond to the TIP3P model:
<blockquote>
Comparison of simple potential functions for simulating liquid water.
Jorgensen WL, Chandrasekhar J, Madura JD.
<i>J Chem Phys.</i> 1983 Jul 15;79(2):926-35.
</blockquote>
Note different solvent charge sets can be obtained using 
<a href="../solvate/solvate.html"><b>Solvate</b></a>.
Charges previously assigned to solvent atoms by
<a href="../solvate/solvate.html"><b>Solvate</b></a> 
are not overwritten by <b>Add Charge</b>.
However,
<a href="../minimize/minimize.html"><b>Minimize Structure</b></a>
will still use the TIP3P model for the remaining water parameters.
</p><p>
<a name="resolved">
Cases of unrecognized atoms in standard residues and/or incorrect net charges
should be examined and resolved</a>. Approaches include:
  <ul>
  <li> editing the input structure file to contain standard atom names 
  that will be recognized
  <li> since hydrogens are frequently the problem, deleting and re-adding 
  hydrogens in Chimera may suffice, for example with commands:
  <blockquote>
  <b><a href="../../UsersGuide/midas/delete.html">delete</a> H
  <br><a href="../../UsersGuide/midas/addh.html">addh</a></b>
  </blockquote>
  <li> using <a href="../dockprep/dockprep.html"><b>Dock Prep</b></a>
  to perform various fixes on a structure before charge assignment
  <li> adding missing atoms with
   <a href="../editing/editing.html"><b>Build Structure</b></a>
  <li> <a href="../editing/editing.html#deletion">deleting</a> 
  unrecognized atoms that are deemed unnecessary for later calculations
  <li> deciding it is acceptable for the unrecognized atoms to have zero charge
  <li> assigning charges directly with 
  <a href="../../UsersGuide/midas/setattr.html"><b>setattr</b></a>,
  <a href="../../UsersGuide/midas/defattr.html"><b>defattr</b></a>, or
  <a href="../defineattrib/defineattrib.html"><b>Define Attribute</b></a>
  </ul>
</p>

<a name="antechamber">
<h3>Nonstandard Residues and Antechamber</h3></a>
<p>
Monatomic ions are simply assigned their net charges, and
<b>ATP</b>, <b>ADP</b>, <b>GTP</b>, <b>GDP</b>, <b>NAD</b>, and <b>NDP</b>
are assigned charges from the 
<a href="http://research.bmh.manchester.ac.uk/bryce/amber"
target="_blank">Amber parameter database</a>.
<!-- for FMN/FAD, see 
https://plato.cgl.ucsf.edu/pipermail/chimera-users/2017-February/013161.html
-->
For other residues not classified as <a href="#standard">standard</a>,
charges and GAFF atom types are determined using Amber's 
<a href="https://ambermd.org/antechamber/antechamber.html" 
target="_blank"><b>Antechamber</b></a> module, which is included with Chimera.
The available charge calculation methods are:
  <ul>
  <li> <b>AM1-BCC</b> - semi-empirical (AM1) with bond charge correction
  (BCC), parametrized to reproduce <i>ab initio</i> (HF/6-31G*) 
  electrostatic potentials 
  <li> <b>Gasteiger</b> - iterative partial equalization of orbital
   electronegativity
  </ul>
While both methods are much faster than <i>ab initio</i> calculations, 
the Gasteiger method is the faster and more approximate of the two.
</p><p>
Note that Antechamber/GAFF are meant to handle most small organic molecules,
but not metal complexes, inorganic compounds, or unstable species such as
radicals, and may not work well on highly charged molecules.
GAFF allows for parametrization of most organic molecules made of
C, N, O, H, S, P, F, Cl, Br and I.
</p><p>
Publications involving
<a href="https://ambermd.org/antechamber/antechamber.html" 
target="_blank"><b>Antechamber</b></a> use should cite:
<a name="antechamber-ref"></a>
<blockquote>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/16458552" target="_blank">
Automatic atom type and bond type perception in molecular mechanical 
calculations.</a>
Wang J, Wang W, Kollman PA, Case DA.
<i>J Mol Graph Model.</i> 2006 Oct;25(2):247-60.
</blockquote>
GAFF atom types and associated parameters are described
<a href="https://ambermd.org/antechamber/gaff.html" target="_blank">online</a>
and in:
<blockquote>
<a href="https://www.ncbi.nlm.nih.gov/pubmed/15116359" target="_blank">
Development and testing of a general amber force field.</a>
Wang J, Wolf RM, Caldwell JW, Kollman PA, Case DA.
<i>J Comput Chem.</i> 2004 Jul 15;25(9):1157-74.
</blockquote>
</p>

<a name="sources">
<h3>Additional Parameter Sources</h3></a>
<p>
Additional sources of charges and other parameters include:
  <ul>
  <li><a href="http://research.bmh.manchester.ac.uk/bryce/amber"
  target="_blank">Amber parameter database</a> (Bryce Group,
  University of Manchester)
  <li> <a href="https://upjv.q4md-forcefieldtools.org/REDDB/index.php"
  target="_blank">REDDB</a> and the
  <a href="https://upjv.q4md-forcefieldtools.org/REDServer/"
  target="_blank">RED Server</a>
  (Universit&eacute; de Picardie Jules Verne
  and Burnham Institute for Medical Research)
  </ul>
</p>

<a name="input">
<h3>Reading Charges from Files</h3></a>
<p>
<b>Attribute assignment</b>.
Charges can be specified arbitrarily in a simple text
<a href="../defineattrib/defineattrib.html#attrfile">attribute 
assignment file</a>, read in with
<a href="../defineattrib/defineattrib.html"><b>Define Attribute</b></a> or
the command <a href="../../UsersGuide/midas/defattr.html"><b>defattr</b></a>.
</p><p>
<b>PQR</b>.
The <a href="../../UsersGuide/tutorials/pdbintro.html#PQR">PQR variant</a>
of <a href="../../UsersGuide/tutorials/framepdbintro.html"
target="_top">PDB format</a> includes atomic partial charges
and radii, which Chimera assigns as the atom
<a href="../defineattrib/defineattrib.html#attribdef">attributes</a>
<b>charge</b> and <b>radius</b>.
</p><p>
<b>Mol2</b>.
When a structure is read from a Mol2 file, the
atom types and any partial charges are stored as the atom
<a href="../defineattrib/defineattrib.html#attribdef">attributes</a> 
<b>mol2type</b> and <b>charge</b>.
</p><p>
<b>PubChem SDF</b>.
When a structure is read from a 3D SDF 
<a href="../../UsersGuide/fetch.html">fetched</a> or obtained directly from
<a href="https://pubchemdocs.ncbi.nlm.nih.gov/pubchem3d"
target="_blank"><b>PubChem3D</b></a>,
<!-- <PUBCHEM_MMFF94_PARTIAL_CHARGES> -->
the partial charges are stored as the atom
<a href="../defineattrib/defineattrib.html#attribdef">attribute</a>
named <b>charge</b>.  
</p><p>
<b>Topology files</b>.
Partial charges in some of the
<a href="../movie/movie.html#movieinput">topology formats</a> (Amber, PSF) 
input to <a href="../movie/framemovie.html" target="_top"><b>MD Movie</b></a>
are stored as the atom
<a href="../defineattrib/defineattrib.html#attribdef">attribute</a> 
named <b>charge</b>.
</p>
<hr>
<address>UCSF Computer Graphics Laboratory / January 2018</address>
</body>
</html>
